{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Sequencing data analysis\n",
    "\n",
    "### IMPORTANT: Please make sure that your are using the bash kernel to run this notebook.\n",
    "\n",
    "### This notebook covers analysis of DNA sequencing data from raw files to processed signals.\n",
    "\n",
    "Although this analysis is for ATAC-seq data, many of the steps (especially the first section) are the same for other types of DNA sequencing experiments.\n",
    "\n",
    "We'll be doing the analysis in Bash, which is the standard language for UNIX command-line scripting."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Part 1: Setting up the data\n",
    "\n",
    "We start with raw `.fastq.gz` files, which are provided by the sequencing instrument. For each DNA molecule (read) that was sequenced, they provide the nucleotide sequence, and information about the quality of the signal of that nucleotide."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": []
    }
   ],
   "source": [
    "### Set up variables storing the location of our data\n",
    "##These are the same as you generated in Notebook 1.3\n",
    "##If you have already logged in and out of jupyterhub, this step may be redundant, be we add it in for good measure. \n",
    "\n",
    "export SUNETID=\"$(whoami)\"\n",
    "export WORK_DIR=\"/tc2016/${SUNETID}\"\n",
    "export DATA_DIR=\"${WORK_DIR}/data\"\n",
    "export FASTQ_DIR=\"${DATA_DIR}/fastq/\"\n",
    "export SRC_DIR=\"${WORK_DIR}/src\"\n",
    "export ANALYSIS_DIR=\"${WORK_DIR}/analysis/\"\n",
    "export YEAST_DIR=\"/srv/scratch/training_camp/saccer3/seq\"\n",
    "export YEAST_INDEX=\"/srv/scratch/training_camp/saccer3/bowtie2_index/saccer3\"\n",
    "export YEAST_CHR=\"/srv/scratch/training_camp/saccer3/sacCer3.chrom.sizes\"\n",
    "export TMP=\"${WORK_DIR}/tmp\"\n",
    "export TEMP=$TMP \n",
    "export TMPDIR=$TMP\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now, let's check exactly which fastqs we have:\n",
    "\n",
    "(recall that the `ls` command lists the contents of a directory)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Ct-1h_S5_L001_R1_001.fastq.gz\t DMSO-3h_S13_L001_R1_001.fastq.gz\r\n",
      "Ct-1h_S5_L001_R2_001.fastq.gz\t DMSO-3h_S13_L001_R2_001.fastq.gz\r\n",
      "Ct-3h_S12_L001_R1_001.fastq.gz\t Kz-1h_S1_L001_R1_001.fastq.gz\r\n",
      "Ct-3h_S12_L001_R2_001.fastq.gz\t Kz-1h_S1_L001_R2_001.fastq.gz\r\n",
      "Cu-1h_S4_L001_R1_001.fastq.gz\t Kz-3h_S8_L001_R1_001.fastq.gz\r\n",
      "Cu-1h_S4_L001_R2_001.fastq.gz\t Kz-3h_S8_L001_R2_001.fastq.gz\r\n",
      "Cu-3h_S11_L001_R1_001.fastq.gz\t Mz-1h_S3_L001_R1_001.fastq.gz\r\n",
      "Cu-3h_S11_L001_R2_001.fastq.gz\t Mz-1h_S3_L001_R2_001.fastq.gz\r\n",
      "Cz-1h_S2_L001_R1_001.fastq.gz\t Mz-3h_S10_L001_R1_001.fastq.gz\r\n",
      "Cz-1h_S2_L001_R2_001.fastq.gz\t Mz-3h_S10_L001_R2_001.fastq.gz\r\n",
      "Cz-3h_S9_L001_R1_001.fastq.gz\t WT-1h_S7_L001_R1_001.fastq.gz\r\n",
      "Cz-3h_S9_L001_R2_001.fastq.gz\t WT-1h_S7_L001_R2_001.fastq.gz\r\n",
      "DMSO-1h_S6_L001_R1_001.fastq.gz  WT-3h_S14_L001_R1_001.fastq.gz\r\n",
      "DMSO-1h_S6_L001_R2_001.fastq.gz  WT-3h_S14_L001_R2_001.fastq.gz\r\n"
     ]
    }
   ],
   "source": [
    "ls $FASTQ_DIR"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "As a sanity check, we can also look at the size and last edited time of some of the fastqs by addind `-lrth` to the `ls` command:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 112K\r\n",
      "lrwxrwxrwx 1 ubuntu ubuntu 80 Sep 12 02:21 WT-3h_S14_L001_R2_001.fastq.gz -> ../../../../srv/scratch/training_camp/data/fastqs/WT-3h_S14_L001_R2_001.fastq.gz\r\n",
      "lrwxrwxrwx 1 ubuntu ubuntu 80 Sep 12 02:21 WT-3h_S14_L001_R1_001.fastq.gz -> ../../../../srv/scratch/training_camp/data/fastqs/WT-3h_S14_L001_R1_001.fastq.gz\r\n",
      "lrwxrwxrwx 1 ubuntu ubuntu 79 Sep 12 02:21 WT-1h_S7_L001_R2_001.fastq.gz -> ../../../../srv/scratch/training_camp/data/fastqs/WT-1h_S7_L001_R2_001.fastq.gz\r\n",
      "lrwxrwxrwx 1 ubuntu ubuntu 79 Sep 12 02:21 WT-1h_S7_L001_R1_001.fastq.gz -> ../../../../srv/scratch/training_camp/data/fastqs/WT-1h_S7_L001_R1_001.fastq.gz\r\n",
      "lrwxrwxrwx 1 ubuntu ubuntu 80 Sep 12 02:21 Mz-3h_S10_L001_R2_001.fastq.gz -> ../../../../srv/scratch/training_camp/data/fastqs/Mz-3h_S10_L001_R2_001.fastq.gz\r\n",
      "lrwxrwxrwx 1 ubuntu ubuntu 80 Sep 12 02:21 Mz-3h_S10_L001_R1_001.fastq.gz -> ../../../../srv/scratch/training_camp/data/fastqs/Mz-3h_S10_L001_R1_001.fastq.gz\r\n",
      "lrwxrwxrwx 1 ubuntu ubuntu 79 Sep 12 02:21 Mz-1h_S3_L001_R2_001.fastq.gz -> ../../../../srv/scratch/training_camp/data/fastqs/Mz-1h_S3_L001_R2_001.fastq.gz\r\n",
      "lrwxrwxrwx 1 ubuntu ubuntu 79 Sep 12 02:21 Mz-1h_S3_L001_R1_001.fastq.gz -> ../../../../srv/scratch/training_camp/data/fastqs/Mz-1h_S3_L001_R1_001.fastq.gz\r\n",
      "lrwxrwxrwx 1 ubuntu ubuntu 79 Sep 12 02:21 Kz-3h_S8_L001_R2_001.fastq.gz -> ../../../../srv/scratch/training_camp/data/fastqs/Kz-3h_S8_L001_R2_001.fastq.gz\r\n"
     ]
    }
   ],
   "source": [
    "ls -lrth $FASTQ_DIR | head"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Let's also inspect the format of one of the fastqs. Notice that each read takes up 4 lines:\n",
    "1. the read name\n",
    "2. the read's nucleotide sequence\n",
    "3. a '+' to indicate the record contains another line\n",
    "4. a quality score for each base (a number encoded as a letter)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@M00653:168:000000000-ATAU0:1:2109:21182:1054 1:N:0:5\r\n",
      "CATTTAGTACACCGTTTCCTTCANTTAACCATCCTTAAATCTGTCTCTTATACACATCTCCGAGCCCACGAGACGG\r\n",
      "+\r\n",
      "CCCCCGGGGGGGCFGCFGGGGGG#:C@FFGGGDECGCFGF9,EFE@FGGFGGGGGGGGGGGGDEEFEGGGGGCCFE\r\n",
      "@M00653:168:000000000-ATAU0:1:2109:17670:1054 1:N:0:5\r\n",
      "GTATAGAGACTAGGCAGATCTGANGATCACCTAGCGACTCTCTCCACCGTTTGACGAGGCCATTTACAAAAACATA\r\n",
      "+\r\n",
      "CCCCCFGCCGGG@FDGCCAF<EF#:CFFGGEFGGFGGGDGFF9FGE,FGFFFGGFGGDEEFE,?C?F,AFFFFGEF\r\n",
      "\r\n",
      "gzip: stdout: Broken pipe\r\n"
     ]
    }
   ],
   "source": [
    "zcat $(ls $FASTQ_DIR* | head -n 1) | head -n 8"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Part 2: Adapter trimming\n",
    "\n",
    "- In many kinds of DNA and RNA sequencing experiments, sometimes the sequences will read through the targeted sequence insert and into sequencing adapter or PCR primer sequences on the end of the fragment. When the insert size is shorter than the read length (like in some of our ATAC-seq reads), the adapter sequence is read by the sequencer.\n",
    "\n",
    "- We need to remove such adapter sequences because they won't align to the genome.\n",
    "\n",
    "- In ATAC-seq (the data we're analyzing), the fragment length follows a periodic distribution. Some reads have very short inserts (only a few basepairs), while other reads have inserts that are much longer (100's of basepairs — much longer than the 77bp reads we're using to read them.\n",
    "\n",
    "- We know ahead of time that the first part of the adapter sequence is `CTGTCTCTTATA`, since our reads are sequenced using a Nextera sample prep kit."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16800\r\n",
      "\r\n",
      "gzip: stdout: Broken pipe\r\n"
     ]
    }
   ],
   "source": [
    "# Let's sanity check our adapter sequence by seeing\n",
    "# how many times it occurs in the first 100000 reads.\n",
    "\n",
    "ADAPTER=\"CTGTCTCTTATA\"\n",
    "\n",
    "NUM_LINES=400000  # 4 * num_reads, since each fastq entry is 4 lines\n",
    "\n",
    "zcat $(ls $FASTQ_DIR*R1* | head -n 1) | head -n $NUM_LINES | grep $ADAPTER | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\r\n",
      "\r\n",
      "gzip: stdout: Broken pipe\r\n"
     ]
    }
   ],
   "source": [
    "# Let's also check how often a permutation (rearrangement)\n",
    "# of the adapter sequence occurs:\n",
    "\n",
    "NOT_ADAPTER=\"CGTTCTTCTATA\"  # A permutation of the adapter sequence\n",
    "\n",
    "zcat $(ls $FASTQ_DIR*R1* | head -n 1) | head -n $NUM_LINES | grep $NOT_ADAPTER | wc -l"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Notice that the correct adapter sequence occurs *many* times more in the reads than a permutation of the adapter sequene — this is an important validation that we have the right sequence.\n",
    "\n",
    "Now, we'll trim the paired-end reads using a tool called `cutadapt`:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cutadapt -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /tc2016/ubuntu/analysis/Ct-1h_S5_L001_R1_001.trimmed.fastq.gz -p /tc2016/ubuntu/analysis/Ct-1h_S5_L001_R2_001.trimmed.fastq.gz /tc2016/ubuntu/data/fastq/Ct-1h_S5_L001_R1_001.fastq.gz /tc2016/ubuntu/data/fastq/Ct-1h_S5_L001_R2_001.fastq.gz\r\n",
      "This is cutadapt 1.10 with Python 3.5.2\r\n",
      "Command line parameters: -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /tc2016/ubuntu/analysis/Ct-1h_S5_L001_R1_001.trimmed.fastq.gz -p /tc2016/ubuntu/analysis/Ct-1h_S5_L001_R2_001.trimmed.fastq.gz /tc2016/ubuntu/data/fastq/Ct-1h_S5_L001_R1_001.fastq.gz /tc2016/ubuntu/data/fastq/Ct-1h_S5_L001_R2_001.fastq.gz\r\n",
      "Trimming 2 adapters with at most 20.0% errors in paired-end mode ...\r\n",
      "Finished in 31.47 s (205 us/read; 0.29 M reads/minute).\r\n",
      "\r\n",
      "=== Summary ===\r\n",
      "\r\n",
      "Total read pairs processed:            153,618\r\n",
      "  Read 1 with adapter:                  41,521 (27.0%)\r\n",
      "  Read 2 with adapter:                  41,872 (27.3%)\r\n",
      "Pairs that were too short:                 142 (0.1%)\r\n",
      "Pairs written (passing filters):       153,476 (99.9%)\r\n",
      "\r\n",
      "Total basepairs processed:    23,349,936 bp\r\n",
      "  Read 1:    11,674,968 bp\r\n",
      "  Read 2:    11,674,968 bp\r\n",
      "Total written (filtered):     21,797,941 bp (93.4%)\r\n",
      "  Read 1:    10,904,658 bp\r\n",
      "  Read 2:    10,893,283 bp\r\n",
      "\r\n",
      "=== First read: Adapter 1 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 41521 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 15.6%\r\n",
      "  C: 35.3%\r\n",
      "  G: 23.9%\r\n",
      "  T: 25.2%\r\n",
      "  none/other: 0.1%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t2311\t2400.3\t0\t2311\r\n",
      "4\t2634\t600.1\t0\t1355 1279\r\n",
      "5\t2840\t150.0\t1\t1009 1831\r\n",
      "6\t1737\t37.5\t1\t851 886\r\n",
      "7\t1161\t9.4\t1\t946 215\r\n",
      "8\t1182\t2.3\t1\t951 210 21\r\n",
      "9\t1089\t0.6\t1\t948 53 88\r\n",
      "10\t1030\t0.1\t2\t857 109 64\r\n",
      "11\t1089\t0.0\t2\t978 55 56\r\n",
      "12\t936\t0.0\t2\t789 122 25\r\n",
      "13\t1007\t0.0\t2\t939 48 20\r\n",
      "14\t1036\t0.0\t2\t977 42 17\r\n",
      "15\t1069\t0.0\t2\t1030 21 18\r\n",
      "16\t984\t0.0\t2\t941 27 16\r\n",
      "17\t1016\t0.0\t2\t957 36 23\r\n",
      "18\t1044\t0.0\t2\t968 43 33\r\n",
      "19\t1033\t0.0\t2\t988 22 23\r\n",
      "20\t1059\t0.0\t2\t1012 22 25\r\n",
      "21\t1003\t0.0\t2\t942 37 24\r\n",
      "22\t990\t0.0\t2\t946 26 18\r\n",
      "23\t914\t0.0\t2\t856 35 23\r\n",
      "24\t1018\t0.0\t2\t963 36 19\r\n",
      "25\t1145\t0.0\t2\t1083 37 25\r\n",
      "26\t1143\t0.0\t2\t1093 31 19\r\n",
      "27\t879\t0.0\t2\t846 18 15\r\n",
      "28\t871\t0.0\t2\t823 28 20\r\n",
      "29\t880\t0.0\t2\t829 31 20\r\n",
      "30\t931\t0.0\t2\t888 26 17\r\n",
      "31\t881\t0.0\t2\t831 32 18\r\n",
      "32\t829\t0.0\t2\t787 32 10\r\n",
      "33\t786\t0.0\t2\t727 17 42\r\n",
      "34\t668\t0.0\t2\t622 30 16\r\n",
      "35\t902\t0.0\t2\t868 23 11\r\n",
      "36\t861\t0.0\t2\t815 28 18\r\n",
      "37\t520\t0.0\t2\t486 15 19\r\n",
      "38\t402\t0.0\t2\t373 19 10\r\n",
      "39\t299\t0.0\t2\t274 5 20\r\n",
      "40\t243\t0.0\t2\t211 13 19\r\n",
      "41\t121\t0.0\t2\t105 0 16\r\n",
      "42\t85\t0.0\t2\t49 2 34\r\n",
      "43\t58\t0.0\t2\t35 3 20\r\n",
      "44\t37\t0.0\t2\t22 1 14\r\n",
      "45\t38\t0.0\t2\t24 2 12\r\n",
      "46\t80\t0.0\t2\t60 5 15\r\n",
      "47\t68\t0.0\t2\t53 4 11\r\n",
      "48\t37\t0.0\t2\t19 3 15\r\n",
      "49\t31\t0.0\t2\t15 2 14\r\n",
      "50\t19\t0.0\t2\t5 2 12\r\n",
      "51\t34\t0.0\t2\t1 1 32\r\n",
      "52\t20\t0.0\t2\t0 0 20\r\n",
      "53\t22\t0.0\t2\t0 0 22\r\n",
      "54\t5\t0.0\t2\t0 1 4\r\n",
      "55\t19\t0.0\t2\t0 1 18\r\n",
      "56\t16\t0.0\t2\t0 1 15\r\n",
      "57\t8\t0.0\t2\t0 4 4\r\n",
      "58\t9\t0.0\t2\t1 1 7\r\n",
      "59\t5\t0.0\t2\t0 1 4\r\n",
      "60\t9\t0.0\t2\t0 0 9\r\n",
      "61\t17\t0.0\t2\t0 0 17\r\n",
      "62\t8\t0.0\t2\t0 0 8\r\n",
      "63\t23\t0.0\t2\t0 0 23\r\n",
      "64\t15\t0.0\t2\t0 2 13\r\n",
      "65\t15\t0.0\t2\t0 2 13\r\n",
      "66\t20\t0.0\t2\t0 0 20\r\n",
      "67\t14\t0.0\t2\t0 1 13\r\n",
      "68\t95\t0.0\t2\t0 2 93\r\n",
      "69\t21\t0.0\t2\t0 2 19\r\n",
      "70\t27\t0.0\t2\t0 1 26\r\n",
      "71\t10\t0.0\t2\t0 1 9\r\n",
      "72\t11\t0.0\t2\t0 1 10\r\n",
      "73\t40\t0.0\t2\t0 2 38\r\n",
      "74\t37\t0.0\t2\t0 1 36\r\n",
      "75\t15\t0.0\t2\t0 0 15\r\n",
      "76\t10\t0.0\t2\t0 0 10\r\n",
      "\r\n",
      "=== Second read: Adapter 2 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 41872 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 16.2%\r\n",
      "  C: 34.4%\r\n",
      "  G: 25.7%\r\n",
      "  T: 23.1%\r\n",
      "  none/other: 0.7%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t2377\t2400.3\t0\t2377\r\n",
      "4\t2757\t600.1\t0\t1367 1390\r\n",
      "5\t2884\t150.0\t1\t1014 1870\r\n",
      "6\t1732\t37.5\t1\t844 888\r\n",
      "7\t1161\t9.4\t1\t969 192\r\n",
      "8\t1192\t2.3\t1\t993 181 18\r\n",
      "9\t1096\t0.6\t1\t978 23 95\r\n",
      "10\t1041\t0.1\t2\t880 99 62\r\n",
      "11\t1080\t0.0\t2\t1033 6 41\r\n",
      "12\t939\t0.0\t2\t823 93 23\r\n",
      "13\t1002\t0.0\t2\t982 8 12\r\n",
      "14\t1037\t0.0\t2\t1015 8 14\r\n",
      "15\t1078\t0.0\t2\t1045 12 21\r\n",
      "16\t995\t0.0\t2\t964 12 19\r\n",
      "17\t1020\t0.0\t2\t986 13 21\r\n",
      "18\t1041\t0.0\t2\t1018 5 18\r\n",
      "19\t1059\t0.0\t2\t1012 7 40\r\n",
      "20\t1060\t0.0\t2\t1028 7 25\r\n",
      "21\t1001\t0.0\t2\t973 8 20\r\n",
      "22\t996\t0.0\t2\t972 8 16\r\n",
      "23\t909\t0.0\t2\t611 278 20\r\n",
      "24\t1019\t0.0\t2\t685 321 13\r\n",
      "25\t1137\t0.0\t2\t761 362 14\r\n",
      "26\t1139\t0.0\t2\t758 367 14\r\n",
      "27\t877\t0.0\t2\t606 250 21\r\n",
      "28\t867\t0.0\t2\t595 259 13\r\n",
      "29\t876\t0.0\t2\t586 269 21\r\n",
      "30\t933\t0.0\t2\t679 241 13\r\n",
      "31\t878\t0.0\t2\t612 243 23\r\n",
      "32\t827\t0.0\t2\t593 225 9\r\n",
      "33\t773\t0.0\t2\t513 235 25\r\n",
      "34\t667\t0.0\t2\t452 5 210\r\n",
      "35\t900\t0.0\t2\t875 7 18\r\n",
      "36\t862\t0.0\t2\t835 8 19\r\n",
      "37\t520\t0.0\t2\t498 7 15\r\n",
      "38\t407\t0.0\t2\t389 6 12\r\n",
      "39\t295\t0.0\t2\t281 1 13\r\n",
      "40\t236\t0.0\t2\t223 2 11\r\n",
      "41\t126\t0.0\t2\t104 1 21\r\n",
      "42\t82\t0.0\t2\t51 0 31\r\n",
      "43\t52\t0.0\t2\t35 0 17\r\n",
      "44\t42\t0.0\t2\t21 1 20\r\n",
      "45\t42\t0.0\t2\t26 0 16\r\n",
      "46\t80\t0.0\t2\t64 1 15\r\n",
      "47\t74\t0.0\t2\t56 0 18\r\n",
      "48\t31\t0.0\t2\t20 1 10\r\n",
      "49\t33\t0.0\t2\t15 1 17\r\n",
      "50\t12\t0.0\t2\t5 0 7\r\n",
      "51\t30\t0.0\t2\t1 1 28\r\n",
      "52\t19\t0.0\t2\t0 0 19\r\n",
      "53\t47\t0.0\t2\t0 1 46\r\n",
      "54\t25\t0.0\t2\t1 0 24\r\n",
      "55\t16\t0.0\t2\t1 0 15\r\n",
      "56\t46\t0.0\t2\t1 0 45\r\n",
      "57\t19\t0.0\t2\t3 0 16\r\n",
      "58\t21\t0.0\t2\t1 1 19\r\n",
      "59\t29\t0.0\t2\t1 1 27\r\n",
      "60\t18\t0.0\t2\t0 0 18\r\n",
      "61\t31\t0.0\t2\t0 3 28\r\n",
      "62\t16\t0.0\t2\t0 0 16\r\n",
      "63\t39\t0.0\t2\t0 0 39\r\n",
      "64\t18\t0.0\t2\t0 1 17\r\n",
      "65\t16\t0.0\t2\t0 0 16\r\n",
      "66\t22\t0.0\t2\t0 2 20\r\n",
      "67\t16\t0.0\t2\t0 1 15\r\n",
      "68\t102\t0.0\t2\t0 4 98\r\n",
      "69\t19\t0.0\t2\t0 2 17\r\n",
      "70\t39\t0.0\t2\t0 0 39\r\n",
      "71\t11\t0.0\t2\t0 0 11\r\n",
      "72\t7\t0.0\t2\t0 0 7\r\n",
      "73\t1\t0.0\t2\t0 0 1\r\n",
      "74\t19\t0.0\t2\t0 0 19\r\n",
      "76\t2\t0.0\t2\t0 0 2\r\n",
      "\r\n",
      "cutadapt -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /tc2016/ubuntu/analysis/Ct-3h_S12_L001_R1_001.trimmed.fastq.gz -p /tc2016/ubuntu/analysis/Ct-3h_S12_L001_R2_001.trimmed.fastq.gz /tc2016/ubuntu/data/fastq/Ct-3h_S12_L001_R1_001.fastq.gz /tc2016/ubuntu/data/fastq/Ct-3h_S12_L001_R2_001.fastq.gz\r\n",
      "This is cutadapt 1.10 with Python 3.5.2\r\n",
      "Command line parameters: -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /tc2016/ubuntu/analysis/Ct-3h_S12_L001_R1_001.trimmed.fastq.gz -p /tc2016/ubuntu/analysis/Ct-3h_S12_L001_R2_001.trimmed.fastq.gz /tc2016/ubuntu/data/fastq/Ct-3h_S12_L001_R1_001.fastq.gz /tc2016/ubuntu/data/fastq/Ct-3h_S12_L001_R2_001.fastq.gz\r\n",
      "Trimming 2 adapters with at most 20.0% errors in paired-end mode ...\r\n",
      "Finished in 86.40 s (192 us/read; 0.31 M reads/minute).\r\n",
      "\r\n",
      "=== Summary ===\r\n",
      "\r\n",
      "Total read pairs processed:            451,176\r\n",
      "  Read 1 with adapter:                 138,083 (30.6%)\r\n",
      "  Read 2 with adapter:                 138,805 (30.8%)\r\n",
      "Pairs that were too short:                 360 (0.1%)\r\n",
      "Pairs written (passing filters):       450,816 (99.9%)\r\n",
      "\r\n",
      "Total basepairs processed:    68,578,752 bp\r\n",
      "  Read 1:    34,289,376 bp\r\n",
      "  Read 2:    34,289,376 bp\r\n",
      "Total written (filtered):     63,097,167 bp (92.0%)\r\n",
      "  Read 1:    31,554,344 bp\r\n",
      "  Read 2:    31,542,823 bp\r\n",
      "\r\n",
      "=== First read: Adapter 1 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 138083 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 14.8%\r\n",
      "  C: 36.3%\r\n",
      "  G: 24.5%\r\n",
      "  T: 24.3%\r\n",
      "  none/other: 0.0%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t6795\t7049.6\t0\t6795\r\n",
      "4\t7560\t1762.4\t0\t3998 3562\r\n",
      "5\t8218\t440.6\t1\t3000 5218\r\n",
      "6\t5089\t110.2\t1\t2654 2435\r\n",
      "7\t3536\t27.5\t1\t3000 536\r\n",
      "8\t3722\t6.9\t1\t2909 779 34\r\n",
      "9\t3389\t1.7\t1\t3075 78 236\r\n",
      "10\t3070\t0.4\t2\t2624 294 152\r\n",
      "11\t2981\t0.1\t2\t2815 35 131\r\n",
      "12\t3056\t0.0\t2\t2709 308 39\r\n",
      "13\t3132\t0.0\t2\t3048 43 41\r\n",
      "14\t3105\t0.0\t2\t3036 25 44\r\n",
      "15\t3245\t0.0\t2\t3172 36 37\r\n",
      "16\t3115\t0.0\t2\t3027 33 55\r\n",
      "17\t3252\t0.0\t2\t3169 41 42\r\n",
      "18\t3230\t0.0\t2\t3150 29 51\r\n",
      "19\t3453\t0.0\t2\t3345 32 76\r\n",
      "20\t3408\t0.0\t2\t3310 48 50\r\n",
      "21\t3419\t0.0\t2\t3323 55 41\r\n",
      "22\t3272\t0.0\t2\t3209 28 35\r\n",
      "23\t3050\t0.0\t2\t2978 30 42\r\n",
      "24\t3534\t0.0\t2\t3451 42 41\r\n",
      "25\t3978\t0.0\t2\t3917 31 30\r\n",
      "26\t3869\t0.0\t2\t3795 37 37\r\n",
      "27\t3375\t0.0\t2\t3313 22 40\r\n",
      "28\t3283\t0.0\t2\t3215 25 43\r\n",
      "29\t3505\t0.0\t2\t3426 32 47\r\n",
      "30\t3603\t0.0\t2\t3528 31 44\r\n",
      "31\t3487\t0.0\t2\t3416 31 40\r\n",
      "32\t3342\t0.0\t2\t3263 40 39\r\n",
      "33\t3139\t0.0\t2\t3050 31 58\r\n",
      "34\t2866\t0.0\t2\t2778 41 47\r\n",
      "35\t3549\t0.0\t2\t3473 41 35\r\n",
      "36\t3377\t0.0\t2\t3303 45 29\r\n",
      "37\t2300\t0.0\t2\t2243 21 36\r\n",
      "38\t1782\t0.0\t2\t1739 14 29\r\n",
      "39\t1263\t0.0\t2\t1208 13 42\r\n",
      "40\t902\t0.0\t2\t832 14 56\r\n",
      "41\t475\t0.0\t2\t413 9 53\r\n",
      "42\t337\t0.0\t2\t231 9 97\r\n",
      "43\t207\t0.0\t2\t149 5 53\r\n",
      "44\t178\t0.0\t2\t124 4 50\r\n",
      "45\t154\t0.0\t2\t131 2 21\r\n",
      "46\t385\t0.0\t2\t351 3 31\r\n",
      "47\t310\t0.0\t2\t260 5 45\r\n",
      "48\t165\t0.0\t2\t132 7 26\r\n",
      "49\t165\t0.0\t2\t110 1 54\r\n",
      "50\t91\t0.0\t2\t67 0 24\r\n",
      "51\t101\t0.0\t2\t4 1 96\r\n",
      "52\t48\t0.0\t2\t2 0 46\r\n",
      "53\t83\t0.0\t2\t2 5 76\r\n",
      "54\t27\t0.0\t2\t1 4 22\r\n",
      "55\t42\t0.0\t2\t1 2 39\r\n",
      "56\t38\t0.0\t2\t3 3 32\r\n",
      "57\t38\t0.0\t2\t3 4 31\r\n",
      "58\t31\t0.0\t2\t2 2 27\r\n",
      "59\t42\t0.0\t2\t0 1 41\r\n",
      "60\t24\t0.0\t2\t0 0 24\r\n",
      "61\t41\t0.0\t2\t0 0 41\r\n",
      "62\t11\t0.0\t2\t0 0 11\r\n",
      "63\t97\t0.0\t2\t0 0 97\r\n",
      "64\t51\t0.0\t2\t0 1 50\r\n",
      "65\t49\t0.0\t2\t0 2 47\r\n",
      "66\t47\t0.0\t2\t0 0 47\r\n",
      "67\t37\t0.0\t2\t0 1 36\r\n",
      "68\t169\t0.0\t2\t0 5 164\r\n",
      "69\t57\t0.0\t2\t0 2 55\r\n",
      "70\t69\t0.0\t2\t0 0 69\r\n",
      "71\t18\t0.0\t2\t0 0 18\r\n",
      "72\t21\t0.0\t2\t0 0 21\r\n",
      "73\t77\t0.0\t2\t0 2 75\r\n",
      "74\t77\t0.0\t2\t0 0 77\r\n",
      "75\t40\t0.0\t2\t0 2 38\r\n",
      "76\t30\t0.0\t2\t0 2 28\r\n",
      "\r\n",
      "=== Second read: Adapter 2 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 138805 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 15.2%\r\n",
      "  C: 36.4%\r\n",
      "  G: 24.8%\r\n",
      "  T: 22.9%\r\n",
      "  none/other: 0.8%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t6821\t7049.6\t0\t6821\r\n",
      "4\t7968\t1762.4\t0\t3987 3981\r\n",
      "5\t8692\t440.6\t1\t3013 5679\r\n",
      "6\t5122\t110.2\t1\t2385 2737\r\n",
      "7\t3599\t27.5\t1\t3000 599\r\n",
      "8\t3574\t6.9\t1\t2843 694 37\r\n",
      "9\t3385\t1.7\t1\t3048 108 229\r\n",
      "10\t3101\t0.4\t2\t2539 380 182\r\n",
      "11\t2977\t0.1\t2\t2779 66 132\r\n",
      "12\t3064\t0.0\t2\t2617 384 63\r\n",
      "13\t3146\t0.0\t2\t3008 79 59\r\n",
      "14\t3087\t0.0\t2\t2759 279 49\r\n",
      "15\t3246\t0.0\t2\t2877 310 59\r\n",
      "16\t3090\t0.0\t2\t2737 299 54\r\n",
      "17\t3250\t0.0\t2\t2895 292 63\r\n",
      "18\t3234\t0.0\t2\t2875 272 87\r\n",
      "19\t3434\t0.0\t2\t3051 304 79\r\n",
      "20\t3380\t0.0\t2\t3028 280 72\r\n",
      "21\t3417\t0.0\t2\t3048 305 64\r\n",
      "22\t3277\t0.0\t2\t2941 276 60\r\n",
      "23\t3030\t0.0\t2\t1979 984 67\r\n",
      "24\t3518\t0.0\t2\t2303 1121 94\r\n",
      "25\t3970\t0.0\t2\t2637 951 382\r\n",
      "26\t3849\t0.0\t2\t2818 967 64\r\n",
      "27\t3354\t0.0\t2\t2437 827 90\r\n",
      "28\t3272\t0.0\t2\t2345 865 62\r\n",
      "29\t3487\t0.0\t2\t2522 843 122\r\n",
      "30\t3584\t0.0\t2\t2556 978 50\r\n",
      "31\t3465\t0.0\t2\t2487 895 83\r\n",
      "32\t3342\t0.0\t2\t2401 871 70\r\n",
      "33\t3107\t0.0\t2\t2208 810 89\r\n",
      "34\t2854\t0.0\t2\t2016 54 784\r\n",
      "35\t3534\t0.0\t2\t3394 77 63\r\n",
      "36\t3378\t0.0\t2\t3247 81 50\r\n",
      "37\t2287\t0.0\t2\t2186 57 44\r\n",
      "38\t1781\t0.0\t2\t1706 38 37\r\n",
      "39\t1256\t0.0\t2\t1176 23 57\r\n",
      "40\t883\t0.0\t2\t823 23 37\r\n",
      "41\t472\t0.0\t2\t411 8 53\r\n",
      "42\t342\t0.0\t2\t229 9 104\r\n",
      "43\t207\t0.0\t2\t145 4 58\r\n",
      "44\t182\t0.0\t2\t121 4 57\r\n",
      "45\t167\t0.0\t2\t127 8 32\r\n",
      "46\t392\t0.0\t2\t341 8 43\r\n",
      "47\t303\t0.0\t2\t253 8 42\r\n",
      "48\t167\t0.0\t2\t131 5 31\r\n",
      "49\t173\t0.0\t2\t105 7 61\r\n",
      "50\t90\t0.0\t2\t64 2 24\r\n",
      "51\t88\t0.0\t2\t5 1 82\r\n",
      "52\t35\t0.0\t2\t2 0 33\r\n",
      "53\t96\t0.0\t2\t3 6 87\r\n",
      "54\t62\t0.0\t2\t3 4 55\r\n",
      "55\t45\t0.0\t2\t2 2 41\r\n",
      "56\t86\t0.0\t2\t6 0 80\r\n",
      "57\t53\t0.0\t2\t6 1 46\r\n",
      "58\t61\t0.0\t2\t2 1 58\r\n",
      "59\t73\t0.0\t2\t0 5 68\r\n",
      "60\t54\t0.0\t2\t0 0 54\r\n",
      "61\t60\t0.0\t2\t0 1 59\r\n",
      "62\t33\t0.0\t2\t0 0 33\r\n",
      "63\t100\t0.0\t2\t0 2 98\r\n",
      "64\t74\t0.0\t2\t0 1 73\r\n",
      "65\t44\t0.0\t2\t0 7 37\r\n",
      "66\t52\t0.0\t2\t0 1 51\r\n",
      "67\t31\t0.0\t2\t0 3 28\r\n",
      "68\t178\t0.0\t2\t0 6 172\r\n",
      "69\t54\t0.0\t2\t0 2 52\r\n",
      "70\t83\t0.0\t2\t0 0 83\r\n",
      "71\t18\t0.0\t2\t0 0 18\r\n",
      "72\t10\t0.0\t2\t0 0 10\r\n",
      "73\t16\t0.0\t2\t0 1 15\r\n",
      "74\t71\t0.0\t2\t0 0 71\r\n",
      "75\t11\t0.0\t2\t0 0 11\r\n",
      "76\t7\t0.0\t2\t0 0 7\r\n",
      "\r\n",
      "cutadapt -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /tc2016/ubuntu/analysis/Cu-1h_S4_L001_R1_001.trimmed.fastq.gz -p /tc2016/ubuntu/analysis/Cu-1h_S4_L001_R2_001.trimmed.fastq.gz /tc2016/ubuntu/data/fastq/Cu-1h_S4_L001_R1_001.fastq.gz /tc2016/ubuntu/data/fastq/Cu-1h_S4_L001_R2_001.fastq.gz\r\n",
      "This is cutadapt 1.10 with Python 3.5.2\r\n",
      "Command line parameters: -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /tc2016/ubuntu/analysis/Cu-1h_S4_L001_R1_001.trimmed.fastq.gz -p /tc2016/ubuntu/analysis/Cu-1h_S4_L001_R2_001.trimmed.fastq.gz /tc2016/ubuntu/data/fastq/Cu-1h_S4_L001_R1_001.fastq.gz /tc2016/ubuntu/data/fastq/Cu-1h_S4_L001_R2_001.fastq.gz\r\n",
      "Trimming 2 adapters with at most 20.0% errors in paired-end mode ...\r\n",
      "Finished in 3.61 s (191 us/read; 0.31 M reads/minute).\r\n",
      "\r\n",
      "=== Summary ===\r\n",
      "\r\n",
      "Total read pairs processed:             18,874\r\n",
      "  Read 1 with adapter:                   3,041 (16.1%)\r\n",
      "  Read 2 with adapter:                   3,287 (17.4%)\r\n",
      "Pairs that were too short:                  20 (0.1%)\r\n",
      "Pairs written (passing filters):        18,854 (99.9%)\r\n",
      "\r\n",
      "Total basepairs processed:     2,868,848 bp\r\n",
      "  Read 1:     1,434,424 bp\r\n",
      "  Read 2:     1,434,424 bp\r\n",
      "Total written (filtered):      2,764,046 bp (96.3%)\r\n",
      "  Read 1:     1,384,900 bp\r\n",
      "  Read 2:     1,379,146 bp\r\n",
      "\r\n",
      "=== First read: Adapter 1 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 3041 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 17.9%\r\n",
      "  C: 31.6%\r\n",
      "  G: 25.0%\r\n",
      "  T: 25.4%\r\n",
      "  none/other: 0.2%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t249\t294.9\t0\t249\r\n",
      "4\t271\t73.7\t0\t120 151\r\n",
      "5\t310\t18.4\t1\t74 236\r\n",
      "6\t210\t4.6\t1\t73 137\r\n",
      "7\t87\t1.2\t1\t57 30\r\n",
      "8\t85\t0.3\t1\t59 24 2\r\n",
      "9\t87\t0.1\t1\t70 6 11\r\n",
      "10\t85\t0.0\t2\t62 14 9\r\n",
      "11\t59\t0.0\t2\t47 5 7\r\n",
      "12\t75\t0.0\t2\t52 18 5\r\n",
      "13\t78\t0.0\t2\t73 3 2\r\n",
      "14\t75\t0.0\t2\t66 3 6\r\n",
      "15\t82\t0.0\t2\t73 8 1\r\n",
      "16\t59\t0.0\t2\t52 4 3\r\n",
      "17\t60\t0.0\t2\t52 5 3\r\n",
      "18\t51\t0.0\t2\t41 6 4\r\n",
      "19\t63\t0.0\t2\t56 5 2\r\n",
      "20\t48\t0.0\t2\t43 3 2\r\n",
      "21\t61\t0.0\t2\t57 4\r\n",
      "22\t69\t0.0\t2\t59 8 2\r\n",
      "23\t51\t0.0\t2\t45 2 4\r\n",
      "24\t60\t0.0\t2\t53 5 2\r\n",
      "25\t79\t0.0\t2\t73 4 2\r\n",
      "26\t47\t0.0\t2\t43 2 2\r\n",
      "27\t58\t0.0\t2\t54 2 2\r\n",
      "28\t49\t0.0\t2\t45 2 2\r\n",
      "29\t48\t0.0\t2\t43 1 4\r\n",
      "30\t54\t0.0\t2\t49 2 3\r\n",
      "31\t51\t0.0\t2\t44 1 6\r\n",
      "32\t47\t0.0\t2\t45 2\r\n",
      "33\t39\t0.0\t2\t32 3 4\r\n",
      "34\t31\t0.0\t2\t25 2 4\r\n",
      "35\t44\t0.0\t2\t43 1\r\n",
      "36\t42\t0.0\t2\t36 2 4\r\n",
      "37\t34\t0.0\t2\t28 3 3\r\n",
      "38\t14\t0.0\t2\t14\r\n",
      "39\t14\t0.0\t2\t10 0 4\r\n",
      "40\t5\t0.0\t2\t3 1 1\r\n",
      "41\t11\t0.0\t2\t7 0 4\r\n",
      "42\t8\t0.0\t2\t3 0 5\r\n",
      "43\t4\t0.0\t2\t1 0 3\r\n",
      "44\t3\t0.0\t2\t1 1 1\r\n",
      "45\t4\t0.0\t2\t2 0 2\r\n",
      "46\t6\t0.0\t2\t2 0 4\r\n",
      "47\t4\t0.0\t2\t3 0 1\r\n",
      "48\t2\t0.0\t2\t1 0 1\r\n",
      "49\t5\t0.0\t2\t0 0 5\r\n",
      "50\t2\t0.0\t2\t1 0 1\r\n",
      "51\t4\t0.0\t2\t0 0 4\r\n",
      "52\t2\t0.0\t2\t0 0 2\r\n",
      "53\t4\t0.0\t2\t0 0 4\r\n",
      "55\t2\t0.0\t2\t0 0 2\r\n",
      "57\t2\t0.0\t2\t0 0 2\r\n",
      "59\t2\t0.0\t2\t0 0 2\r\n",
      "63\t3\t0.0\t2\t0 0 3\r\n",
      "65\t3\t0.0\t2\t0 0 3\r\n",
      "66\t3\t0.0\t2\t0 0 3\r\n",
      "67\t2\t0.0\t2\t0 0 2\r\n",
      "68\t9\t0.0\t2\t0 0 9\r\n",
      "69\t4\t0.0\t2\t0 1 3\r\n",
      "70\t4\t0.0\t2\t0 0 4\r\n",
      "71\t1\t0.0\t2\t0 0 1\r\n",
      "72\t2\t0.0\t2\t0 0 2\r\n",
      "73\t4\t0.0\t2\t0 0 4\r\n",
      "74\t4\t0.0\t2\t0 1 3\r\n",
      "75\t3\t0.0\t2\t0 0 3\r\n",
      "76\t3\t0.0\t2\t0 0 3\r\n",
      "\r\n",
      "=== Second read: Adapter 2 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 3287 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 17.2%\r\n",
      "  C: 33.1%\r\n",
      "  G: 25.7%\r\n",
      "  T: 23.4%\r\n",
      "  none/other: 0.6%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t273\t294.9\t0\t273\r\n",
      "4\t378\t73.7\t0\t149 229\r\n",
      "5\t324\t18.4\t1\t77 247\r\n",
      "6\t200\t4.6\t1\t82 118\r\n",
      "7\t108\t1.2\t1\t61 47\r\n",
      "8\t88\t0.3\t1\t68 18 2\r\n",
      "9\t93\t0.1\t1\t75 8 10\r\n",
      "10\t83\t0.0\t2\t68 11 4\r\n",
      "11\t61\t0.0\t2\t51 0 10\r\n",
      "12\t73\t0.0\t2\t62 9 2\r\n",
      "13\t77\t0.0\t2\t72 3 2\r\n",
      "14\t72\t0.0\t2\t65 4 3\r\n",
      "15\t79\t0.0\t2\t79\r\n",
      "16\t62\t0.0\t2\t57 1 4\r\n",
      "17\t73\t0.0\t2\t55 2 16\r\n",
      "18\t49\t0.0\t2\t47 1 1\r\n",
      "19\t71\t0.0\t2\t61 1 9\r\n",
      "20\t51\t0.0\t2\t47 0 4\r\n",
      "21\t66\t0.0\t2\t59 1 6\r\n",
      "22\t68\t0.0\t2\t64 2 2\r\n",
      "23\t47\t0.0\t2\t37 8 2\r\n",
      "24\t61\t0.0\t2\t47 12 2\r\n",
      "25\t80\t0.0\t2\t65 13 2\r\n",
      "26\t46\t0.0\t2\t34 9 3\r\n",
      "27\t59\t0.0\t2\t45 12 2\r\n",
      "28\t48\t0.0\t2\t34 12 2\r\n",
      "29\t47\t0.0\t2\t32 9 6\r\n",
      "30\t53\t0.0\t2\t41 10 2\r\n",
      "31\t48\t0.0\t2\t38 7 3\r\n",
      "32\t49\t0.0\t2\t35 10 4\r\n",
      "33\t36\t0.0\t2\t29 6 1\r\n",
      "34\t28\t0.0\t2\t20 2 6\r\n",
      "35\t45\t0.0\t2\t40 3 2\r\n",
      "36\t42\t0.0\t2\t38 1 3\r\n",
      "37\t34\t0.0\t2\t32 0 2\r\n",
      "38\t16\t0.0\t2\t13 0 3\r\n",
      "39\t11\t0.0\t2\t8 0 3\r\n",
      "40\t7\t0.0\t2\t5 0 2\r\n",
      "41\t9\t0.0\t2\t7 0 2\r\n",
      "42\t5\t0.0\t2\t3 0 2\r\n",
      "43\t5\t0.0\t2\t1 1 3\r\n",
      "44\t4\t0.0\t2\t1 0 3\r\n",
      "45\t3\t0.0\t2\t2 0 1\r\n",
      "46\t12\t0.0\t2\t2 1 9\r\n",
      "47\t6\t0.0\t2\t3 0 3\r\n",
      "48\t7\t0.0\t2\t1 1 5\r\n",
      "49\t4\t0.0\t2\t0 0 4\r\n",
      "50\t4\t0.0\t2\t1 0 3\r\n",
      "51\t5\t0.0\t2\t0 0 5\r\n",
      "52\t5\t0.0\t2\t0 0 5\r\n",
      "53\t5\t0.0\t2\t0 0 5\r\n",
      "54\t8\t0.0\t2\t0 0 8\r\n",
      "56\t14\t0.0\t2\t0 0 14\r\n",
      "57\t1\t0.0\t2\t0 0 1\r\n",
      "58\t3\t0.0\t2\t0 0 3\r\n",
      "59\t11\t0.0\t2\t0 0 11\r\n",
      "60\t7\t0.0\t2\t0 0 7\r\n",
      "61\t5\t0.0\t2\t0 0 5\r\n",
      "62\t5\t0.0\t2\t0 0 5\r\n",
      "63\t7\t0.0\t2\t0 0 7\r\n",
      "64\t2\t0.0\t2\t0 0 2\r\n",
      "65\t1\t0.0\t2\t0 1\r\n",
      "66\t2\t0.0\t2\t0 0 2\r\n",
      "67\t6\t0.0\t2\t0 1 5\r\n",
      "68\t16\t0.0\t2\t0 0 16\r\n",
      "69\t5\t0.0\t2\t0 2 3\r\n",
      "70\t9\t0.0\t2\t0 0 9\r\n",
      "71\t1\t0.0\t2\t0 0 1\r\n",
      "72\t1\t0.0\t2\t0 0 1\r\n",
      "74\t3\t0.0\t2\t0 0 3\r\n",
      "\r\n",
      "cutadapt -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /tc2016/ubuntu/analysis/Cu-3h_S11_L001_R1_001.trimmed.fastq.gz -p /tc2016/ubuntu/analysis/Cu-3h_S11_L001_R2_001.trimmed.fastq.gz /tc2016/ubuntu/data/fastq/Cu-3h_S11_L001_R1_001.fastq.gz /tc2016/ubuntu/data/fastq/Cu-3h_S11_L001_R2_001.fastq.gz\r\n",
      "This is cutadapt 1.10 with Python 3.5.2\r\n",
      "Command line parameters: -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /tc2016/ubuntu/analysis/Cu-3h_S11_L001_R1_001.trimmed.fastq.gz -p /tc2016/ubuntu/analysis/Cu-3h_S11_L001_R2_001.trimmed.fastq.gz /tc2016/ubuntu/data/fastq/Cu-3h_S11_L001_R1_001.fastq.gz /tc2016/ubuntu/data/fastq/Cu-3h_S11_L001_R2_001.fastq.gz\r\n",
      "Trimming 2 adapters with at most 20.0% errors in paired-end mode ...\r\n",
      "Finished in 83.62 s (192 us/read; 0.31 M reads/minute).\r\n",
      "\r\n",
      "=== Summary ===\r\n",
      "\r\n",
      "Total read pairs processed:            434,576\r\n",
      "  Read 1 with adapter:                 145,703 (33.5%)\r\n",
      "  Read 2 with adapter:                 146,583 (33.7%)\r\n",
      "Pairs that were too short:                 358 (0.1%)\r\n",
      "Pairs written (passing filters):       434,218 (99.9%)\r\n",
      "\r\n",
      "Total basepairs processed:    66,055,552 bp\r\n",
      "  Read 1:    33,027,776 bp\r\n",
      "  Read 2:    33,027,776 bp\r\n",
      "Total written (filtered):     59,971,599 bp (90.8%)\r\n",
      "  Read 1:    29,991,858 bp\r\n",
      "  Read 2:    29,979,741 bp\r\n",
      "\r\n",
      "=== First read: Adapter 1 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 145703 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 14.4%\r\n",
      "  C: 36.7%\r\n",
      "  G: 24.5%\r\n",
      "  T: 24.3%\r\n",
      "  none/other: 0.0%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t6238\t6790.2\t0\t6238\r\n",
      "4\t7198\t1697.6\t0\t3902 3296\r\n",
      "5\t7652\t424.4\t1\t2905 4747\r\n",
      "6\t4847\t106.1\t1\t2594 2253\r\n",
      "7\t3216\t26.5\t1\t2731 485\r\n",
      "8\t3338\t6.6\t1\t2747 549 42\r\n",
      "9\t3236\t1.7\t1\t2937 81 218\r\n",
      "10\t3050\t0.4\t2\t2616 280 154\r\n",
      "11\t3096\t0.1\t2\t2938 34 124\r\n",
      "12\t3025\t0.0\t2\t2668 300 57\r\n",
      "13\t3194\t0.0\t2\t3101 45 48\r\n",
      "14\t3378\t0.0\t2\t3304 37 37\r\n",
      "15\t3658\t0.0\t2\t3583 34 41\r\n",
      "16\t3295\t0.0\t2\t3190 47 58\r\n",
      "17\t3229\t0.0\t2\t3154 38 37\r\n",
      "18\t3186\t0.0\t2\t3107 35 44\r\n",
      "19\t3393\t0.0\t2\t3301 33 59\r\n",
      "20\t3581\t0.0\t2\t3483 57 41\r\n",
      "21\t3475\t0.0\t2\t3381 32 62\r\n",
      "22\t3302\t0.0\t2\t3235 24 43\r\n",
      "23\t3232\t0.0\t2\t3158 28 46\r\n",
      "24\t3943\t0.0\t2\t3847 50 46\r\n",
      "25\t4839\t0.0\t2\t4760 47 32\r\n",
      "26\t4784\t0.0\t2\t4696 42 46\r\n",
      "27\t3731\t0.0\t2\t3662 34 35\r\n",
      "28\t3442\t0.0\t2\t3369 39 34\r\n",
      "29\t3601\t0.0\t2\t3523 29 49\r\n",
      "30\t3978\t0.0\t2\t3894 41 43\r\n",
      "31\t3811\t0.0\t2\t3733 39 39\r\n",
      "32\t3525\t0.0\t2\t3453 33 39\r\n",
      "33\t3046\t0.0\t2\t2964 34 48\r\n",
      "34\t3200\t0.0\t2\t3131 37 32\r\n",
      "35\t4797\t0.0\t2\t4716 43 38\r\n",
      "36\t5180\t0.0\t2\t5102 37 41\r\n",
      "37\t3094\t0.0\t2\t3038 28 28\r\n",
      "38\t1940\t0.0\t2\t1896 19 25\r\n",
      "39\t1381\t0.0\t2\t1319 14 48\r\n",
      "40\t1013\t0.0\t2\t953 18 42\r\n",
      "41\t511\t0.0\t2\t467 5 39\r\n",
      "42\t358\t0.0\t2\t274 5 79\r\n",
      "43\t197\t0.0\t2\t158 1 38\r\n",
      "44\t165\t0.0\t2\t126 1 38\r\n",
      "45\t234\t0.0\t2\t194 8 32\r\n",
      "46\t769\t0.0\t2\t711 7 51\r\n",
      "47\t527\t0.0\t2\t457 7 63\r\n",
      "48\t203\t0.0\t2\t169 4 30\r\n",
      "49\t161\t0.0\t2\t109 3 49\r\n",
      "50\t91\t0.0\t2\t65 0 26\r\n",
      "51\t83\t0.0\t2\t14 2 67\r\n",
      "52\t36\t0.0\t2\t5 2 29\r\n",
      "53\t75\t0.0\t2\t1 2 72\r\n",
      "54\t40\t0.0\t2\t0 4 36\r\n",
      "55\t42\t0.0\t2\t1 1 40\r\n",
      "56\t47\t0.0\t2\t3 7 37\r\n",
      "57\t37\t0.0\t2\t11 5 21\r\n",
      "58\t32\t0.0\t2\t1 6 25\r\n",
      "59\t53\t0.0\t2\t0 4 49\r\n",
      "60\t36\t0.0\t2\t0 1 35\r\n",
      "61\t32\t0.0\t2\t0 0 32\r\n",
      "62\t17\t0.0\t2\t0 0 17\r\n",
      "63\t78\t0.0\t2\t0 2 76\r\n",
      "64\t38\t0.0\t2\t0 0 38\r\n",
      "65\t44\t0.0\t2\t0 8 36\r\n",
      "66\t54\t0.0\t2\t0 0 54\r\n",
      "67\t37\t0.0\t2\t0 2 35\r\n",
      "68\t176\t0.0\t2\t0 5 171\r\n",
      "69\t57\t0.0\t2\t0 0 57\r\n",
      "70\t59\t0.0\t2\t0 0 59\r\n",
      "71\t23\t0.0\t2\t0 0 23\r\n",
      "72\t30\t0.0\t2\t0 0 30\r\n",
      "73\t78\t0.0\t2\t0 2 76\r\n",
      "74\t73\t0.0\t2\t0 0 73\r\n",
      "75\t38\t0.0\t2\t0 1 37\r\n",
      "76\t48\t0.0\t2\t0 2 46\r\n",
      "\r\n",
      "=== Second read: Adapter 2 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 146583 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 15.0%\r\n",
      "  C: 36.4%\r\n",
      "  G: 24.8%\r\n",
      "  T: 23.1%\r\n",
      "  none/other: 0.7%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t6286\t6790.2\t0\t6286\r\n",
      "4\t7565\t1697.6\t0\t3943 3622\r\n",
      "5\t8035\t424.4\t1\t2900 5135\r\n",
      "6\t5077\t106.1\t1\t2372 2705\r\n",
      "7\t3261\t26.5\t1\t2694 567\r\n",
      "8\t3344\t6.6\t1\t2691 617 36\r\n",
      "9\t3225\t1.7\t1\t2914 106 205\r\n",
      "10\t3073\t0.4\t2\t2484 409 180\r\n",
      "11\t3079\t0.1\t2\t2888 77 114\r\n",
      "12\t3018\t0.0\t2\t2537 420 61\r\n",
      "13\t3199\t0.0\t2\t3063 82 54\r\n",
      "14\t3372\t0.0\t2\t2993 324 55\r\n",
      "15\t3654\t0.0\t2\t3274 319 61\r\n",
      "16\t3270\t0.0\t2\t2920 284 66\r\n",
      "17\t3226\t0.0\t2\t2866 298 62\r\n",
      "18\t3174\t0.0\t2\t2810 290 74\r\n",
      "19\t3398\t0.0\t2\t2985 319 94\r\n",
      "20\t3581\t0.0\t2\t3182 314 85\r\n",
      "21\t3465\t0.0\t2\t3077 316 72\r\n",
      "22\t3323\t0.0\t2\t2953 281 89\r\n",
      "23\t3215\t0.0\t2\t2141 1002 72\r\n",
      "24\t3928\t0.0\t2\t2572 1254 102\r\n",
      "25\t4823\t0.0\t2\t3117 1255 451\r\n",
      "26\t4759\t0.0\t2\t3477 1193 89\r\n",
      "27\t3704\t0.0\t2\t2682 928 94\r\n",
      "28\t3444\t0.0\t2\t2480 907 57\r\n",
      "29\t3575\t0.0\t2\t2605 861 109\r\n",
      "30\t3964\t0.0\t2\t2847 1041 76\r\n",
      "31\t3795\t0.0\t2\t2758 958 79\r\n",
      "32\t3493\t0.0\t2\t2503 939 51\r\n",
      "33\t3031\t0.0\t2\t2181 751 99\r\n",
      "34\t3196\t0.0\t2\t2284 66 846\r\n",
      "35\t4785\t0.0\t2\t4593 113 79\r\n",
      "36\t5161\t0.0\t2\t4946 154 61\r\n",
      "37\t3111\t0.0\t2\t2973 79 59\r\n",
      "38\t1937\t0.0\t2\t1855 50 32\r\n",
      "39\t1377\t0.0\t2\t1282 40 55\r\n",
      "40\t989\t0.0\t2\t942 16 31\r\n",
      "41\t524\t0.0\t2\t462 6 56\r\n",
      "42\t361\t0.0\t2\t266 9 86\r\n",
      "43\t205\t0.0\t2\t155 7 43\r\n",
      "44\t166\t0.0\t2\t124 3 39\r\n",
      "45\t239\t0.0\t2\t190 8 41\r\n",
      "46\t765\t0.0\t2\t698 16 51\r\n",
      "47\t497\t0.0\t2\t452 7 38\r\n",
      "48\t206\t0.0\t2\t164 5 37\r\n",
      "49\t166\t0.0\t2\t109 2 55\r\n",
      "50\t94\t0.0\t2\t62 0 32\r\n",
      "51\t97\t0.0\t2\t13 2 82\r\n",
      "52\t49\t0.0\t2\t7 0 42\r\n",
      "53\t118\t0.0\t2\t2 8 108\r\n",
      "54\t55\t0.0\t2\t3 3 49\r\n",
      "55\t37\t0.0\t2\t2 1 34\r\n",
      "56\t95\t0.0\t2\t9 1 85\r\n",
      "57\t50\t0.0\t2\t16 0 34\r\n",
      "58\t61\t0.0\t2\t6 1 54\r\n",
      "59\t54\t0.0\t2\t2 2 50\r\n",
      "60\t39\t0.0\t2\t0 1 38\r\n",
      "61\t70\t0.0\t2\t0 1 69\r\n",
      "62\t29\t0.0\t2\t0 0 29\r\n",
      "63\t109\t0.0\t2\t0 2 107\r\n",
      "64\t77\t0.0\t2\t0 1 76\r\n",
      "65\t41\t0.0\t2\t0 1 40\r\n",
      "66\t32\t0.0\t2\t0 0 32\r\n",
      "67\t34\t0.0\t2\t0 3 31\r\n",
      "68\t174\t0.0\t2\t0 8 166\r\n",
      "69\t57\t0.0\t2\t0 4 53\r\n",
      "70\t66\t0.0\t2\t0 4 62\r\n",
      "71\t13\t0.0\t2\t0 0 13\r\n",
      "72\t16\t0.0\t2\t0 0 16\r\n",
      "73\t11\t0.0\t2\t0 0 11\r\n",
      "74\t53\t0.0\t2\t0 0 53\r\n",
      "75\t7\t0.0\t2\t0 0 7\r\n",
      "76\t4\t0.0\t2\t0 1 3\r\n",
      "\r\n",
      "cutadapt -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /tc2016/ubuntu/analysis/Cz-1h_S2_L001_R1_001.trimmed.fastq.gz -p /tc2016/ubuntu/analysis/Cz-1h_S2_L001_R2_001.trimmed.fastq.gz /tc2016/ubuntu/data/fastq/Cz-1h_S2_L001_R1_001.fastq.gz /tc2016/ubuntu/data/fastq/Cz-1h_S2_L001_R2_001.fastq.gz\r\n",
      "This is cutadapt 1.10 with Python 3.5.2\r\n",
      "Command line parameters: -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /tc2016/ubuntu/analysis/Cz-1h_S2_L001_R1_001.trimmed.fastq.gz -p /tc2016/ubuntu/analysis/Cz-1h_S2_L001_R2_001.trimmed.fastq.gz /tc2016/ubuntu/data/fastq/Cz-1h_S2_L001_R1_001.fastq.gz /tc2016/ubuntu/data/fastq/Cz-1h_S2_L001_R2_001.fastq.gz\r\n",
      "Trimming 2 adapters with at most 20.0% errors in paired-end mode ...\r\n",
      "Finished in 32.92 s (209 us/read; 0.29 M reads/minute).\r\n",
      "\r\n",
      "=== Summary ===\r\n",
      "\r\n",
      "Total read pairs processed:            157,409\r\n",
      "  Read 1 with adapter:                  44,291 (28.1%)\r\n",
      "  Read 2 with adapter:                  44,482 (28.3%)\r\n",
      "Pairs that were too short:                 138 (0.1%)\r\n",
      "Pairs written (passing filters):       157,271 (99.9%)\r\n",
      "\r\n",
      "Total basepairs processed:    23,926,168 bp\r\n",
      "  Read 1:    11,963,084 bp\r\n",
      "  Read 2:    11,963,084 bp\r\n",
      "Total written (filtered):     22,247,598 bp (93.0%)\r\n",
      "  Read 1:    11,127,051 bp\r\n",
      "  Read 2:    11,120,547 bp\r\n",
      "\r\n",
      "=== First read: Adapter 1 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 44291 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 15.2%\r\n",
      "  C: 35.4%\r\n",
      "  G: 24.4%\r\n",
      "  T: 25.0%\r\n",
      "  none/other: 0.1%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t2455\t2459.5\t0\t2455\r\n",
      "4\t2794\t614.9\t0\t1459 1335\r\n",
      "5\t2921\t153.7\t1\t1076 1845\r\n",
      "6\t1821\t38.4\t1\t906 915\r\n",
      "7\t1182\t9.6\t1\t992 190\r\n",
      "8\t1139\t2.4\t1\t901 219 19\r\n",
      "9\t1203\t0.6\t1\t1043 48 112\r\n",
      "10\t1129\t0.2\t2\t934 138 57\r\n",
      "11\t1050\t0.0\t2\t973 34 43\r\n",
      "12\t1020\t0.0\t2\t859 133 28\r\n",
      "13\t1071\t0.0\t2\t1014 34 23\r\n",
      "14\t1138\t0.0\t2\t1091 27 20\r\n",
      "15\t1087\t0.0\t2\t1037 31 19\r\n",
      "16\t1075\t0.0\t2\t1018 26 31\r\n",
      "17\t1083\t0.0\t2\t1031 31 21\r\n",
      "18\t1019\t0.0\t2\t962 37 20\r\n",
      "19\t1108\t0.0\t2\t1040 43 25\r\n",
      "20\t1077\t0.0\t2\t1036 24 17\r\n",
      "21\t1102\t0.0\t2\t1052 35 15\r\n",
      "22\t1036\t0.0\t2\t985 25 26\r\n",
      "23\t920\t0.0\t2\t865 30 25\r\n",
      "24\t1083\t0.0\t2\t1029 27 27\r\n",
      "25\t1316\t0.0\t2\t1270 29 17\r\n",
      "26\t1152\t0.0\t2\t1100 33 19\r\n",
      "27\t927\t0.0\t2\t880 27 20\r\n",
      "28\t903\t0.0\t2\t862 26 15\r\n",
      "29\t909\t0.0\t2\t861 24 24\r\n",
      "30\t949\t0.0\t2\t902 25 22\r\n",
      "31\t989\t0.0\t2\t942 31 16\r\n",
      "32\t837\t0.0\t2\t798 21 18\r\n",
      "33\t787\t0.0\t2\t735 25 27\r\n",
      "34\t735\t0.0\t2\t695 14 26\r\n",
      "35\t1084\t0.0\t2\t1048 17 19\r\n",
      "36\t1153\t0.0\t2\t1104 27 22\r\n",
      "37\t678\t0.0\t2\t640 22 16\r\n",
      "38\t474\t0.0\t2\t446 10 18\r\n",
      "39\t387\t0.0\t2\t352 12 23\r\n",
      "40\t267\t0.0\t2\t244 9 14\r\n",
      "41\t136\t0.0\t2\t118 2 16\r\n",
      "42\t129\t0.0\t2\t74 2 53\r\n",
      "43\t52\t0.0\t2\t33 2 17\r\n",
      "44\t46\t0.0\t2\t26 0 20\r\n",
      "45\t38\t0.0\t2\t25 0 13\r\n",
      "46\t116\t0.0\t2\t93 4 19\r\n",
      "47\t77\t0.0\t2\t61 1 15\r\n",
      "48\t34\t0.0\t2\t25 2 7\r\n",
      "49\t58\t0.0\t2\t27 2 29\r\n",
      "50\t31\t0.0\t2\t18 0 13\r\n",
      "51\t40\t0.0\t2\t1 0 39\r\n",
      "52\t18\t0.0\t2\t0 0 18\r\n",
      "53\t23\t0.0\t2\t0 1 22\r\n",
      "54\t8\t0.0\t2\t0 1 7\r\n",
      "55\t9\t0.0\t2\t0 0 9\r\n",
      "56\t20\t0.0\t2\t0 1 19\r\n",
      "57\t12\t0.0\t2\t0 3 9\r\n",
      "58\t6\t0.0\t2\t0 1 5\r\n",
      "59\t7\t0.0\t2\t0 0 7\r\n",
      "60\t11\t0.0\t2\t0 0 11\r\n",
      "61\t11\t0.0\t2\t0 0 11\r\n",
      "62\t7\t0.0\t2\t0 0 7\r\n",
      "63\t20\t0.0\t2\t0 1 19\r\n",
      "64\t24\t0.0\t2\t0 0 24\r\n",
      "65\t27\t0.0\t2\t0 2 25\r\n",
      "66\t20\t0.0\t2\t0 0 20\r\n",
      "67\t15\t0.0\t2\t0 0 15\r\n",
      "68\t81\t0.0\t2\t0 1 80\r\n",
      "69\t14\t0.0\t2\t0 1 13\r\n",
      "70\t30\t0.0\t2\t0 1 29\r\n",
      "71\t12\t0.0\t2\t0 0 12\r\n",
      "72\t8\t0.0\t2\t0 1 7\r\n",
      "73\t33\t0.0\t2\t0 1 32\r\n",
      "74\t25\t0.0\t2\t0 1 24\r\n",
      "75\t17\t0.0\t2\t0 0 17\r\n",
      "76\t16\t0.0\t2\t0 1 15\r\n",
      "\r\n",
      "=== Second read: Adapter 2 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 44482 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 15.9%\r\n",
      "  C: 34.7%\r\n",
      "  G: 25.2%\r\n",
      "  T: 23.5%\r\n",
      "  none/other: 0.8%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t2384\t2459.5\t0\t2384\r\n",
      "4\t2864\t614.9\t0\t1458 1406\r\n",
      "5\t3095\t153.7\t1\t1077 2018\r\n",
      "6\t1839\t38.4\t1\t886 953\r\n",
      "7\t1213\t9.6\t1\t1010 203\r\n",
      "8\t1135\t2.4\t1\t915 202 18\r\n",
      "9\t1193\t0.6\t1\t1072 30 91\r\n",
      "10\t1128\t0.2\t2\t942 123 63\r\n",
      "11\t1060\t0.0\t2\t994 16 50\r\n",
      "12\t1014\t0.0\t2\t876 121 17\r\n",
      "13\t1060\t0.0\t2\t1030 16 14\r\n",
      "14\t1147\t0.0\t2\t1108 13 26\r\n",
      "15\t1084\t0.0\t2\t1053 16 15\r\n",
      "16\t1069\t0.0\t2\t1035 15 19\r\n",
      "17\t1087\t0.0\t2\t1058 13 16\r\n",
      "18\t1013\t0.0\t2\t986 16 11\r\n",
      "19\t1110\t0.0\t2\t1071 10 29\r\n",
      "20\t1078\t0.0\t2\t1039 20 19\r\n",
      "21\t1103\t0.0\t2\t1078 10 15\r\n",
      "22\t1034\t0.0\t2\t1008 10 16\r\n",
      "23\t912\t0.0\t2\t621 275 16\r\n",
      "24\t1073\t0.0\t2\t719 330 24\r\n",
      "25\t1311\t0.0\t2\t899 401 11\r\n",
      "26\t1147\t0.0\t2\t763 372 12\r\n",
      "27\t919\t0.0\t2\t595 302 22\r\n",
      "28\t901\t0.0\t2\t600 288 13\r\n",
      "29\t896\t0.0\t2\t627 251 18\r\n",
      "30\t944\t0.0\t2\t605 324 15\r\n",
      "31\t987\t0.0\t2\t678 286 23\r\n",
      "32\t836\t0.0\t2\t553 267 16\r\n",
      "33\t780\t0.0\t2\t537 214 29\r\n",
      "34\t726\t0.0\t2\t498 6 222\r\n",
      "35\t1083\t0.0\t2\t1053 10 20\r\n",
      "36\t1140\t0.0\t2\t1114 17 9\r\n",
      "37\t673\t0.0\t2\t656 7 10\r\n",
      "38\t472\t0.0\t2\t454 5 13\r\n",
      "39\t385\t0.0\t2\t361 5 19\r\n",
      "40\t271\t0.0\t2\t247 5 19\r\n",
      "41\t141\t0.0\t2\t121 2 18\r\n",
      "42\t114\t0.0\t2\t76 2 36\r\n",
      "43\t49\t0.0\t2\t32 2 15\r\n",
      "44\t47\t0.0\t2\t26 0 21\r\n",
      "45\t44\t0.0\t2\t24 2 18\r\n",
      "46\t115\t0.0\t2\t96 0 19\r\n",
      "47\t82\t0.0\t2\t62 0 20\r\n",
      "48\t42\t0.0\t2\t28 2 12\r\n",
      "49\t57\t0.0\t2\t29 0 28\r\n",
      "50\t32\t0.0\t2\t18 0 14\r\n",
      "51\t35\t0.0\t2\t1 1 33\r\n",
      "52\t16\t0.0\t2\t0 0 16\r\n",
      "53\t40\t0.0\t2\t1 1 38\r\n",
      "54\t26\t0.0\t2\t1 1 24\r\n",
      "55\t14\t0.0\t2\t0 0 14\r\n",
      "56\t38\t0.0\t2\t1 0 37\r\n",
      "57\t24\t0.0\t2\t2 1 21\r\n",
      "58\t27\t0.0\t2\t0 0 27\r\n",
      "59\t22\t0.0\t2\t0 0 22\r\n",
      "60\t21\t0.0\t2\t0 1 20\r\n",
      "61\t22\t0.0\t2\t0 2 20\r\n",
      "62\t16\t0.0\t2\t0 0 16\r\n",
      "63\t30\t0.0\t2\t0 0 30\r\n",
      "64\t27\t0.0\t2\t0 0 27\r\n",
      "65\t17\t0.0\t2\t0 3 14\r\n",
      "66\t28\t0.0\t2\t0 0 28\r\n",
      "67\t18\t0.0\t2\t0 2 16\r\n",
      "68\t79\t0.0\t2\t0 6 73\r\n",
      "69\t23\t0.0\t2\t0 1 22\r\n",
      "70\t24\t0.0\t2\t0 1 23\r\n",
      "71\t7\t0.0\t2\t0 0 7\r\n",
      "72\t4\t0.0\t2\t0 0 4\r\n",
      "73\t4\t0.0\t2\t0 0 4\r\n",
      "74\t27\t0.0\t2\t0 0 27\r\n",
      "75\t1\t0.0\t2\t0 0 1\r\n",
      "76\t3\t0.0\t2\t0 0 3\r\n",
      "\r\n",
      "cutadapt -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /tc2016/ubuntu/analysis/Cz-3h_S9_L001_R1_001.trimmed.fastq.gz -p /tc2016/ubuntu/analysis/Cz-3h_S9_L001_R2_001.trimmed.fastq.gz /tc2016/ubuntu/data/fastq/Cz-3h_S9_L001_R1_001.fastq.gz /tc2016/ubuntu/data/fastq/Cz-3h_S9_L001_R2_001.fastq.gz\r\n",
      "This is cutadapt 1.10 with Python 3.5.2\r\n",
      "Command line parameters: -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /tc2016/ubuntu/analysis/Cz-3h_S9_L001_R1_001.trimmed.fastq.gz -p /tc2016/ubuntu/analysis/Cz-3h_S9_L001_R2_001.trimmed.fastq.gz /tc2016/ubuntu/data/fastq/Cz-3h_S9_L001_R1_001.fastq.gz /tc2016/ubuntu/data/fastq/Cz-3h_S9_L001_R2_001.fastq.gz\r\n",
      "Trimming 2 adapters with at most 20.0% errors in paired-end mode ...\r\n",
      "Finished in 81.55 s (196 us/read; 0.31 M reads/minute).\r\n",
      "\r\n",
      "=== Summary ===\r\n",
      "\r\n",
      "Total read pairs processed:            416,435\r\n",
      "  Read 1 with adapter:                 126,998 (30.5%)\r\n",
      "  Read 2 with adapter:                 127,870 (30.7%)\r\n",
      "Pairs that were too short:                 395 (0.1%)\r\n",
      "Pairs written (passing filters):       416,040 (99.9%)\r\n",
      "\r\n",
      "Total basepairs processed:    63,298,120 bp\r\n",
      "  Read 1:    31,649,060 bp\r\n",
      "  Read 2:    31,649,060 bp\r\n",
      "Total written (filtered):     58,266,477 bp (92.1%)\r\n",
      "  Read 1:    29,140,628 bp\r\n",
      "  Read 2:    29,125,849 bp\r\n",
      "\r\n",
      "=== First read: Adapter 1 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 126998 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 14.9%\r\n",
      "  C: 35.9%\r\n",
      "  G: 24.3%\r\n",
      "  T: 24.8%\r\n",
      "  none/other: 0.1%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t6283\t6506.8\t0\t6283\r\n",
      "4\t6963\t1626.7\t0\t3766 3197\r\n",
      "5\t7411\t406.7\t1\t2652 4759\r\n",
      "6\t4585\t101.7\t1\t2369 2216\r\n",
      "7\t3123\t25.4\t1\t2647 476\r\n",
      "8\t3324\t6.4\t1\t2794 486 44\r\n",
      "9\t3160\t1.6\t1\t2840 92 228\r\n",
      "10\t2888\t0.4\t2\t2447 282 159\r\n",
      "11\t2932\t0.1\t2\t2762 42 128\r\n",
      "12\t2947\t0.0\t2\t2587 311 49\r\n",
      "13\t2790\t0.0\t2\t2700 53 37\r\n",
      "14\t2937\t0.0\t2\t2850 39 48\r\n",
      "15\t3100\t0.0\t2\t3016 46 38\r\n",
      "16\t2895\t0.0\t2\t2813 40 42\r\n",
      "17\t2993\t0.0\t2\t2906 32 55\r\n",
      "18\t3115\t0.0\t2\t3023 35 57\r\n",
      "19\t3244\t0.0\t2\t3124 40 80\r\n",
      "20\t3260\t0.0\t2\t3172 42 46\r\n",
      "21\t3152\t0.0\t2\t3060 37 55\r\n",
      "22\t3172\t0.0\t2\t3103 32 37\r\n",
      "23\t2940\t0.0\t2\t2851 27 62\r\n",
      "24\t3241\t0.0\t2\t3150 39 52\r\n",
      "25\t3779\t0.0\t2\t3697 40 42\r\n",
      "26\t3547\t0.0\t2\t3468 34 45\r\n",
      "27\t3206\t0.0\t2\t3144 31 31\r\n",
      "28\t2985\t0.0\t2\t2903 29 53\r\n",
      "29\t3211\t0.0\t2\t3133 29 49\r\n",
      "30\t3328\t0.0\t2\t3252 31 45\r\n",
      "31\t3272\t0.0\t2\t3195 32 45\r\n",
      "32\t3081\t0.0\t2\t3006 43 32\r\n",
      "33\t2697\t0.0\t2\t2608 31 58\r\n",
      "34\t2518\t0.0\t2\t2439 25 54\r\n",
      "35\t2997\t0.0\t2\t2923 37 37\r\n",
      "36\t2963\t0.0\t2\t2906 26 31\r\n",
      "37\t1999\t0.0\t2\t1934 22 43\r\n",
      "38\t1474\t0.0\t2\t1417 10 47\r\n",
      "39\t1072\t0.0\t2\t999 22 51\r\n",
      "40\t775\t0.0\t2\t728 6 41\r\n",
      "41\t394\t0.0\t2\t331 2 61\r\n",
      "42\t283\t0.0\t2\t183 6 94\r\n",
      "43\t199\t0.0\t2\t131 6 62\r\n",
      "44\t154\t0.0\t2\t104 1 49\r\n",
      "45\t164\t0.0\t2\t123 4 37\r\n",
      "46\t369\t0.0\t2\t317 1 51\r\n",
      "47\t273\t0.0\t2\t231 3 39\r\n",
      "48\t156\t0.0\t2\t122 4 30\r\n",
      "49\t147\t0.0\t2\t88 4 55\r\n",
      "50\t80\t0.0\t2\t54 0 26\r\n",
      "51\t98\t0.0\t2\t4 1 93\r\n",
      "52\t34\t0.0\t2\t6 0 28\r\n",
      "53\t61\t0.0\t2\t2 2 57\r\n",
      "54\t33\t0.0\t2\t2 3 28\r\n",
      "55\t30\t0.0\t2\t1 1 28\r\n",
      "56\t55\t0.0\t2\t2 2 51\r\n",
      "57\t34\t0.0\t2\t4 4 26\r\n",
      "58\t39\t0.0\t2\t1 4 34\r\n",
      "59\t42\t0.0\t2\t0 1 41\r\n",
      "60\t25\t0.0\t2\t0 1 24\r\n",
      "61\t35\t0.0\t2\t0 0 35\r\n",
      "62\t29\t0.0\t2\t1 2 26\r\n",
      "63\t87\t0.0\t2\t0 2 85\r\n",
      "64\t48\t0.0\t2\t0 0 48\r\n",
      "65\t60\t0.0\t2\t0 2 58\r\n",
      "66\t57\t0.0\t2\t0 1 56\r\n",
      "67\t47\t0.0\t2\t0 2 45\r\n",
      "68\t195\t0.0\t2\t0 6 189\r\n",
      "69\t41\t0.0\t2\t0 3 38\r\n",
      "70\t74\t0.0\t2\t0 1 73\r\n",
      "71\t16\t0.0\t2\t0 0 16\r\n",
      "72\t26\t0.0\t2\t0 2 24\r\n",
      "73\t91\t0.0\t2\t0 3 88\r\n",
      "74\t77\t0.0\t2\t0 1 76\r\n",
      "75\t33\t0.0\t2\t0 1 32\r\n",
      "76\t53\t0.0\t2\t1 1 51\r\n",
      "\r\n",
      "=== Second read: Adapter 2 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 127870 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 15.2%\r\n",
      "  C: 35.9%\r\n",
      "  G: 24.4%\r\n",
      "  T: 23.7%\r\n",
      "  none/other: 0.8%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t6183\t6506.8\t0\t6183\r\n",
      "4\t7392\t1626.7\t0\t3752 3640\r\n",
      "5\t7849\t406.7\t1\t2638 5211\r\n",
      "6\t4774\t101.7\t1\t2117 2657\r\n",
      "7\t3199\t25.4\t1\t2624 575\r\n",
      "8\t3340\t6.4\t1\t2718 560 62\r\n",
      "9\t3130\t1.6\t1\t2803 97 230\r\n",
      "10\t2907\t0.4\t2\t2322 390 195\r\n",
      "11\t2917\t0.1\t2\t2708 78 131\r\n",
      "12\t2961\t0.0\t2\t2444 447 70\r\n",
      "13\t2800\t0.0\t2\t2652 93 55\r\n",
      "14\t2919\t0.0\t2\t2609 255 55\r\n",
      "15\t3098\t0.0\t2\t2746 288 64\r\n",
      "16\t2894\t0.0\t2\t2544 276 74\r\n",
      "17\t2989\t0.0\t2\t2633 289 67\r\n",
      "18\t3100\t0.0\t2\t2739 278 83\r\n",
      "19\t3235\t0.0\t2\t2851 298 86\r\n",
      "20\t3251\t0.0\t2\t2880 282 89\r\n",
      "21\t3141\t0.0\t2\t2751 321 69\r\n",
      "22\t3181\t0.0\t2\t2833 274 74\r\n",
      "23\t2903\t0.0\t2\t1919 921 63\r\n",
      "24\t3204\t0.0\t2\t2103 1022 79\r\n",
      "25\t3754\t0.0\t2\t2459 945 350\r\n",
      "26\t3539\t0.0\t2\t2546 909 84\r\n",
      "27\t3195\t0.0\t2\t2348 758 89\r\n",
      "28\t2947\t0.0\t2\t2107 790 50\r\n",
      "29\t3173\t0.0\t2\t2298 775 100\r\n",
      "30\t3307\t0.0\t2\t2384 855 68\r\n",
      "31\t3251\t0.0\t2\t2296 847 108\r\n",
      "32\t3072\t0.0\t2\t2195 820 57\r\n",
      "33\t2694\t0.0\t2\t1890 692 112\r\n",
      "34\t2520\t0.0\t2\t1746 61 713\r\n",
      "35\t3010\t0.0\t2\t2842 78 90\r\n",
      "36\t2968\t0.0\t2\t2832 69 67\r\n",
      "37\t1991\t0.0\t2\t1885 49 57\r\n",
      "38\t1468\t0.0\t2\t1376 36 56\r\n",
      "39\t1072\t0.0\t2\t985 29 58\r\n",
      "40\t784\t0.0\t2\t704 24 56\r\n",
      "41\t385\t0.0\t2\t320 15 50\r\n",
      "42\t272\t0.0\t2\t178 10 84\r\n",
      "43\t195\t0.0\t2\t126 8 61\r\n",
      "44\t156\t0.0\t2\t102 1 53\r\n",
      "45\t168\t0.0\t2\t118 7 43\r\n",
      "46\t360\t0.0\t2\t307 13 40\r\n",
      "47\t270\t0.0\t2\t225 7 38\r\n",
      "48\t164\t0.0\t2\t114 10 40\r\n",
      "49\t170\t0.0\t2\t84 6 80\r\n",
      "50\t79\t0.0\t2\t52 2 25\r\n",
      "51\t91\t0.0\t2\t4 1 86\r\n",
      "52\t41\t0.0\t2\t6 0 35\r\n",
      "53\t97\t0.0\t2\t2 9 86\r\n",
      "54\t64\t0.0\t2\t3 2 59\r\n",
      "55\t42\t0.0\t2\t2 2 38\r\n",
      "56\t94\t0.0\t2\t4 0 90\r\n",
      "57\t58\t0.0\t2\t7 0 51\r\n",
      "58\t64\t0.0\t2\t3 1 60\r\n",
      "59\t69\t0.0\t2\t0 2 67\r\n",
      "60\t44\t0.0\t2\t0 0 44\r\n",
      "61\t80\t0.0\t2\t0 0 80\r\n",
      "62\t26\t0.0\t2\t0 2 24\r\n",
      "63\t86\t0.0\t2\t0 2 84\r\n",
      "64\t70\t0.0\t2\t0 3 67\r\n",
      "65\t51\t0.0\t2\t0 4 47\r\n",
      "66\t56\t0.0\t2\t0 1 55\r\n",
      "67\t39\t0.0\t2\t0 0 39\r\n",
      "68\t202\t0.0\t2\t0 5 197\r\n",
      "69\t50\t0.0\t2\t0 2 48\r\n",
      "70\t70\t0.0\t2\t0 1 69\r\n",
      "71\t29\t0.0\t2\t0 0 29\r\n",
      "72\t12\t0.0\t2\t0 0 12\r\n",
      "73\t12\t0.0\t2\t0 0 12\r\n",
      "74\t75\t0.0\t2\t0 0 75\r\n",
      "75\t7\t0.0\t2\t0 0 7\r\n",
      "76\t10\t0.0\t2\t0 0 10\r\n",
      "\r\n",
      "cutadapt -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /tc2016/ubuntu/analysis/DMSO-1h_S6_L001_R1_001.trimmed.fastq.gz -p /tc2016/ubuntu/analysis/DMSO-1h_S6_L001_R2_001.trimmed.fastq.gz /tc2016/ubuntu/data/fastq/DMSO-1h_S6_L001_R1_001.fastq.gz /tc2016/ubuntu/data/fastq/DMSO-1h_S6_L001_R2_001.fastq.gz\r\n",
      "This is cutadapt 1.10 with Python 3.5.2\r\n",
      "Command line parameters: -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /tc2016/ubuntu/analysis/DMSO-1h_S6_L001_R1_001.trimmed.fastq.gz -p /tc2016/ubuntu/analysis/DMSO-1h_S6_L001_R2_001.trimmed.fastq.gz /tc2016/ubuntu/data/fastq/DMSO-1h_S6_L001_R1_001.fastq.gz /tc2016/ubuntu/data/fastq/DMSO-1h_S6_L001_R2_001.fastq.gz\r\n",
      "Trimming 2 adapters with at most 20.0% errors in paired-end mode ...\r\n",
      "Finished in 26.01 s (203 us/read; 0.29 M reads/minute).\r\n",
      "\r\n",
      "=== Summary ===\r\n",
      "\r\n",
      "Total read pairs processed:            127,835\r\n",
      "  Read 1 with adapter:                  31,575 (24.7%)\r\n",
      "  Read 2 with adapter:                  32,088 (25.1%)\r\n",
      "Pairs that were too short:                 135 (0.1%)\r\n",
      "Pairs written (passing filters):       127,700 (99.9%)\r\n",
      "\r\n",
      "Total basepairs processed:    19,430,920 bp\r\n",
      "  Read 1:     9,715,460 bp\r\n",
      "  Read 2:     9,715,460 bp\r\n",
      "Total written (filtered):     18,291,881 bp (94.1%)\r\n",
      "  Read 1:     9,149,968 bp\r\n",
      "  Read 2:     9,141,913 bp\r\n",
      "\r\n",
      "=== First read: Adapter 1 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 31575 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 15.9%\r\n",
      "  C: 34.9%\r\n",
      "  G: 24.3%\r\n",
      "  T: 24.8%\r\n",
      "  none/other: 0.1%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t1948\t1997.4\t0\t1948\r\n",
      "4\t2219\t499.4\t0\t1083 1136\r\n",
      "5\t2294\t124.8\t1\t756 1538\r\n",
      "6\t1387\t31.2\t1\t664 723\r\n",
      "7\t923\t7.8\t1\t749 174\r\n",
      "8\t948\t2.0\t1\t766 169 13\r\n",
      "9\t909\t0.5\t1\t802 41 66\r\n",
      "10\t883\t0.1\t2\t697 115 71\r\n",
      "11\t812\t0.0\t2\t753 27 32\r\n",
      "12\t707\t0.0\t2\t600 93 14\r\n",
      "13\t782\t0.0\t2\t731 35 16\r\n",
      "14\t797\t0.0\t2\t750 25 22\r\n",
      "15\t754\t0.0\t2\t714 26 14\r\n",
      "16\t806\t0.0\t2\t748 34 24\r\n",
      "17\t791\t0.0\t2\t738 34 19\r\n",
      "18\t746\t0.0\t2\t694 27 25\r\n",
      "19\t767\t0.0\t2\t726 24 17\r\n",
      "20\t817\t0.0\t2\t765 34 18\r\n",
      "21\t804\t0.0\t2\t748 34 22\r\n",
      "22\t810\t0.0\t2\t750 36 24\r\n",
      "23\t670\t0.0\t2\t622 23 25\r\n",
      "24\t748\t0.0\t2\t704 27 17\r\n",
      "25\t798\t0.0\t2\t763 16 19\r\n",
      "26\t793\t0.0\t2\t757 28 8\r\n",
      "27\t629\t0.0\t2\t595 21 13\r\n",
      "28\t585\t0.0\t2\t548 24 13\r\n",
      "29\t638\t0.0\t2\t597 26 15\r\n",
      "30\t665\t0.0\t2\t632 21 12\r\n",
      "31\t603\t0.0\t2\t570 22 11\r\n",
      "32\t555\t0.0\t2\t527 15 13\r\n",
      "33\t498\t0.0\t2\t462 14 22\r\n",
      "34\t456\t0.0\t2\t413 25 18\r\n",
      "35\t591\t0.0\t2\t560 18 13\r\n",
      "36\t547\t0.0\t2\t504 21 22\r\n",
      "37\t370\t0.0\t2\t328 23 19\r\n",
      "38\t268\t0.0\t2\t258 6 4\r\n",
      "39\t200\t0.0\t2\t169 6 25\r\n",
      "40\t137\t0.0\t2\t124 6 7\r\n",
      "41\t85\t0.0\t2\t61 4 20\r\n",
      "42\t74\t0.0\t2\t38 1 35\r\n",
      "43\t40\t0.0\t2\t17 2 21\r\n",
      "44\t23\t0.0\t2\t7 1 15\r\n",
      "45\t25\t0.0\t2\t13 0 12\r\n",
      "46\t48\t0.0\t2\t31 1 16\r\n",
      "47\t52\t0.0\t2\t32 2 18\r\n",
      "48\t31\t0.0\t2\t24 1 6\r\n",
      "49\t47\t0.0\t2\t17 0 30\r\n",
      "50\t27\t0.0\t2\t13 0 14\r\n",
      "51\t34\t0.0\t2\t1 0 33\r\n",
      "52\t17\t0.0\t2\t0 0 17\r\n",
      "53\t16\t0.0\t2\t0 0 16\r\n",
      "54\t5\t0.0\t2\t0 1 4\r\n",
      "55\t11\t0.0\t2\t0 0 11\r\n",
      "56\t7\t0.0\t2\t0 1 6\r\n",
      "57\t9\t0.0\t2\t0 1 8\r\n",
      "58\t11\t0.0\t2\t0 0 11\r\n",
      "59\t10\t0.0\t2\t0 1 9\r\n",
      "60\t5\t0.0\t2\t0 0 5\r\n",
      "61\t10\t0.0\t2\t0 0 10\r\n",
      "62\t5\t0.0\t2\t0 0 5\r\n",
      "63\t24\t0.0\t2\t0 0 24\r\n",
      "64\t11\t0.0\t2\t0 1 10\r\n",
      "65\t14\t0.0\t2\t0 1 13\r\n",
      "66\t12\t0.0\t2\t0 0 12\r\n",
      "67\t13\t0.0\t2\t0 0 13\r\n",
      "68\t78\t0.0\t2\t0 2 76\r\n",
      "69\t27\t0.0\t2\t0 1 26\r\n",
      "70\t28\t0.0\t2\t0 0 28\r\n",
      "71\t10\t0.0\t2\t0 0 10\r\n",
      "72\t16\t0.0\t2\t0 0 16\r\n",
      "73\t35\t0.0\t2\t0 0 35\r\n",
      "74\t36\t0.0\t2\t0 0 36\r\n",
      "75\t12\t0.0\t2\t0 0 12\r\n",
      "76\t12\t0.0\t2\t0 0 12\r\n",
      "\r\n",
      "=== Second read: Adapter 2 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 32088 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 17.0%\r\n",
      "  C: 33.3%\r\n",
      "  G: 25.6%\r\n",
      "  T: 23.4%\r\n",
      "  none/other: 0.7%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t1996\t1997.4\t0\t1996\r\n",
      "4\t2329\t499.4\t0\t1113 1216\r\n",
      "5\t2480\t124.8\t1\t786 1694\r\n",
      "6\t1485\t31.2\t1\t681 804\r\n",
      "7\t958\t7.8\t1\t768 190\r\n",
      "8\t970\t2.0\t1\t797 157 16\r\n",
      "9\t932\t0.5\t1\t827 22 83\r\n",
      "10\t877\t0.1\t2\t731 84 62\r\n",
      "11\t817\t0.0\t2\t775 8 34\r\n",
      "12\t705\t0.0\t2\t609 86 10\r\n",
      "13\t791\t0.0\t2\t765 5 21\r\n",
      "14\t791\t0.0\t2\t767 15 9\r\n",
      "15\t761\t0.0\t2\t739 7 15\r\n",
      "16\t808\t0.0\t2\t783 6 19\r\n",
      "17\t794\t0.0\t2\t774 3 17\r\n",
      "18\t735\t0.0\t2\t719 1 15\r\n",
      "19\t774\t0.0\t2\t738 8 28\r\n",
      "20\t818\t0.0\t2\t794 12 12\r\n",
      "21\t800\t0.0\t2\t777 12 11\r\n",
      "22\t801\t0.0\t2\t780 8 13\r\n",
      "23\t665\t0.0\t2\t488 162 15\r\n",
      "24\t744\t0.0\t2\t521 212 11\r\n",
      "25\t799\t0.0\t2\t574 213 12\r\n",
      "26\t789\t0.0\t2\t592 183 14\r\n",
      "27\t627\t0.0\t2\t447 167 13\r\n",
      "28\t588\t0.0\t2\t422 151 15\r\n",
      "29\t636\t0.0\t2\t471 150 15\r\n",
      "30\t665\t0.0\t2\t491 160 14\r\n",
      "31\t605\t0.0\t2\t449 139 17\r\n",
      "32\t559\t0.0\t2\t423 125 11\r\n",
      "33\t489\t0.0\t2\t369 101 19\r\n",
      "34\t455\t0.0\t2\t336 5 114\r\n",
      "35\t590\t0.0\t2\t569 8 13\r\n",
      "36\t545\t0.0\t2\t524 6 15\r\n",
      "37\t362\t0.0\t2\t347 2 13\r\n",
      "38\t277\t0.0\t2\t258 1 18\r\n",
      "39\t191\t0.0\t2\t171 5 15\r\n",
      "40\t144\t0.0\t2\t128 3 13\r\n",
      "41\t80\t0.0\t2\t65 0 15\r\n",
      "42\t61\t0.0\t2\t36 2 23\r\n",
      "43\t41\t0.0\t2\t18 1 22\r\n",
      "44\t21\t0.0\t2\t7 0 14\r\n",
      "45\t21\t0.0\t2\t12 1 8\r\n",
      "46\t53\t0.0\t2\t32 0 21\r\n",
      "47\t54\t0.0\t2\t31 1 22\r\n",
      "48\t38\t0.0\t2\t24 2 12\r\n",
      "49\t36\t0.0\t2\t16 1 19\r\n",
      "50\t18\t0.0\t2\t12 2 4\r\n",
      "51\t33\t0.0\t2\t1 0 32\r\n",
      "52\t15\t0.0\t2\t0 0 15\r\n",
      "53\t39\t0.0\t2\t0 1 38\r\n",
      "54\t22\t0.0\t2\t1 2 19\r\n",
      "55\t5\t0.0\t2\t0 0 5\r\n",
      "56\t43\t0.0\t2\t1 0 42\r\n",
      "57\t20\t0.0\t2\t1 0 19\r\n",
      "58\t15\t0.0\t2\t0 0 15\r\n",
      "59\t23\t0.0\t2\t0 2 21\r\n",
      "60\t17\t0.0\t2\t0 1 16\r\n",
      "61\t24\t0.0\t2\t0 0 24\r\n",
      "62\t12\t0.0\t2\t0 0 12\r\n",
      "63\t35\t0.0\t2\t0 0 35\r\n",
      "64\t25\t0.0\t2\t0 2 23\r\n",
      "65\t13\t0.0\t2\t0 1 12\r\n",
      "66\t15\t0.0\t2\t0 0 15\r\n",
      "67\t16\t0.0\t2\t0 2 14\r\n",
      "68\t72\t0.0\t2\t0 5 67\r\n",
      "69\t15\t0.0\t2\t0 0 15\r\n",
      "70\t24\t0.0\t2\t0 0 24\r\n",
      "71\t6\t0.0\t2\t0 0 6\r\n",
      "72\t5\t0.0\t2\t0 0 5\r\n",
      "73\t1\t0.0\t2\t0 0 1\r\n",
      "74\t15\t0.0\t2\t0 0 15\r\n",
      "75\t1\t0.0\t2\t0 0 1\r\n",
      "76\t2\t0.0\t2\t0 0 2\r\n",
      "\r\n",
      "cutadapt -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /tc2016/ubuntu/analysis/DMSO-3h_S13_L001_R1_001.trimmed.fastq.gz -p /tc2016/ubuntu/analysis/DMSO-3h_S13_L001_R2_001.trimmed.fastq.gz /tc2016/ubuntu/data/fastq/DMSO-3h_S13_L001_R1_001.fastq.gz /tc2016/ubuntu/data/fastq/DMSO-3h_S13_L001_R2_001.fastq.gz\r\n",
      "This is cutadapt 1.10 with Python 3.5.2\r\n",
      "Command line parameters: -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /tc2016/ubuntu/analysis/DMSO-3h_S13_L001_R1_001.trimmed.fastq.gz -p /tc2016/ubuntu/analysis/DMSO-3h_S13_L001_R2_001.trimmed.fastq.gz /tc2016/ubuntu/data/fastq/DMSO-3h_S13_L001_R1_001.fastq.gz /tc2016/ubuntu/data/fastq/DMSO-3h_S13_L001_R2_001.fastq.gz\r\n",
      "Trimming 2 adapters with at most 20.0% errors in paired-end mode ...\r\n",
      "Finished in 84.67 s (196 us/read; 0.31 M reads/minute).\r\n",
      "\r\n",
      "=== Summary ===\r\n",
      "\r\n",
      "Total read pairs processed:            432,427\r\n",
      "  Read 1 with adapter:                 131,949 (30.5%)\r\n",
      "  Read 2 with adapter:                 132,578 (30.7%)\r\n",
      "Pairs that were too short:                 356 (0.1%)\r\n",
      "Pairs written (passing filters):       432,071 (99.9%)\r\n",
      "\r\n",
      "Total basepairs processed:    65,728,904 bp\r\n",
      "  Read 1:    32,864,452 bp\r\n",
      "  Read 2:    32,864,452 bp\r\n",
      "Total written (filtered):     60,503,216 bp (92.0%)\r\n",
      "  Read 1:    30,255,983 bp\r\n",
      "  Read 2:    30,247,233 bp\r\n",
      "\r\n",
      "=== First read: Adapter 1 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 131949 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 14.7%\r\n",
      "  C: 36.4%\r\n",
      "  G: 24.4%\r\n",
      "  T: 24.4%\r\n",
      "  none/other: 0.0%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t6495\t6756.7\t0\t6495\r\n",
      "4\t7323\t1689.2\t0\t3880 3443\r\n",
      "5\t7831\t422.3\t1\t2905 4926\r\n",
      "6\t5002\t105.6\t1\t2571 2431\r\n",
      "7\t3222\t26.4\t1\t2727 495\r\n",
      "8\t3396\t6.6\t1\t2777 569 50\r\n",
      "9\t3189\t1.6\t1\t2858 66 265\r\n",
      "10\t3022\t0.4\t2\t2573 311 138\r\n",
      "11\t2985\t0.1\t2\t2816 40 129\r\n",
      "12\t3021\t0.0\t2\t2617 340 64\r\n",
      "13\t2969\t0.0\t2\t2874 50 45\r\n",
      "14\t3124\t0.0\t2\t3045 41 38\r\n",
      "15\t3117\t0.0\t2\t3048 31 38\r\n",
      "16\t3071\t0.0\t2\t2991 37 43\r\n",
      "17\t3089\t0.0\t2\t2997 33 59\r\n",
      "18\t3053\t0.0\t2\t2973 33 47\r\n",
      "19\t3340\t0.0\t2\t3225 31 84\r\n",
      "20\t3273\t0.0\t2\t3188 36 49\r\n",
      "21\t3334\t0.0\t2\t3252 43 39\r\n",
      "22\t3125\t0.0\t2\t3047 39 39\r\n",
      "23\t3061\t0.0\t2\t2956 29 76\r\n",
      "24\t3298\t0.0\t2\t3227 23 48\r\n",
      "25\t3794\t0.0\t2\t3724 48 22\r\n",
      "26\t3712\t0.0\t2\t3623 30 59\r\n",
      "27\t3054\t0.0\t2\t2990 21 43\r\n",
      "28\t2994\t0.0\t2\t2912 27 55\r\n",
      "29\t3180\t0.0\t2\t3108 30 42\r\n",
      "30\t3367\t0.0\t2\t3283 35 49\r\n",
      "31\t3347\t0.0\t2\t3273 42 32\r\n",
      "32\t3262\t0.0\t2\t3177 51 34\r\n",
      "33\t2862\t0.0\t2\t2776 22 64\r\n",
      "34\t2689\t0.0\t2\t2608 24 57\r\n",
      "35\t3375\t0.0\t2\t3311 33 31\r\n",
      "36\t3357\t0.0\t2\t3268 33 56\r\n",
      "37\t2299\t0.0\t2\t2243 21 35\r\n",
      "38\t1541\t0.0\t2\t1476 23 42\r\n",
      "39\t1168\t0.0\t2\t1090 13 65\r\n",
      "40\t844\t0.0\t2\t782 10 52\r\n",
      "41\t446\t0.0\t2\t394 7 45\r\n",
      "42\t296\t0.0\t2\t203 7 86\r\n",
      "43\t186\t0.0\t2\t131 2 53\r\n",
      "44\t127\t0.0\t2\t87 2 38\r\n",
      "45\t175\t0.0\t2\t141 4 30\r\n",
      "46\t453\t0.0\t2\t392 7 54\r\n",
      "47\t304\t0.0\t2\t255 2 47\r\n",
      "48\t167\t0.0\t2\t123 8 36\r\n",
      "49\t147\t0.0\t2\t97 3 47\r\n",
      "50\t89\t0.0\t2\t51 0 38\r\n",
      "51\t109\t0.0\t2\t2 3 104\r\n",
      "52\t37\t0.0\t2\t2 2 33\r\n",
      "53\t79\t0.0\t2\t1 3 75\r\n",
      "54\t28\t0.0\t2\t2 0 26\r\n",
      "55\t33\t0.0\t2\t0 2 31\r\n",
      "56\t47\t0.0\t2\t1 2 44\r\n",
      "57\t31\t0.0\t2\t5 5 21\r\n",
      "58\t30\t0.0\t2\t1 5 24\r\n",
      "59\t45\t0.0\t2\t0 1 44\r\n",
      "60\t27\t0.0\t2\t0 1 26\r\n",
      "61\t45\t0.0\t2\t0 4 41\r\n",
      "62\t23\t0.0\t2\t0 0 23\r\n",
      "63\t85\t0.0\t2\t0 2 83\r\n",
      "64\t42\t0.0\t2\t0 0 42\r\n",
      "65\t48\t0.0\t2\t0 1 47\r\n",
      "66\t56\t0.0\t2\t0 0 56\r\n",
      "67\t39\t0.0\t2\t0 1 38\r\n",
      "68\t162\t0.0\t2\t0 3 159\r\n",
      "69\t53\t0.0\t2\t0 2 51\r\n",
      "70\t80\t0.0\t2\t0 0 80\r\n",
      "71\t25\t0.0\t2\t0 0 25\r\n",
      "72\t20\t0.0\t2\t0 0 20\r\n",
      "73\t81\t0.0\t2\t0 4 77\r\n",
      "74\t75\t0.0\t2\t0 2 73\r\n",
      "75\t44\t0.0\t2\t0 2 42\r\n",
      "76\t30\t0.0\t2\t0 0 30\r\n",
      "\r\n",
      "=== Second read: Adapter 2 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 132578 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 15.3%\r\n",
      "  C: 36.0%\r\n",
      "  G: 24.6%\r\n",
      "  T: 23.4%\r\n",
      "  none/other: 0.8%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t6441\t6756.7\t0\t6441\r\n",
      "4\t7733\t1689.2\t0\t3871 3862\r\n",
      "5\t8400\t422.3\t1\t2879 5521\r\n",
      "6\t5015\t105.6\t1\t2228 2787\r\n",
      "7\t3299\t26.4\t1\t2697 602\r\n",
      "8\t3329\t6.6\t1\t2706 586 37\r\n",
      "9\t3139\t1.6\t1\t2818 103 218\r\n",
      "10\t3022\t0.4\t2\t2436 411 175\r\n",
      "11\t2993\t0.1\t2\t2758 92 143\r\n",
      "12\t2984\t0.0\t2\t2530 402 52\r\n",
      "13\t2971\t0.0\t2\t2835 74 62\r\n",
      "14\t3112\t0.0\t2\t2779 275 58\r\n",
      "15\t3112\t0.0\t2\t2813 253 46\r\n",
      "16\t3069\t0.0\t2\t2728 270 71\r\n",
      "17\t3088\t0.0\t2\t2770 245 73\r\n",
      "18\t3048\t0.0\t2\t2734 229 85\r\n",
      "19\t3349\t0.0\t2\t2973 257 119\r\n",
      "20\t3267\t0.0\t2\t2948 239 80\r\n",
      "21\t3337\t0.0\t2\t2972 296 69\r\n",
      "22\t3135\t0.0\t2\t2797 262 76\r\n",
      "23\t3005\t0.0\t2\t1943 987 75\r\n",
      "24\t3287\t0.0\t2\t2148 1049 90\r\n",
      "25\t3784\t0.0\t2\t2487 986 311\r\n",
      "26\t3684\t0.0\t2\t2663 955 66\r\n",
      "27\t3035\t0.0\t2\t2208 731 96\r\n",
      "28\t2975\t0.0\t2\t2144 768 63\r\n",
      "29\t3164\t0.0\t2\t2289 774 101\r\n",
      "30\t3338\t0.0\t2\t2413 867 58\r\n",
      "31\t3332\t0.0\t2\t2374 879 79\r\n",
      "32\t3244\t0.0\t2\t2302 869 73\r\n",
      "33\t2832\t0.0\t2\t2035 700 97\r\n",
      "34\t2669\t0.0\t2\t1903 58 708\r\n",
      "35\t3358\t0.0\t2\t3212 93 53\r\n",
      "36\t3344\t0.0\t2\t3188 82 74\r\n",
      "37\t2288\t0.0\t2\t2190 57 41\r\n",
      "38\t1527\t0.0\t2\t1424 62 41\r\n",
      "39\t1153\t0.0\t2\t1067 26 60\r\n",
      "40\t829\t0.0\t2\t749 27 53\r\n",
      "41\t448\t0.0\t2\t385 13 50\r\n",
      "42\t291\t0.0\t2\t193 14 84\r\n",
      "43\t172\t0.0\t2\t129 5 38\r\n",
      "44\t135\t0.0\t2\t81 3 51\r\n",
      "45\t182\t0.0\t2\t140 4 38\r\n",
      "46\t440\t0.0\t2\t376 16 48\r\n",
      "47\t305\t0.0\t2\t250 5 50\r\n",
      "48\t158\t0.0\t2\t119 11 28\r\n",
      "49\t167\t0.0\t2\t95 4 68\r\n",
      "50\t83\t0.0\t2\t48 4 31\r\n",
      "51\t80\t0.0\t2\t2 1 77\r\n",
      "52\t37\t0.0\t2\t3 1 33\r\n",
      "53\t118\t0.0\t2\t2 5 111\r\n",
      "54\t60\t0.0\t2\t2 3 55\r\n",
      "55\t42\t0.0\t2\t1 0 41\r\n",
      "56\t83\t0.0\t2\t2 1 80\r\n",
      "57\t53\t0.0\t2\t9 1 43\r\n",
      "58\t58\t0.0\t2\t3 0 55\r\n",
      "59\t48\t0.0\t2\t0 4 44\r\n",
      "60\t57\t0.0\t2\t0 0 57\r\n",
      "61\t70\t0.0\t2\t1 2 67\r\n",
      "62\t40\t0.0\t2\t0 2 38\r\n",
      "63\t87\t0.0\t2\t0 1 86\r\n",
      "64\t64\t0.0\t2\t0 1 63\r\n",
      "65\t56\t0.0\t2\t0 3 53\r\n",
      "66\t68\t0.0\t2\t0 2 66\r\n",
      "67\t26\t0.0\t2\t0 0 26\r\n",
      "68\t217\t0.0\t2\t0 10 207\r\n",
      "69\t45\t0.0\t2\t0 5 40\r\n",
      "70\t69\t0.0\t2\t0 1 68\r\n",
      "71\t22\t0.0\t2\t0 0 22\r\n",
      "72\t11\t0.0\t2\t0 0 11\r\n",
      "73\t12\t0.0\t2\t0 1 11\r\n",
      "74\t71\t0.0\t2\t0 2 69\r\n",
      "75\t3\t0.0\t2\t0 0 3\r\n",
      "76\t9\t0.0\t2\t0 0 9\r\n",
      "\r\n",
      "cutadapt -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /tc2016/ubuntu/analysis/Kz-1h_S1_L001_R1_001.trimmed.fastq.gz -p /tc2016/ubuntu/analysis/Kz-1h_S1_L001_R2_001.trimmed.fastq.gz /tc2016/ubuntu/data/fastq/Kz-1h_S1_L001_R1_001.fastq.gz /tc2016/ubuntu/data/fastq/Kz-1h_S1_L001_R2_001.fastq.gz\r\n",
      "This is cutadapt 1.10 with Python 3.5.2\r\n",
      "Command line parameters: -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /tc2016/ubuntu/analysis/Kz-1h_S1_L001_R1_001.trimmed.fastq.gz -p /tc2016/ubuntu/analysis/Kz-1h_S1_L001_R2_001.trimmed.fastq.gz /tc2016/ubuntu/data/fastq/Kz-1h_S1_L001_R1_001.fastq.gz /tc2016/ubuntu/data/fastq/Kz-1h_S1_L001_R2_001.fastq.gz\r\n",
      "Trimming 2 adapters with at most 20.0% errors in paired-end mode ...\r\n",
      "Finished in 12.27 s (195 us/read; 0.31 M reads/minute).\r\n",
      "\r\n",
      "=== Summary ===\r\n",
      "\r\n",
      "Total read pairs processed:             63,018\r\n",
      "  Read 1 with adapter:                  15,855 (25.2%)\r\n",
      "  Read 2 with adapter:                  16,048 (25.5%)\r\n",
      "Pairs that were too short:                  50 (0.1%)\r\n",
      "Pairs written (passing filters):        62,968 (99.9%)\r\n",
      "\r\n",
      "Total basepairs processed:     9,578,736 bp\r\n",
      "  Read 1:     4,789,368 bp\r\n",
      "  Read 2:     4,789,368 bp\r\n",
      "Total written (filtered):      9,006,059 bp (94.0%)\r\n",
      "  Read 1:     4,505,288 bp\r\n",
      "  Read 2:     4,500,771 bp\r\n",
      "\r\n",
      "=== First read: Adapter 1 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 15855 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 14.7%\r\n",
      "  C: 35.9%\r\n",
      "  G: 24.3%\r\n",
      "  T: 25.1%\r\n",
      "  none/other: 0.0%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t976\t984.7\t0\t976\r\n",
      "4\t1141\t246.2\t0\t597 544\r\n",
      "5\t1147\t61.5\t1\t385 762\r\n",
      "6\t727\t15.4\t1\t349 378\r\n",
      "7\t474\t3.8\t1\t373 101\r\n",
      "8\t470\t1.0\t1\t379 83 8\r\n",
      "9\t428\t0.2\t1\t362 22 44\r\n",
      "10\t392\t0.1\t2\t291 67 34\r\n",
      "11\t392\t0.0\t2\t349 21 22\r\n",
      "12\t402\t0.0\t2\t340 56 6\r\n",
      "13\t417\t0.0\t2\t380 29 8\r\n",
      "14\t409\t0.0\t2\t377 23 9\r\n",
      "15\t425\t0.0\t2\t391 22 12\r\n",
      "16\t391\t0.0\t2\t358 14 19\r\n",
      "17\t376\t0.0\t2\t349 15 12\r\n",
      "18\t368\t0.0\t2\t346 13 9\r\n",
      "19\t414\t0.0\t2\t370 28 16\r\n",
      "20\t388\t0.0\t2\t361 19 8\r\n",
      "21\t404\t0.0\t2\t374 15 15\r\n",
      "22\t370\t0.0\t2\t338 20 12\r\n",
      "23\t303\t0.0\t2\t279 14 10\r\n",
      "24\t363\t0.0\t2\t353 6 4\r\n",
      "25\t432\t0.0\t2\t403 18 11\r\n",
      "26\t386\t0.0\t2\t354 19 13\r\n",
      "27\t304\t0.0\t2\t287 12 5\r\n",
      "28\t274\t0.0\t2\t253 14 7\r\n",
      "29\t285\t0.0\t2\t268 8 9\r\n",
      "30\t320\t0.0\t2\t288 18 14\r\n",
      "31\t295\t0.0\t2\t276 8 11\r\n",
      "32\t262\t0.0\t2\t249 9 4\r\n",
      "33\t285\t0.0\t2\t263 8 14\r\n",
      "34\t225\t0.0\t2\t213 7 5\r\n",
      "35\t294\t0.0\t2\t280 10 4\r\n",
      "36\t303\t0.0\t2\t282 16 5\r\n",
      "37\t202\t0.0\t2\t188 11 3\r\n",
      "38\t141\t0.0\t2\t123 9 9\r\n",
      "39\t109\t0.0\t2\t88 7 14\r\n",
      "40\t84\t0.0\t2\t76 2 6\r\n",
      "41\t41\t0.0\t2\t31 2 8\r\n",
      "42\t32\t0.0\t2\t13 2 17\r\n",
      "43\t20\t0.0\t2\t10 0 10\r\n",
      "44\t12\t0.0\t2\t5 0 7\r\n",
      "45\t12\t0.0\t2\t8 0 4\r\n",
      "46\t28\t0.0\t2\t21 1 6\r\n",
      "47\t44\t0.0\t2\t28 1 15\r\n",
      "48\t10\t0.0\t2\t4 0 6\r\n",
      "49\t23\t0.0\t2\t8 1 14\r\n",
      "50\t12\t0.0\t2\t5 1 6\r\n",
      "51\t17\t0.0\t2\t1 0 16\r\n",
      "52\t5\t0.0\t2\t1 0 4\r\n",
      "53\t13\t0.0\t2\t0 0 13\r\n",
      "54\t7\t0.0\t2\t0 0 7\r\n",
      "55\t6\t0.0\t2\t0 0 6\r\n",
      "56\t6\t0.0\t2\t0 0 6\r\n",
      "57\t3\t0.0\t2\t0 0 3\r\n",
      "58\t4\t0.0\t2\t0 1 3\r\n",
      "59\t1\t0.0\t2\t0 0 1\r\n",
      "60\t4\t0.0\t2\t0 0 4\r\n",
      "61\t7\t0.0\t2\t0 0 7\r\n",
      "62\t3\t0.0\t2\t0 0 3\r\n",
      "63\t11\t0.0\t2\t0 0 11\r\n",
      "64\t9\t0.0\t2\t0 0 9\r\n",
      "65\t10\t0.0\t2\t0 0 10\r\n",
      "66\t7\t0.0\t2\t0 0 7\r\n",
      "67\t5\t0.0\t2\t0 0 5\r\n",
      "68\t40\t0.0\t2\t0 0 40\r\n",
      "69\t20\t0.0\t2\t0 1 19\r\n",
      "70\t21\t0.0\t2\t0 3 18\r\n",
      "71\t3\t0.0\t2\t0 0 3\r\n",
      "72\t2\t0.0\t2\t0 0 2\r\n",
      "73\t17\t0.0\t2\t0 2 15\r\n",
      "74\t16\t0.0\t2\t0 0 16\r\n",
      "75\t4\t0.0\t2\t0 0 4\r\n",
      "76\t2\t0.0\t2\t1 0 1\r\n",
      "\r\n",
      "=== Second read: Adapter 2 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 16048 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 17.3%\r\n",
      "  C: 33.4%\r\n",
      "  G: 26.4%\r\n",
      "  T: 22.3%\r\n",
      "  none/other: 0.6%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t1005\t984.7\t0\t1005\r\n",
      "4\t1239\t246.2\t0\t608 631\r\n",
      "5\t1171\t61.5\t1\t394 777\r\n",
      "6\t778\t15.4\t1\t344 434\r\n",
      "7\t465\t3.8\t1\t377 88\r\n",
      "8\t483\t1.0\t1\t406 74 3\r\n",
      "9\t404\t0.2\t1\t374 6 24\r\n",
      "10\t383\t0.1\t2\t324 33 26\r\n",
      "11\t399\t0.0\t2\t372 4 23\r\n",
      "12\t401\t0.0\t2\t357 39 5\r\n",
      "13\t421\t0.0\t2\t409 4 8\r\n",
      "14\t410\t0.0\t2\t401 3 6\r\n",
      "15\t420\t0.0\t2\t410 4 6\r\n",
      "16\t380\t0.0\t2\t375 2 3\r\n",
      "17\t376\t0.0\t2\t367 1 8\r\n",
      "18\t365\t0.0\t2\t357 1 7\r\n",
      "19\t413\t0.0\t2\t397 3 13\r\n",
      "20\t390\t0.0\t2\t378 6 6\r\n",
      "21\t401\t0.0\t2\t391 4 6\r\n",
      "22\t371\t0.0\t2\t360 0 11\r\n",
      "23\t299\t0.0\t2\t224 71 4\r\n",
      "24\t366\t0.0\t2\t277 84 5\r\n",
      "25\t431\t0.0\t2\t329 95 7\r\n",
      "26\t377\t0.0\t2\t289 83 5\r\n",
      "27\t305\t0.0\t2\t234 67 4\r\n",
      "28\t270\t0.0\t2\t207 58 5\r\n",
      "29\t281\t0.0\t2\t219 57 5\r\n",
      "30\t310\t0.0\t2\t234 72 4\r\n",
      "31\t301\t0.0\t2\t219 70 12\r\n",
      "32\t270\t0.0\t2\t205 52 13\r\n",
      "33\t276\t0.0\t2\t213 55 8\r\n",
      "34\t226\t0.0\t2\t169 0 57\r\n",
      "35\t296\t0.0\t2\t286 3 7\r\n",
      "36\t307\t0.0\t2\t295 2 10\r\n",
      "37\t205\t0.0\t2\t195 2 8\r\n",
      "38\t139\t0.0\t2\t131 1 7\r\n",
      "39\t101\t0.0\t2\t92 3 6\r\n",
      "40\t86\t0.0\t2\t78 0 8\r\n",
      "41\t43\t0.0\t2\t32 1 10\r\n",
      "42\t25\t0.0\t2\t13 2 10\r\n",
      "43\t20\t0.0\t2\t10 0 10\r\n",
      "44\t9\t0.0\t2\t5 0 4\r\n",
      "45\t18\t0.0\t2\t7 0 11\r\n",
      "46\t33\t0.0\t2\t21 0 12\r\n",
      "47\t36\t0.0\t2\t29 0 7\r\n",
      "48\t8\t0.0\t2\t5 1 2\r\n",
      "49\t21\t0.0\t2\t9 1 11\r\n",
      "50\t9\t0.0\t2\t5 0 4\r\n",
      "51\t17\t0.0\t2\t1 0 16\r\n",
      "52\t3\t0.0\t2\t1 0 2\r\n",
      "53\t31\t0.0\t2\t0 1 30\r\n",
      "54\t15\t0.0\t2\t0 0 15\r\n",
      "55\t9\t0.0\t2\t0 1 8\r\n",
      "56\t20\t0.0\t2\t0 1 19\r\n",
      "57\t7\t0.0\t2\t0 1 6\r\n",
      "58\t19\t0.0\t2\t1 0 18\r\n",
      "59\t20\t0.0\t2\t0 0 20\r\n",
      "60\t8\t0.0\t2\t0 0 8\r\n",
      "61\t17\t0.0\t2\t0 0 17\r\n",
      "62\t11\t0.0\t2\t0 0 11\r\n",
      "63\t14\t0.0\t2\t0 0 14\r\n",
      "64\t9\t0.0\t2\t0 0 9\r\n",
      "65\t8\t0.0\t2\t0 0 8\r\n",
      "66\t7\t0.0\t2\t0 0 7\r\n",
      "67\t5\t0.0\t2\t0 0 5\r\n",
      "68\t47\t0.0\t2\t0 2 45\r\n",
      "69\t13\t0.0\t2\t0 3 10\r\n",
      "70\t12\t0.0\t2\t0 0 12\r\n",
      "71\t3\t0.0\t2\t0 0 3\r\n",
      "72\t2\t0.0\t2\t0 0 2\r\n",
      "74\t6\t0.0\t2\t0 0 6\r\n",
      "76\t2\t0.0\t2\t0 1 1\r\n",
      "\r\n",
      "cutadapt -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /tc2016/ubuntu/analysis/Kz-3h_S8_L001_R1_001.trimmed.fastq.gz -p /tc2016/ubuntu/analysis/Kz-3h_S8_L001_R2_001.trimmed.fastq.gz /tc2016/ubuntu/data/fastq/Kz-3h_S8_L001_R1_001.fastq.gz /tc2016/ubuntu/data/fastq/Kz-3h_S8_L001_R2_001.fastq.gz\r\n",
      "This is cutadapt 1.10 with Python 3.5.2\r\n",
      "Command line parameters: -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /tc2016/ubuntu/analysis/Kz-3h_S8_L001_R1_001.trimmed.fastq.gz -p /tc2016/ubuntu/analysis/Kz-3h_S8_L001_R2_001.trimmed.fastq.gz /tc2016/ubuntu/data/fastq/Kz-3h_S8_L001_R1_001.fastq.gz /tc2016/ubuntu/data/fastq/Kz-3h_S8_L001_R2_001.fastq.gz\r\n",
      "Trimming 2 adapters with at most 20.0% errors in paired-end mode ...\r\n",
      "Finished in 69.35 s (194 us/read; 0.31 M reads/minute).\r\n",
      "\r\n",
      "=== Summary ===\r\n",
      "\r\n",
      "Total read pairs processed:            357,262\r\n",
      "  Read 1 with adapter:                 113,602 (31.8%)\r\n",
      "  Read 2 with adapter:                 114,259 (32.0%)\r\n",
      "Pairs that were too short:                 289 (0.1%)\r\n",
      "Pairs written (passing filters):       356,973 (99.9%)\r\n",
      "\r\n",
      "Total basepairs processed:    54,303,824 bp\r\n",
      "  Read 1:    27,151,912 bp\r\n",
      "  Read 2:    27,151,912 bp\r\n",
      "Total written (filtered):     49,723,017 bp (91.6%)\r\n",
      "  Read 1:    24,863,741 bp\r\n",
      "  Read 2:    24,859,276 bp\r\n",
      "\r\n",
      "=== First read: Adapter 1 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 113602 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 14.8%\r\n",
      "  C: 36.5%\r\n",
      "  G: 24.2%\r\n",
      "  T: 24.4%\r\n",
      "  none/other: 0.1%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t5457\t5582.2\t0\t5457\r\n",
      "4\t5969\t1395.6\t0\t3193 2776\r\n",
      "5\t6407\t348.9\t1\t2348 4059\r\n",
      "6\t4027\t87.2\t1\t2127 1900\r\n",
      "7\t2715\t21.8\t1\t2304 411\r\n",
      "8\t2976\t5.5\t1\t2404 536 36\r\n",
      "9\t2701\t1.4\t1\t2408 68 225\r\n",
      "10\t2458\t0.3\t2\t2130 205 123\r\n",
      "11\t2576\t0.1\t2\t2448 44 84\r\n",
      "12\t2425\t0.0\t2\t2125 266 34\r\n",
      "13\t2567\t0.0\t2\t2489 42 36\r\n",
      "14\t2594\t0.0\t2\t2528 40 26\r\n",
      "15\t2663\t0.0\t2\t2605 27 31\r\n",
      "16\t2551\t0.0\t2\t2469 36 46\r\n",
      "17\t2626\t0.0\t2\t2543 33 50\r\n",
      "18\t2716\t0.0\t2\t2654 26 36\r\n",
      "19\t2786\t0.0\t2\t2705 29 52\r\n",
      "20\t2871\t0.0\t2\t2798 37 36\r\n",
      "21\t2883\t0.0\t2\t2814 32 37\r\n",
      "22\t2764\t0.0\t2\t2706 26 32\r\n",
      "23\t2691\t0.0\t2\t2606 25 60\r\n",
      "24\t2977\t0.0\t2\t2911 26 40\r\n",
      "25\t3202\t0.0\t2\t3146 26 30\r\n",
      "26\t3207\t0.0\t2\t3135 31 41\r\n",
      "27\t2749\t0.0\t2\t2683 29 37\r\n",
      "28\t2652\t0.0\t2\t2589 21 42\r\n",
      "29\t2813\t0.0\t2\t2743 32 38\r\n",
      "30\t2947\t0.0\t2\t2882 36 29\r\n",
      "31\t2945\t0.0\t2\t2877 35 33\r\n",
      "32\t2776\t0.0\t2\t2717 31 28\r\n",
      "33\t2555\t0.0\t2\t2493 24 38\r\n",
      "34\t2425\t0.0\t2\t2363 18 44\r\n",
      "35\t3104\t0.0\t2\t3053 28 23\r\n",
      "36\t3069\t0.0\t2\t3015 22 32\r\n",
      "37\t2017\t0.0\t2\t1966 24 27\r\n",
      "38\t1497\t0.0\t2\t1460 14 23\r\n",
      "39\t1135\t0.0\t2\t1075 16 44\r\n",
      "40\t807\t0.0\t2\t767 13 27\r\n",
      "41\t388\t0.0\t2\t351 5 32\r\n",
      "42\t260\t0.0\t2\t175 4 81\r\n",
      "43\t167\t0.0\t2\t118 2 47\r\n",
      "44\t131\t0.0\t2\t93 0 38\r\n",
      "45\t168\t0.0\t2\t139 5 24\r\n",
      "46\t383\t0.0\t2\t351 4 28\r\n",
      "47\t269\t0.0\t2\t232 5 32\r\n",
      "48\t140\t0.0\t2\t115 3 22\r\n",
      "49\t137\t0.0\t2\t93 1 43\r\n",
      "50\t91\t0.0\t2\t59 4 28\r\n",
      "51\t78\t0.0\t2\t9 0 69\r\n",
      "52\t33\t0.0\t2\t0 0 33\r\n",
      "53\t78\t0.0\t2\t1 3 74\r\n",
      "54\t24\t0.0\t2\t0 0 24\r\n",
      "55\t41\t0.0\t2\t0 2 39\r\n",
      "56\t45\t0.0\t2\t1 1 43\r\n",
      "57\t26\t0.0\t2\t0 2 24\r\n",
      "58\t29\t0.0\t2\t0 1 28\r\n",
      "59\t32\t0.0\t2\t0 1 31\r\n",
      "60\t32\t0.0\t2\t1 0 31\r\n",
      "61\t40\t0.0\t2\t0 2 38\r\n",
      "62\t12\t0.0\t2\t0 0 12\r\n",
      "63\t81\t0.0\t2\t0 0 81\r\n",
      "64\t31\t0.0\t2\t0 0 31\r\n",
      "65\t38\t0.0\t2\t0 2 36\r\n",
      "66\t46\t0.0\t2\t0 1 45\r\n",
      "67\t36\t0.0\t2\t0 5 31\r\n",
      "68\t161\t0.0\t2\t0 8 153\r\n",
      "69\t37\t0.0\t2\t0 2 35\r\n",
      "70\t64\t0.0\t2\t0 0 64\r\n",
      "71\t15\t0.0\t2\t0 0 15\r\n",
      "72\t21\t0.0\t2\t0 1 20\r\n",
      "73\t59\t0.0\t2\t0 3 56\r\n",
      "74\t50\t0.0\t2\t0 1 49\r\n",
      "75\t23\t0.0\t2\t0 0 23\r\n",
      "76\t36\t0.0\t2\t1 1 34\r\n",
      "\r\n",
      "=== Second read: Adapter 2 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 114259 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 15.2%\r\n",
      "  C: 36.0%\r\n",
      "  G: 24.6%\r\n",
      "  T: 23.3%\r\n",
      "  none/other: 0.8%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t5400\t5582.2\t0\t5400\r\n",
      "4\t6355\t1395.6\t0\t3272 3083\r\n",
      "5\t6814\t348.9\t1\t2373 4441\r\n",
      "6\t4124\t87.2\t1\t1920 2204\r\n",
      "7\t2772\t21.8\t1\t2285 487\r\n",
      "8\t2929\t5.5\t1\t2335 550 44\r\n",
      "9\t2661\t1.4\t1\t2374 99 188\r\n",
      "10\t2495\t0.3\t2\t2025 292 178\r\n",
      "11\t2598\t0.1\t2\t2407 73 118\r\n",
      "12\t2423\t0.0\t2\t2039 339 45\r\n",
      "13\t2562\t0.0\t2\t2477 52 33\r\n",
      "14\t2598\t0.0\t2\t2284 264 50\r\n",
      "15\t2672\t0.0\t2\t2369 250 53\r\n",
      "16\t2538\t0.0\t2\t2268 221 49\r\n",
      "17\t2625\t0.0\t2\t2331 224 70\r\n",
      "18\t2717\t0.0\t2\t2418 233 66\r\n",
      "19\t2782\t0.0\t2\t2461 261 60\r\n",
      "20\t2860\t0.0\t2\t2549 243 68\r\n",
      "21\t2888\t0.0\t2\t2612 228 48\r\n",
      "22\t2779\t0.0\t2\t2468 255 56\r\n",
      "23\t2659\t0.0\t2\t1732 865 62\r\n",
      "24\t2956\t0.0\t2\t1937 944 75\r\n",
      "25\t3185\t0.0\t2\t2085 800 300\r\n",
      "26\t3188\t0.0\t2\t2346 789 53\r\n",
      "27\t2740\t0.0\t2\t1970 675 95\r\n",
      "28\t2630\t0.0\t2\t1873 710 47\r\n",
      "29\t2796\t0.0\t2\t2042 677 77\r\n",
      "30\t2930\t0.0\t2\t2145 741 44\r\n",
      "31\t2922\t0.0\t2\t2111 731 80\r\n",
      "32\t2763\t0.0\t2\t1979 735 49\r\n",
      "33\t2538\t0.0\t2\t1833 633 72\r\n",
      "34\t2406\t0.0\t2\t1692 52 662\r\n",
      "35\t3095\t0.0\t2\t2959 87 49\r\n",
      "36\t3065\t0.0\t2\t2935 79 51\r\n",
      "37\t2012\t0.0\t2\t1923 52 37\r\n",
      "38\t1500\t0.0\t2\t1428 37 35\r\n",
      "39\t1144\t0.0\t2\t1058 24 62\r\n",
      "40\t822\t0.0\t2\t758 18 46\r\n",
      "41\t390\t0.0\t2\t345 6 39\r\n",
      "42\t232\t0.0\t2\t175 2 55\r\n",
      "43\t164\t0.0\t2\t116 4 44\r\n",
      "44\t126\t0.0\t2\t90 3 33\r\n",
      "45\t165\t0.0\t2\t135 1 29\r\n",
      "46\t387\t0.0\t2\t345 9 33\r\n",
      "47\t277\t0.0\t2\t228 6 43\r\n",
      "48\t131\t0.0\t2\t111 6 14\r\n",
      "49\t142\t0.0\t2\t90 1 51\r\n",
      "50\t86\t0.0\t2\t58 2 26\r\n",
      "51\t75\t0.0\t2\t9 1 65\r\n",
      "52\t30\t0.0\t2\t0 2 28\r\n",
      "53\t111\t0.0\t2\t2 5 104\r\n",
      "54\t44\t0.0\t2\t0 2 42\r\n",
      "55\t33\t0.0\t2\t1 0 32\r\n",
      "56\t74\t0.0\t2\t2 1 71\r\n",
      "57\t41\t0.0\t2\t2 0 39\r\n",
      "58\t46\t0.0\t2\t1 0 45\r\n",
      "59\t39\t0.0\t2\t0 2 37\r\n",
      "60\t34\t0.0\t2\t1 0 33\r\n",
      "61\t46\t0.0\t2\t0 1 45\r\n",
      "62\t29\t0.0\t2\t0 0 29\r\n",
      "63\t93\t0.0\t2\t0 2 91\r\n",
      "64\t54\t0.0\t2\t0 0 54\r\n",
      "65\t42\t0.0\t2\t0 1 41\r\n",
      "66\t40\t0.0\t2\t0 1 39\r\n",
      "67\t30\t0.0\t2\t0 2 28\r\n",
      "68\t154\t0.0\t2\t0 4 150\r\n",
      "69\t29\t0.0\t2\t0 2 27\r\n",
      "70\t55\t0.0\t2\t0 2 53\r\n",
      "71\t16\t0.0\t2\t0 0 16\r\n",
      "72\t14\t0.0\t2\t0 0 14\r\n",
      "73\t8\t0.0\t2\t0 0 8\r\n",
      "74\t58\t0.0\t2\t0 0 58\r\n",
      "75\t10\t0.0\t2\t0 0 10\r\n",
      "76\t11\t0.0\t2\t0 2 9\r\n",
      "\r\n",
      "cutadapt -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /tc2016/ubuntu/analysis/Mz-1h_S3_L001_R1_001.trimmed.fastq.gz -p /tc2016/ubuntu/analysis/Mz-1h_S3_L001_R2_001.trimmed.fastq.gz /tc2016/ubuntu/data/fastq/Mz-1h_S3_L001_R1_001.fastq.gz /tc2016/ubuntu/data/fastq/Mz-1h_S3_L001_R2_001.fastq.gz\r\n",
      "This is cutadapt 1.10 with Python 3.5.2\r\n",
      "Command line parameters: -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /tc2016/ubuntu/analysis/Mz-1h_S3_L001_R1_001.trimmed.fastq.gz -p /tc2016/ubuntu/analysis/Mz-1h_S3_L001_R2_001.trimmed.fastq.gz /tc2016/ubuntu/data/fastq/Mz-1h_S3_L001_R1_001.fastq.gz /tc2016/ubuntu/data/fastq/Mz-1h_S3_L001_R2_001.fastq.gz\r\n",
      "Trimming 2 adapters with at most 20.0% errors in paired-end mode ...\r\n",
      "Finished in 44.06 s (210 us/read; 0.29 M reads/minute).\r\n",
      "\r\n",
      "=== Summary ===\r\n",
      "\r\n",
      "Total read pairs processed:            209,953\r\n",
      "  Read 1 with adapter:                  60,982 (29.0%)\r\n",
      "  Read 2 with adapter:                  61,238 (29.2%)\r\n",
      "Pairs that were too short:                 195 (0.1%)\r\n",
      "Pairs written (passing filters):       209,758 (99.9%)\r\n",
      "\r\n",
      "Total basepairs processed:    31,912,856 bp\r\n",
      "  Read 1:    15,956,428 bp\r\n",
      "  Read 2:    15,956,428 bp\r\n",
      "Total written (filtered):     29,532,721 bp (92.5%)\r\n",
      "  Read 1:    14,770,115 bp\r\n",
      "  Read 2:    14,762,606 bp\r\n",
      "\r\n",
      "=== First read: Adapter 1 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 60982 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 14.7%\r\n",
      "  C: 36.8%\r\n",
      "  G: 24.2%\r\n",
      "  T: 24.4%\r\n",
      "  none/other: 0.1%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t3167\t3280.5\t0\t3167\r\n",
      "4\t3588\t820.1\t0\t1835 1753\r\n",
      "5\t3839\t205.0\t1\t1375 2464\r\n",
      "6\t2283\t51.3\t1\t1168 1115\r\n",
      "7\t1610\t12.8\t1\t1371 239\r\n",
      "8\t1565\t3.2\t1\t1286 265 14\r\n",
      "9\t1547\t0.8\t1\t1389 44 114\r\n",
      "10\t1454\t0.2\t2\t1232 139 83\r\n",
      "11\t1423\t0.1\t2\t1338 30 55\r\n",
      "12\t1396\t0.0\t2\t1209 165 22\r\n",
      "13\t1356\t0.0\t2\t1301 32 23\r\n",
      "14\t1383\t0.0\t2\t1345 30 8\r\n",
      "15\t1432\t0.0\t2\t1382 27 23\r\n",
      "16\t1379\t0.0\t2\t1320 35 24\r\n",
      "17\t1427\t0.0\t2\t1378 27 22\r\n",
      "18\t1514\t0.0\t2\t1456 24 34\r\n",
      "19\t1587\t0.0\t2\t1530 27 30\r\n",
      "20\t1564\t0.0\t2\t1501 34 29\r\n",
      "21\t1551\t0.0\t2\t1486 37 28\r\n",
      "22\t1418\t0.0\t2\t1360 34 24\r\n",
      "23\t1327\t0.0\t2\t1289 13 25\r\n",
      "24\t1516\t0.0\t2\t1465 25 26\r\n",
      "25\t1688\t0.0\t2\t1632 28 28\r\n",
      "26\t1603\t0.0\t2\t1546 22 35\r\n",
      "27\t1450\t0.0\t2\t1404 23 23\r\n",
      "28\t1289\t0.0\t2\t1241 27 21\r\n",
      "29\t1414\t0.0\t2\t1356 26 32\r\n",
      "30\t1525\t0.0\t2\t1473 27 25\r\n",
      "31\t1437\t0.0\t2\t1393 22 22\r\n",
      "32\t1360\t0.0\t2\t1306 24 30\r\n",
      "33\t1307\t0.0\t2\t1267 18 22\r\n",
      "34\t1201\t0.0\t2\t1140 23 38\r\n",
      "35\t1355\t0.0\t2\t1314 23 18\r\n",
      "36\t1410\t0.0\t2\t1368 25 17\r\n",
      "37\t950\t0.0\t2\t915 19 16\r\n",
      "38\t792\t0.0\t2\t754 14 24\r\n",
      "39\t625\t0.0\t2\t582 13 30\r\n",
      "40\t467\t0.0\t2\t429 7 31\r\n",
      "41\t251\t0.0\t2\t223 3 25\r\n",
      "42\t159\t0.0\t2\t106 5 48\r\n",
      "43\t92\t0.0\t2\t59 1 32\r\n",
      "44\t71\t0.0\t2\t46 1 24\r\n",
      "45\t65\t0.0\t2\t45 1 19\r\n",
      "46\t150\t0.0\t2\t120 5 25\r\n",
      "47\t140\t0.0\t2\t108 1 31\r\n",
      "48\t81\t0.0\t2\t58 3 20\r\n",
      "49\t70\t0.0\t2\t41 1 28\r\n",
      "50\t54\t0.0\t2\t33 0 21\r\n",
      "51\t51\t0.0\t2\t2 1 48\r\n",
      "52\t15\t0.0\t2\t2 0 13\r\n",
      "53\t31\t0.0\t2\t0 0 31\r\n",
      "54\t10\t0.0\t2\t0 0 10\r\n",
      "55\t19\t0.0\t2\t0 0 19\r\n",
      "56\t16\t0.0\t2\t0 1 15\r\n",
      "57\t12\t0.0\t2\t1 1 10\r\n",
      "58\t17\t0.0\t2\t1 1 15\r\n",
      "59\t11\t0.0\t2\t0 2 9\r\n",
      "60\t12\t0.0\t2\t0 0 12\r\n",
      "61\t10\t0.0\t2\t0 0 10\r\n",
      "62\t8\t0.0\t2\t0 0 8\r\n",
      "63\t23\t0.0\t2\t0 0 23\r\n",
      "64\t16\t0.0\t2\t0 0 16\r\n",
      "65\t27\t0.0\t2\t0 4 23\r\n",
      "66\t31\t0.0\t2\t0 0 31\r\n",
      "67\t23\t0.0\t2\t0 1 22\r\n",
      "68\t93\t0.0\t2\t0 4 89\r\n",
      "69\t21\t0.0\t2\t0 0 21\r\n",
      "70\t49\t0.0\t2\t0 1 48\r\n",
      "71\t7\t0.0\t2\t0 0 7\r\n",
      "72\t10\t0.0\t2\t0 0 10\r\n",
      "73\t53\t0.0\t2\t0 3 50\r\n",
      "74\t47\t0.0\t2\t0 0 47\r\n",
      "75\t17\t0.0\t2\t0 1 16\r\n",
      "76\t21\t0.0\t2\t0 1 20\r\n",
      "\r\n",
      "=== Second read: Adapter 2 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 61238 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 15.9%\r\n",
      "  C: 35.2%\r\n",
      "  G: 24.8%\r\n",
      "  T: 23.3%\r\n",
      "  none/other: 0.8%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t3174\t3280.5\t0\t3174\r\n",
      "4\t3685\t820.1\t0\t1872 1813\r\n",
      "5\t3936\t205.0\t1\t1350 2586\r\n",
      "6\t2367\t51.3\t1\t1134 1233\r\n",
      "7\t1653\t12.8\t1\t1372 281\r\n",
      "8\t1513\t3.2\t1\t1293 204 16\r\n",
      "9\t1552\t0.8\t1\t1393 41 118\r\n",
      "10\t1453\t0.2\t2\t1197 178 78\r\n",
      "11\t1438\t0.1\t2\t1355 18 65\r\n",
      "12\t1399\t0.0\t2\t1175 199 25\r\n",
      "13\t1358\t0.0\t2\t1302 25 31\r\n",
      "14\t1391\t0.0\t2\t1354 14 23\r\n",
      "15\t1424\t0.0\t2\t1384 21 19\r\n",
      "16\t1381\t0.0\t2\t1336 21 24\r\n",
      "17\t1438\t0.0\t2\t1383 19 36\r\n",
      "18\t1511\t0.0\t2\t1456 27 28\r\n",
      "19\t1595\t0.0\t2\t1536 21 38\r\n",
      "20\t1562\t0.0\t2\t1513 18 31\r\n",
      "21\t1548\t0.0\t2\t1497 21 30\r\n",
      "22\t1426\t0.0\t2\t1374 25 27\r\n",
      "23\t1311\t0.0\t2\t859 432 20\r\n",
      "24\t1512\t0.0\t2\t1000 477 35\r\n",
      "25\t1669\t0.0\t2\t1074 558 37\r\n",
      "26\t1593\t0.0\t2\t1049 517 27\r\n",
      "27\t1437\t0.0\t2\t915 481 41\r\n",
      "28\t1287\t0.0\t2\t818 439 30\r\n",
      "29\t1401\t0.0\t2\t905 452 44\r\n",
      "30\t1519\t0.0\t2\t985 503 31\r\n",
      "31\t1432\t0.0\t2\t934 455 43\r\n",
      "32\t1351\t0.0\t2\t881 434 36\r\n",
      "33\t1313\t0.0\t2\t841 421 51\r\n",
      "34\t1192\t0.0\t2\t765 14 413\r\n",
      "35\t1356\t0.0\t2\t1301 24 31\r\n",
      "36\t1413\t0.0\t2\t1366 21 26\r\n",
      "37\t956\t0.0\t2\t918 18 20\r\n",
      "38\t788\t0.0\t2\t747 16 25\r\n",
      "39\t620\t0.0\t2\t577 17 26\r\n",
      "40\t464\t0.0\t2\t424 13 27\r\n",
      "41\t255\t0.0\t2\t225 0 30\r\n",
      "42\t155\t0.0\t2\t107 5 43\r\n",
      "43\t80\t0.0\t2\t59 2 19\r\n",
      "44\t76\t0.0\t2\t44 3 29\r\n",
      "45\t66\t0.0\t2\t42 5 19\r\n",
      "46\t142\t0.0\t2\t119 4 19\r\n",
      "47\t131\t0.0\t2\t106 4 21\r\n",
      "48\t71\t0.0\t2\t56 4 11\r\n",
      "49\t78\t0.0\t2\t40 4 34\r\n",
      "50\t47\t0.0\t2\t33 0 14\r\n",
      "51\t35\t0.0\t2\t2 0 33\r\n",
      "52\t26\t0.0\t2\t2 0 24\r\n",
      "53\t46\t0.0\t2\t0 2 44\r\n",
      "54\t25\t0.0\t2\t0 1 24\r\n",
      "55\t30\t0.0\t2\t0 1 29\r\n",
      "56\t30\t0.0\t2\t1 1 28\r\n",
      "57\t27\t0.0\t2\t2 1 24\r\n",
      "58\t33\t0.0\t2\t2 2 29\r\n",
      "59\t33\t0.0\t2\t1 1 31\r\n",
      "60\t21\t0.0\t2\t0 0 21\r\n",
      "61\t20\t0.0\t2\t0 0 20\r\n",
      "62\t26\t0.0\t2\t0 1 25\r\n",
      "63\t50\t0.0\t2\t0 0 50\r\n",
      "64\t43\t0.0\t2\t0 1 42\r\n",
      "65\t26\t0.0\t2\t0 2 24\r\n",
      "66\t30\t0.0\t2\t0 1 29\r\n",
      "67\t18\t0.0\t2\t0 1 17\r\n",
      "68\t86\t0.0\t2\t0 0 86\r\n",
      "69\t27\t0.0\t2\t0 2 25\r\n",
      "70\t30\t0.0\t2\t0 1 29\r\n",
      "71\t10\t0.0\t2\t0 0 10\r\n",
      "72\t5\t0.0\t2\t0 0 5\r\n",
      "73\t7\t0.0\t2\t0 0 7\r\n",
      "74\t29\t0.0\t2\t0 0 29\r\n",
      "75\t4\t0.0\t2\t0 0 4\r\n",
      "76\t2\t0.0\t2\t0 0 2\r\n",
      "\r\n",
      "cutadapt -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /tc2016/ubuntu/analysis/Mz-3h_S10_L001_R1_001.trimmed.fastq.gz -p /tc2016/ubuntu/analysis/Mz-3h_S10_L001_R2_001.trimmed.fastq.gz /tc2016/ubuntu/data/fastq/Mz-3h_S10_L001_R1_001.fastq.gz /tc2016/ubuntu/data/fastq/Mz-3h_S10_L001_R2_001.fastq.gz\r\n",
      "This is cutadapt 1.10 with Python 3.5.2\r\n",
      "Command line parameters: -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /tc2016/ubuntu/analysis/Mz-3h_S10_L001_R1_001.trimmed.fastq.gz -p /tc2016/ubuntu/analysis/Mz-3h_S10_L001_R2_001.trimmed.fastq.gz /tc2016/ubuntu/data/fastq/Mz-3h_S10_L001_R1_001.fastq.gz /tc2016/ubuntu/data/fastq/Mz-3h_S10_L001_R2_001.fastq.gz\r\n",
      "Trimming 2 adapters with at most 20.0% errors in paired-end mode ...\r\n",
      "Finished in 63.23 s (194 us/read; 0.31 M reads/minute).\r\n",
      "\r\n",
      "=== Summary ===\r\n",
      "\r\n",
      "Total read pairs processed:            325,123\r\n",
      "  Read 1 with adapter:                 102,194 (31.4%)\r\n",
      "  Read 2 with adapter:                 102,273 (31.5%)\r\n",
      "Pairs that were too short:                 277 (0.1%)\r\n",
      "Pairs written (passing filters):       324,846 (99.9%)\r\n",
      "\r\n",
      "Total basepairs processed:    49,418,696 bp\r\n",
      "  Read 1:    24,709,348 bp\r\n",
      "  Read 2:    24,709,348 bp\r\n",
      "Total written (filtered):     45,311,358 bp (91.7%)\r\n",
      "  Read 1:    22,657,711 bp\r\n",
      "  Read 2:    22,653,647 bp\r\n",
      "\r\n",
      "=== First read: Adapter 1 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 102194 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 14.7%\r\n",
      "  C: 36.8%\r\n",
      "  G: 24.2%\r\n",
      "  T: 24.2%\r\n",
      "  none/other: 0.0%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t4891\t5080.0\t0\t4891\r\n",
      "4\t5410\t1270.0\t0\t2847 2563\r\n",
      "5\t5838\t317.5\t1\t2157 3681\r\n",
      "6\t3705\t79.4\t1\t1906 1799\r\n",
      "7\t2467\t19.8\t1\t2058 409\r\n",
      "8\t2457\t5.0\t1\t2039 374 44\r\n",
      "9\t2434\t1.2\t1\t2193 47 194\r\n",
      "10\t2264\t0.3\t2\t1959 194 111\r\n",
      "11\t2358\t0.1\t2\t2228 29 101\r\n",
      "12\t2219\t0.0\t2\t1950 224 45\r\n",
      "13\t2199\t0.0\t2\t2134 28 37\r\n",
      "14\t2390\t0.0\t2\t2332 31 27\r\n",
      "15\t2434\t0.0\t2\t2379 26 29\r\n",
      "16\t2302\t0.0\t2\t2245 26 31\r\n",
      "17\t2325\t0.0\t2\t2265 21 39\r\n",
      "18\t2457\t0.0\t2\t2396 25 36\r\n",
      "19\t2603\t0.0\t2\t2522 29 52\r\n",
      "20\t2605\t0.0\t2\t2546 32 27\r\n",
      "21\t2577\t0.0\t2\t2515 28 34\r\n",
      "22\t2481\t0.0\t2\t2422 20 39\r\n",
      "23\t2405\t0.0\t2\t2320 36 49\r\n",
      "24\t2621\t0.0\t2\t2571 29 21\r\n",
      "25\t2996\t0.0\t2\t2927 32 37\r\n",
      "26\t2824\t0.0\t2\t2766 22 36\r\n",
      "27\t2511\t0.0\t2\t2469 21 21\r\n",
      "28\t2484\t0.0\t2\t2426 13 45\r\n",
      "29\t2587\t0.0\t2\t2516 26 45\r\n",
      "30\t2724\t0.0\t2\t2673 23 28\r\n",
      "31\t2711\t0.0\t2\t2640 27 44\r\n",
      "32\t2526\t0.0\t2\t2473 22 31\r\n",
      "33\t2350\t0.0\t2\t2287 22 41\r\n",
      "34\t2078\t0.0\t2\t2017 15 46\r\n",
      "35\t2528\t0.0\t2\t2466 28 34\r\n",
      "36\t2562\t0.0\t2\t2494 27 41\r\n",
      "37\t1743\t0.0\t2\t1697 16 30\r\n",
      "38\t1416\t0.0\t2\t1372 20 24\r\n",
      "39\t1031\t0.0\t2\t970 11 50\r\n",
      "40\t715\t0.0\t2\t660 11 44\r\n",
      "41\t369\t0.0\t2\t323 5 41\r\n",
      "42\t226\t0.0\t2\t169 2 55\r\n",
      "43\t152\t0.0\t2\t100 3 49\r\n",
      "44\t117\t0.0\t2\t91 4 22\r\n",
      "45\t166\t0.0\t2\t125 2 39\r\n",
      "46\t311\t0.0\t2\t265 3 43\r\n",
      "47\t215\t0.0\t2\t174 3 38\r\n",
      "48\t137\t0.0\t2\t103 2 32\r\n",
      "49\t122\t0.0\t2\t81 3 38\r\n",
      "50\t81\t0.0\t2\t57 1 23\r\n",
      "51\t82\t0.0\t2\t6 1 75\r\n",
      "52\t30\t0.0\t2\t2 1 27\r\n",
      "53\t62\t0.0\t2\t2 2 58\r\n",
      "54\t25\t0.0\t2\t2 1 22\r\n",
      "55\t20\t0.0\t2\t0 1 19\r\n",
      "56\t39\t0.0\t2\t0 4 35\r\n",
      "57\t36\t0.0\t2\t5 4 27\r\n",
      "58\t18\t0.0\t2\t0 0 18\r\n",
      "59\t28\t0.0\t2\t0 3 25\r\n",
      "60\t22\t0.0\t2\t0 1 21\r\n",
      "61\t30\t0.0\t2\t0 0 30\r\n",
      "62\t23\t0.0\t2\t0 0 23\r\n",
      "63\t57\t0.0\t2\t0 1 56\r\n",
      "64\t29\t0.0\t2\t0 2 27\r\n",
      "65\t45\t0.0\t2\t0 0 45\r\n",
      "66\t38\t0.0\t2\t0 0 38\r\n",
      "67\t32\t0.0\t2\t0 1 31\r\n",
      "68\t138\t0.0\t2\t0 8 130\r\n",
      "69\t41\t0.0\t2\t0 3 38\r\n",
      "70\t59\t0.0\t2\t0 1 58\r\n",
      "71\t11\t0.0\t2\t0 0 11\r\n",
      "72\t21\t0.0\t2\t0 1 20\r\n",
      "73\t68\t0.0\t2\t0 2 66\r\n",
      "74\t64\t0.0\t2\t0 1 63\r\n",
      "75\t28\t0.0\t2\t0 1 27\r\n",
      "76\t24\t0.0\t2\t0 2 22\r\n",
      "\r\n",
      "=== Second read: Adapter 2 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 102273 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 15.2%\r\n",
      "  C: 36.2%\r\n",
      "  G: 24.5%\r\n",
      "  T: 23.3%\r\n",
      "  none/other: 0.8%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t4734\t5080.0\t0\t4734\r\n",
      "4\t5735\t1270.0\t0\t2929 2806\r\n",
      "5\t6071\t317.5\t1\t2130 3941\r\n",
      "6\t3638\t79.4\t1\t1658 1980\r\n",
      "7\t2454\t19.8\t1\t2024 430\r\n",
      "8\t2407\t5.0\t1\t1967 406 34\r\n",
      "9\t2398\t1.2\t1\t2147 83 168\r\n",
      "10\t2288\t0.3\t2\t1815 322 151\r\n",
      "11\t2364\t0.1\t2\t2179 65 120\r\n",
      "12\t2201\t0.0\t2\t1839 312 50\r\n",
      "13\t2193\t0.0\t2\t2097 65 31\r\n",
      "14\t2386\t0.0\t2\t2131 206 49\r\n",
      "15\t2430\t0.0\t2\t2166 205 59\r\n",
      "16\t2305\t0.0\t2\t2053 190 62\r\n",
      "17\t2311\t0.0\t2\t2033 225 53\r\n",
      "18\t2443\t0.0\t2\t2160 222 61\r\n",
      "19\t2606\t0.0\t2\t2289 237 80\r\n",
      "20\t2611\t0.0\t2\t2274 259 78\r\n",
      "21\t2594\t0.0\t2\t2286 232 76\r\n",
      "22\t2489\t0.0\t2\t2196 230 63\r\n",
      "23\t2388\t0.0\t2\t1521 791 76\r\n",
      "24\t2636\t0.0\t2\t1652 907 77\r\n",
      "25\t2974\t0.0\t2\t1985 709 280\r\n",
      "26\t2810\t0.0\t2\t2043 707 60\r\n",
      "27\t2505\t0.0\t2\t1810 628 67\r\n",
      "28\t2470\t0.0\t2\t1772 638 60\r\n",
      "29\t2569\t0.0\t2\t1804 665 100\r\n",
      "30\t2723\t0.0\t2\t1966 693 64\r\n",
      "31\t2689\t0.0\t2\t1929 699 61\r\n",
      "32\t2505\t0.0\t2\t1803 664 38\r\n",
      "33\t2338\t0.0\t2\t1653 591 94\r\n",
      "34\t2068\t0.0\t2\t1473 34 561\r\n",
      "35\t2525\t0.0\t2\t2397 70 58\r\n",
      "36\t2546\t0.0\t2\t2415 80 51\r\n",
      "37\t1740\t0.0\t2\t1632 60 48\r\n",
      "38\t1413\t0.0\t2\t1327 42 44\r\n",
      "39\t1017\t0.0\t2\t947 21 49\r\n",
      "40\t703\t0.0\t2\t640 22 41\r\n",
      "41\t357\t0.0\t2\t313 14 30\r\n",
      "42\t258\t0.0\t2\t166 11 81\r\n",
      "43\t153\t0.0\t2\t98 5 50\r\n",
      "44\t119\t0.0\t2\t94 1 24\r\n",
      "45\t153\t0.0\t2\t119 7 27\r\n",
      "46\t305\t0.0\t2\t252 10 43\r\n",
      "47\t212\t0.0\t2\t164 9 39\r\n",
      "48\t142\t0.0\t2\t100 8 34\r\n",
      "49\t134\t0.0\t2\t78 4 52\r\n",
      "50\t80\t0.0\t2\t55 3 22\r\n",
      "51\t64\t0.0\t2\t5 1 58\r\n",
      "52\t20\t0.0\t2\t2 1 17\r\n",
      "53\t75\t0.0\t2\t3 2 70\r\n",
      "54\t37\t0.0\t2\t3 0 34\r\n",
      "55\t29\t0.0\t2\t1 1 27\r\n",
      "56\t73\t0.0\t2\t2 1 70\r\n",
      "57\t59\t0.0\t2\t8 1 50\r\n",
      "58\t37\t0.0\t2\t0 0 37\r\n",
      "59\t49\t0.0\t2\t1 1 47\r\n",
      "60\t33\t0.0\t2\t0 0 33\r\n",
      "61\t56\t0.0\t2\t0 0 56\r\n",
      "62\t34\t0.0\t2\t0 2 32\r\n",
      "63\t58\t0.0\t2\t0 1 57\r\n",
      "64\t47\t0.0\t2\t0 0 47\r\n",
      "65\t29\t0.0\t2\t0 0 29\r\n",
      "66\t31\t0.0\t2\t0 0 31\r\n",
      "67\t30\t0.0\t2\t0 2 28\r\n",
      "68\t134\t0.0\t2\t0 3 131\r\n",
      "69\t43\t0.0\t2\t0 3 40\r\n",
      "70\t62\t0.0\t2\t0 1 61\r\n",
      "71\t11\t0.0\t2\t0 0 11\r\n",
      "72\t9\t0.0\t2\t0 0 9\r\n",
      "73\t11\t0.0\t2\t0 0 11\r\n",
      "74\t44\t0.0\t2\t0 0 44\r\n",
      "75\t3\t0.0\t2\t0 1 2\r\n",
      "76\t5\t0.0\t2\t0 0 5\r\n",
      "\r\n",
      "cutadapt -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /tc2016/ubuntu/analysis/WT-1h_S7_L001_R1_001.trimmed.fastq.gz -p /tc2016/ubuntu/analysis/WT-1h_S7_L001_R2_001.trimmed.fastq.gz /tc2016/ubuntu/data/fastq/WT-1h_S7_L001_R1_001.fastq.gz /tc2016/ubuntu/data/fastq/WT-1h_S7_L001_R2_001.fastq.gz\r\n",
      "This is cutadapt 1.10 with Python 3.5.2\r\n",
      "Command line parameters: -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /tc2016/ubuntu/analysis/WT-1h_S7_L001_R1_001.trimmed.fastq.gz -p /tc2016/ubuntu/analysis/WT-1h_S7_L001_R2_001.trimmed.fastq.gz /tc2016/ubuntu/data/fastq/WT-1h_S7_L001_R1_001.fastq.gz /tc2016/ubuntu/data/fastq/WT-1h_S7_L001_R2_001.fastq.gz\r\n",
      "Trimming 2 adapters with at most 20.0% errors in paired-end mode ...\r\n",
      "Finished in 17.68 s (199 us/read; 0.30 M reads/minute).\r\n",
      "\r\n",
      "=== Summary ===\r\n",
      "\r\n",
      "Total read pairs processed:             88,835\r\n",
      "  Read 1 with adapter:                  20,488 (23.1%)\r\n",
      "  Read 2 with adapter:                  20,916 (23.5%)\r\n",
      "Pairs that were too short:                  89 (0.1%)\r\n",
      "Pairs written (passing filters):        88,746 (99.9%)\r\n",
      "\r\n",
      "Total basepairs processed:    13,502,920 bp\r\n",
      "  Read 1:     6,751,460 bp\r\n",
      "  Read 2:     6,751,460 bp\r\n",
      "Total written (filtered):     12,770,694 bp (94.6%)\r\n",
      "  Read 1:     6,391,645 bp\r\n",
      "  Read 2:     6,379,049 bp\r\n",
      "\r\n",
      "=== First read: Adapter 1 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 20488 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 16.3%\r\n",
      "  C: 33.4%\r\n",
      "  G: 24.4%\r\n",
      "  T: 25.7%\r\n",
      "  none/other: 0.1%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t1354\t1388.0\t0\t1354\r\n",
      "4\t1479\t347.0\t0\t670 809\r\n",
      "5\t1639\t86.8\t1\t509 1130\r\n",
      "6\t1003\t21.7\t1\t437 566\r\n",
      "7\t583\t5.4\t1\t454 129\r\n",
      "8\t581\t1.4\t1\t448 122 11\r\n",
      "9\t583\t0.3\t1\t503 30 50\r\n",
      "10\t525\t0.1\t2\t417 67 41\r\n",
      "11\t557\t0.0\t2\t494 29 34\r\n",
      "12\t490\t0.0\t2\t404 66 20\r\n",
      "13\t499\t0.0\t2\t454 30 15\r\n",
      "14\t483\t0.0\t2\t453 20 10\r\n",
      "15\t574\t0.0\t2\t536 27 11\r\n",
      "16\t500\t0.0\t2\t460 21 19\r\n",
      "17\t462\t0.0\t2\t442 14 6\r\n",
      "18\t469\t0.0\t2\t431 17 21\r\n",
      "19\t505\t0.0\t2\t472 22 11\r\n",
      "20\t506\t0.0\t2\t471 23 12\r\n",
      "21\t452\t0.0\t2\t418 19 15\r\n",
      "22\t520\t0.0\t2\t480 23 17\r\n",
      "23\t453\t0.0\t2\t411 21 21\r\n",
      "24\t491\t0.0\t2\t456 15 20\r\n",
      "25\t518\t0.0\t2\t487 21 10\r\n",
      "26\t460\t0.0\t2\t424 25 11\r\n",
      "27\t400\t0.0\t2\t380 14 6\r\n",
      "28\t363\t0.0\t2\t332 18 13\r\n",
      "29\t346\t0.0\t2\t320 12 14\r\n",
      "30\t446\t0.0\t2\t412 18 16\r\n",
      "31\t402\t0.0\t2\t378 16 8\r\n",
      "32\t378\t0.0\t2\t353 13 12\r\n",
      "33\t340\t0.0\t2\t313 13 14\r\n",
      "34\t271\t0.0\t2\t244 14 13\r\n",
      "35\t359\t0.0\t2\t335 13 11\r\n",
      "36\t322\t0.0\t2\t301 9 12\r\n",
      "37\t216\t0.0\t2\t202 6 8\r\n",
      "38\t137\t0.0\t2\t125 6 6\r\n",
      "39\t130\t0.0\t2\t104 5 21\r\n",
      "40\t93\t0.0\t2\t76 4 13\r\n",
      "41\t56\t0.0\t2\t41 3 12\r\n",
      "42\t44\t0.0\t2\t26 1 17\r\n",
      "43\t20\t0.0\t2\t9 0 11\r\n",
      "44\t23\t0.0\t2\t13 0 10\r\n",
      "45\t15\t0.0\t2\t7 0 8\r\n",
      "46\t26\t0.0\t2\t17 1 8\r\n",
      "47\t29\t0.0\t2\t15 2 12\r\n",
      "48\t27\t0.0\t2\t15 1 11\r\n",
      "49\t20\t0.0\t2\t9 0 11\r\n",
      "50\t16\t0.0\t2\t6 0 10\r\n",
      "51\t22\t0.0\t2\t1 1 20\r\n",
      "52\t7\t0.0\t2\t0 0 7\r\n",
      "53\t16\t0.0\t2\t0 0 16\r\n",
      "54\t8\t0.0\t2\t0 2 6\r\n",
      "55\t8\t0.0\t2\t0 0 8\r\n",
      "56\t4\t0.0\t2\t0 0 4\r\n",
      "57\t3\t0.0\t2\t0 0 3\r\n",
      "58\t8\t0.0\t2\t0 0 8\r\n",
      "59\t6\t0.0\t2\t0 0 6\r\n",
      "60\t3\t0.0\t2\t0 0 3\r\n",
      "61\t5\t0.0\t2\t0 0 5\r\n",
      "62\t1\t0.0\t2\t0 0 1\r\n",
      "63\t17\t0.0\t2\t0 0 17\r\n",
      "64\t6\t0.0\t2\t0 1 5\r\n",
      "65\t12\t0.0\t2\t0 1 11\r\n",
      "66\t17\t0.0\t2\t0 0 17\r\n",
      "67\t8\t0.0\t2\t0 0 8\r\n",
      "68\t55\t0.0\t2\t0 1 54\r\n",
      "69\t24\t0.0\t2\t0 0 24\r\n",
      "70\t17\t0.0\t2\t0 1 16\r\n",
      "71\t6\t0.0\t2\t0 1 5\r\n",
      "72\t3\t0.0\t2\t0 0 3\r\n",
      "73\t19\t0.0\t2\t0 0 19\r\n",
      "74\t27\t0.0\t2\t0 1 26\r\n",
      "75\t12\t0.0\t2\t0 0 12\r\n",
      "76\t9\t0.0\t2\t0 0 9\r\n",
      "\r\n",
      "=== Second read: Adapter 2 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 20916 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 17.1%\r\n",
      "  C: 33.2%\r\n",
      "  G: 25.9%\r\n",
      "  T: 23.2%\r\n",
      "  none/other: 0.5%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t1404\t1388.0\t0\t1404\r\n",
      "4\t1580\t347.0\t0\t702 878\r\n",
      "5\t1691\t86.8\t1\t505 1186\r\n",
      "6\t1065\t21.7\t1\t472 593\r\n",
      "7\t602\t5.4\t1\t464 138\r\n",
      "8\t569\t1.4\t1\t470 88 11\r\n",
      "9\t584\t0.3\t1\t511 18 55\r\n",
      "10\t524\t0.1\t2\t441 50 33\r\n",
      "11\t559\t0.0\t2\t528 0 31\r\n",
      "12\t493\t0.0\t2\t430 48 15\r\n",
      "13\t498\t0.0\t2\t486 4 8\r\n",
      "14\t491\t0.0\t2\t473 2 16\r\n",
      "15\t582\t0.0\t2\t565 1 16\r\n",
      "16\t499\t0.0\t2\t479 8 12\r\n",
      "17\t468\t0.0\t2\t447 7 14\r\n",
      "18\t469\t0.0\t2\t449 6 14\r\n",
      "19\t510\t0.0\t2\t491 4 15\r\n",
      "20\t517\t0.0\t2\t497 8 12\r\n",
      "21\t451\t0.0\t2\t439 3 9\r\n",
      "22\t519\t0.0\t2\t503 5 11\r\n",
      "23\t450\t0.0\t2\t339 97 14\r\n",
      "24\t487\t0.0\t2\t351 125 11\r\n",
      "25\t514\t0.0\t2\t404 107 3\r\n",
      "26\t458\t0.0\t2\t360 90 8\r\n",
      "27\t400\t0.0\t2\t300 91 9\r\n",
      "28\t361\t0.0\t2\t275 77 9\r\n",
      "29\t346\t0.0\t2\t252 79 15\r\n",
      "30\t441\t0.0\t2\t334 99 8\r\n",
      "31\t403\t0.0\t2\t311 83 9\r\n",
      "32\t373\t0.0\t2\t287 76 10\r\n",
      "33\t339\t0.0\t2\t247 69 23\r\n",
      "34\t266\t0.0\t2\t215 3 48\r\n",
      "35\t358\t0.0\t2\t342 7 9\r\n",
      "36\t322\t0.0\t2\t309 0 13\r\n",
      "37\t224\t0.0\t2\t205 7 12\r\n",
      "38\t138\t0.0\t2\t128 4 6\r\n",
      "39\t123\t0.0\t2\t109 0 14\r\n",
      "40\t85\t0.0\t2\t80 0 5\r\n",
      "41\t63\t0.0\t2\t44 1 18\r\n",
      "42\t40\t0.0\t2\t26 1 13\r\n",
      "43\t23\t0.0\t2\t9 1 13\r\n",
      "44\t27\t0.0\t2\t13 1 13\r\n",
      "45\t17\t0.0\t2\t7 1 9\r\n",
      "46\t33\t0.0\t2\t18 0 15\r\n",
      "47\t26\t0.0\t2\t16 0 10\r\n",
      "48\t26\t0.0\t2\t15 1 10\r\n",
      "49\t28\t0.0\t2\t9 0 19\r\n",
      "50\t15\t0.0\t2\t6 0 9\r\n",
      "51\t23\t0.0\t2\t1 2 20\r\n",
      "52\t8\t0.0\t2\t0 0 8\r\n",
      "53\t34\t0.0\t2\t0 3 31\r\n",
      "54\t23\t0.0\t2\t2 0 21\r\n",
      "55\t12\t0.0\t2\t0 0 12\r\n",
      "56\t20\t0.0\t2\t0 1 19\r\n",
      "57\t13\t0.0\t2\t0 0 13\r\n",
      "58\t27\t0.0\t2\t0 1 26\r\n",
      "59\t25\t0.0\t2\t0 1 24\r\n",
      "60\t12\t0.0\t2\t0 0 12\r\n",
      "61\t18\t0.0\t2\t0 1 17\r\n",
      "62\t5\t0.0\t2\t0 0 5\r\n",
      "63\t26\t0.0\t2\t0 0 26\r\n",
      "64\t17\t0.0\t2\t0 0 17\r\n",
      "65\t20\t0.0\t2\t0 1 19\r\n",
      "66\t17\t0.0\t2\t0 0 17\r\n",
      "67\t11\t0.0\t2\t0 1 10\r\n",
      "68\t81\t0.0\t2\t0 6 75\r\n",
      "69\t13\t0.0\t2\t0 3 10\r\n",
      "70\t26\t0.0\t2\t0 0 26\r\n",
      "71\t5\t0.0\t2\t0 0 5\r\n",
      "72\t1\t0.0\t2\t0 0 1\r\n",
      "73\t3\t0.0\t2\t0 0 3\r\n",
      "74\t15\t0.0\t2\t0 0 15\r\n",
      "\r\n",
      "cutadapt -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /tc2016/ubuntu/analysis/WT-3h_S14_L001_R1_001.trimmed.fastq.gz -p /tc2016/ubuntu/analysis/WT-3h_S14_L001_R2_001.trimmed.fastq.gz /tc2016/ubuntu/data/fastq/WT-3h_S14_L001_R1_001.fastq.gz /tc2016/ubuntu/data/fastq/WT-3h_S14_L001_R2_001.fastq.gz\r\n",
      "This is cutadapt 1.10 with Python 3.5.2\r\n",
      "Command line parameters: -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /tc2016/ubuntu/analysis/WT-3h_S14_L001_R1_001.trimmed.fastq.gz -p /tc2016/ubuntu/analysis/WT-3h_S14_L001_R2_001.trimmed.fastq.gz /tc2016/ubuntu/data/fastq/WT-3h_S14_L001_R1_001.fastq.gz /tc2016/ubuntu/data/fastq/WT-3h_S14_L001_R2_001.fastq.gz\r\n",
      "Trimming 2 adapters with at most 20.0% errors in paired-end mode ...\r\n",
      "Finished in 85.42 s (193 us/read; 0.31 M reads/minute).\r\n",
      "\r\n",
      "=== Summary ===\r\n",
      "\r\n",
      "Total read pairs processed:            443,115\r\n",
      "  Read 1 with adapter:                 135,800 (30.6%)\r\n",
      "  Read 2 with adapter:                 136,454 (30.8%)\r\n",
      "Pairs that were too short:                 357 (0.1%)\r\n",
      "Pairs written (passing filters):       442,758 (99.9%)\r\n",
      "\r\n",
      "Total basepairs processed:    67,353,480 bp\r\n",
      "  Read 1:    33,676,740 bp\r\n",
      "  Read 2:    33,676,740 bp\r\n",
      "Total written (filtered):     61,955,869 bp (92.0%)\r\n",
      "  Read 1:    30,982,457 bp\r\n",
      "  Read 2:    30,973,412 bp\r\n",
      "\r\n",
      "=== First read: Adapter 1 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 135800 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 14.6%\r\n",
      "  C: 36.9%\r\n",
      "  G: 24.6%\r\n",
      "  T: 23.8%\r\n",
      "  none/other: 0.0%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t6694\t6923.7\t0\t6694\r\n",
      "4\t7627\t1730.9\t0\t4024 3603\r\n",
      "5\t7987\t432.7\t1\t2899 5088\r\n",
      "6\t5140\t108.2\t1\t2581 2559\r\n",
      "7\t3226\t27.0\t1\t2717 509\r\n",
      "8\t3629\t6.8\t1\t2894 696 39\r\n",
      "9\t3155\t1.7\t1\t2823 81 251\r\n",
      "10\t3008\t0.4\t2\t2545 281 182\r\n",
      "11\t3067\t0.1\t2\t2908 45 114\r\n",
      "12\t3030\t0.0\t2\t2675 311 44\r\n",
      "13\t3078\t0.0\t2\t2991 45 42\r\n",
      "14\t3121\t0.0\t2\t3043 36 42\r\n",
      "15\t3169\t0.0\t2\t3099 36 34\r\n",
      "16\t3092\t0.0\t2\t2996 48 48\r\n",
      "17\t3141\t0.0\t2\t3053 33 55\r\n",
      "18\t3147\t0.0\t2\t3065 40 42\r\n",
      "19\t3371\t0.0\t2\t3252 41 78\r\n",
      "20\t3452\t0.0\t2\t3366 39 47\r\n",
      "21\t3437\t0.0\t2\t3340 50 47\r\n",
      "22\t3315\t0.0\t2\t3224 41 50\r\n",
      "23\t3204\t0.0\t2\t3125 24 55\r\n",
      "24\t3445\t0.0\t2\t3367 46 32\r\n",
      "25\t3948\t0.0\t2\t3879 40 29\r\n",
      "26\t3709\t0.0\t2\t3635 36 38\r\n",
      "27\t3234\t0.0\t2\t3179 28 27\r\n",
      "28\t3060\t0.0\t2\t2987 23 50\r\n",
      "29\t3268\t0.0\t2\t3180 40 48\r\n",
      "30\t3444\t0.0\t2\t3372 37 35\r\n",
      "31\t3355\t0.0\t2\t3275 37 43\r\n",
      "32\t3194\t0.0\t2\t3120 44 30\r\n",
      "33\t2886\t0.0\t2\t2803 28 55\r\n",
      "34\t2735\t0.0\t2\t2654 29 52\r\n",
      "35\t3619\t0.0\t2\t3560 27 32\r\n",
      "36\t3660\t0.0\t2\t3581 36 43\r\n",
      "37\t2317\t0.0\t2\t2254 32 31\r\n",
      "38\t1754\t0.0\t2\t1702 19 33\r\n",
      "39\t1261\t0.0\t2\t1200 9 52\r\n",
      "40\t937\t0.0\t2\t878 11 48\r\n",
      "41\t463\t0.0\t2\t413 9 41\r\n",
      "42\t303\t0.0\t2\t198 4 101\r\n",
      "43\t202\t0.0\t2\t156 3 43\r\n",
      "44\t141\t0.0\t2\t100 3 38\r\n",
      "45\t173\t0.0\t2\t143 3 27\r\n",
      "46\t456\t0.0\t2\t415 3 38\r\n",
      "47\t333\t0.0\t2\t293 1 39\r\n",
      "48\t177\t0.0\t2\t150 3 24\r\n",
      "49\t186\t0.0\t2\t119 3 64\r\n",
      "50\t84\t0.0\t2\t62 2 20\r\n",
      "51\t85\t0.0\t2\t3 0 82\r\n",
      "52\t41\t0.0\t2\t4 0 37\r\n",
      "53\t98\t0.0\t2\t1 4 93\r\n",
      "54\t28\t0.0\t2\t1 1 26\r\n",
      "55\t31\t0.0\t2\t1 1 29\r\n",
      "56\t50\t0.0\t2\t4 3 43\r\n",
      "57\t23\t0.0\t2\t2 1 20\r\n",
      "58\t37\t0.0\t2\t0 6 31\r\n",
      "59\t41\t0.0\t2\t0 2 39\r\n",
      "60\t25\t0.0\t2\t0 0 25\r\n",
      "61\t30\t0.0\t2\t0 0 30\r\n",
      "62\t25\t0.0\t2\t0 0 25\r\n",
      "63\t83\t0.0\t2\t0 2 81\r\n",
      "64\t56\t0.0\t2\t0 1 55\r\n",
      "65\t45\t0.0\t2\t0 1 44\r\n",
      "66\t51\t0.0\t2\t0 1 50\r\n",
      "67\t25\t0.0\t2\t0 0 25\r\n",
      "68\t197\t0.0\t2\t0 9 188\r\n",
      "69\t67\t0.0\t2\t0 3 64\r\n",
      "70\t68\t0.0\t2\t0 0 68\r\n",
      "71\t19\t0.0\t2\t0 0 19\r\n",
      "72\t25\t0.0\t2\t0 2 23\r\n",
      "73\t72\t0.0\t2\t0 1 71\r\n",
      "74\t93\t0.0\t2\t0 1 92\r\n",
      "75\t28\t0.0\t2\t0 0 28\r\n",
      "76\t23\t0.0\t2\t0 1 22\r\n",
      "\r\n",
      "=== Second read: Adapter 2 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 136454 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 15.2%\r\n",
      "  C: 36.4%\r\n",
      "  G: 24.8%\r\n",
      "  T: 22.9%\r\n",
      "  none/other: 0.8%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t6714\t6923.7\t0\t6714\r\n",
      "4\t7999\t1730.9\t0\t4027 3972\r\n",
      "5\t8412\t432.7\t1\t2890 5522\r\n",
      "6\t5166\t108.2\t1\t2333 2833\r\n",
      "7\t3327\t27.0\t1\t2711 616\r\n",
      "8\t3562\t6.8\t1\t2807 725 30\r\n",
      "9\t3117\t1.7\t1\t2786 93 238\r\n",
      "10\t2975\t0.4\t2\t2408 399 168\r\n",
      "11\t3072\t0.1\t2\t2882 60 130\r\n",
      "12\t3035\t0.0\t2\t2584 387 64\r\n",
      "13\t3077\t0.0\t2\t2941 83 53\r\n",
      "14\t3106\t0.0\t2\t2777 276 53\r\n",
      "15\t3152\t0.0\t2\t2818 287 47\r\n",
      "16\t3090\t0.0\t2\t2756 271 63\r\n",
      "17\t3119\t0.0\t2\t2769 286 64\r\n",
      "18\t3142\t0.0\t2\t2803 263 76\r\n",
      "19\t3360\t0.0\t2\t2966 309 85\r\n",
      "20\t3432\t0.0\t2\t3076 273 83\r\n",
      "21\t3442\t0.0\t2\t3037 334 71\r\n",
      "22\t3309\t0.0\t2\t2938 301 70\r\n",
      "23\t3178\t0.0\t2\t2084 1022 72\r\n",
      "24\t3438\t0.0\t2\t2195 1145 98\r\n",
      "25\t3939\t0.0\t2\t2612 995 332\r\n",
      "26\t3689\t0.0\t2\t2668 963 58\r\n",
      "27\t3229\t0.0\t2\t2355 766 108\r\n",
      "28\t3036\t0.0\t2\t2183 809 44\r\n",
      "29\t3241\t0.0\t2\t2305 839 97\r\n",
      "30\t3438\t0.0\t2\t2487 876 75\r\n",
      "31\t3342\t0.0\t2\t2438 828 76\r\n",
      "32\t3192\t0.0\t2\t2274 851 67\r\n",
      "33\t2879\t0.0\t2\t2032 737 110\r\n",
      "34\t2719\t0.0\t2\t1949 51 719\r\n",
      "35\t3614\t0.0\t2\t3443 91 80\r\n",
      "36\t3652\t0.0\t2\t3486 111 55\r\n",
      "37\t2319\t0.0\t2\t2212 57 50\r\n",
      "38\t1747\t0.0\t2\t1671 41 35\r\n",
      "39\t1267\t0.0\t2\t1171 30 66\r\n",
      "40\t935\t0.0\t2\t846 30 59\r\n",
      "41\t470\t0.0\t2\t408 13 49\r\n",
      "42\t311\t0.0\t2\t192 12 107\r\n",
      "43\t199\t0.0\t2\t150 6 43\r\n",
      "44\t147\t0.0\t2\t97 5 45\r\n",
      "45\t196\t0.0\t2\t143 1 52\r\n",
      "46\t463\t0.0\t2\t406 15 42\r\n",
      "47\t330\t0.0\t2\t282 13 35\r\n",
      "48\t180\t0.0\t2\t148 7 25\r\n",
      "49\t170\t0.0\t2\t116 7 47\r\n",
      "50\t93\t0.0\t2\t60 3 30\r\n",
      "51\t96\t0.0\t2\t3 1 92\r\n",
      "52\t37\t0.0\t2\t4 0 33\r\n",
      "53\t127\t0.0\t2\t2 7 118\r\n",
      "54\t49\t0.0\t2\t2 2 45\r\n",
      "55\t49\t0.0\t2\t1 0 48\r\n",
      "56\t81\t0.0\t2\t7 0 74\r\n",
      "57\t40\t0.0\t2\t2 1 37\r\n",
      "58\t50\t0.0\t2\t4 3 43\r\n",
      "59\t63\t0.0\t2\t0 2 61\r\n",
      "60\t50\t0.0\t2\t0 0 50\r\n",
      "61\t56\t0.0\t2\t0 0 56\r\n",
      "62\t33\t0.0\t2\t0 1 32\r\n",
      "63\t87\t0.0\t2\t0 1 86\r\n",
      "64\t61\t0.0\t2\t0 2 59\r\n",
      "65\t38\t0.0\t2\t0 4 34\r\n",
      "66\t58\t0.0\t2\t0 0 58\r\n",
      "67\t37\t0.0\t2\t0 3 34\r\n",
      "68\t169\t0.0\t2\t0 7 162\r\n",
      "69\t45\t0.0\t2\t0 5 40\r\n",
      "70\t74\t0.0\t2\t0 1 73\r\n",
      "71\t17\t0.0\t2\t0 0 17\r\n",
      "72\t14\t0.0\t2\t0 0 14\r\n",
      "73\t12\t0.0\t2\t0 0 12\r\n",
      "74\t77\t0.0\t2\t0 0 77\r\n",
      "75\t7\t0.0\t2\t0 0 7\r\n",
      "76\t6\t0.0\t2\t0 0 6\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "for R1_fastq in ${FASTQ_DIR}*_R1*fastq.gz; do\n",
    "    \n",
    "    # Get the read 2 fastq file from the filename of read 1\n",
    "    R2_fastq=$(echo $R1_fastq | sed -e 's/R1/R2/')\n",
    "    \n",
    "    # Generate names for the trimmed fastq files\n",
    "\n",
    "    trimmed_R1_fastq=$ANALYSIS_DIR$(echo $(basename $R1_fastq)| sed -e 's/.fastq.gz/.trimmed.fastq.gz/')\n",
    "    trimmed_R2_fastq=$ANALYSIS_DIR$(echo $(basename $R2_fastq)| sed -e 's/.fastq.gz/.trimmed.fastq.gz/')   \n",
    "    echo cutadapt -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA \\\n",
    "        -o ${trimmed_R1_fastq} \\\n",
    "        -p ${trimmed_R2_fastq} \\\n",
    "        $R1_fastq \\\n",
    "        $R2_fastq\n",
    "    cutadapt -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA \\\n",
    "        -o ${trimmed_R1_fastq} \\\n",
    "        -p ${trimmed_R2_fastq} \\\n",
    "        $R1_fastq \\\n",
    "        $R2_fastq\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": false
   },
   "source": [
    "## Part 3: Alignment\n",
    "\n",
    "Now, we're ready to align our trimmed reads to the Yeast SacCer3 reference genome.\n",
    "\n",
    "We'll use [Bowtie2](http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml), which is a [Burrows-Wheeler](https://en.wikipedia.org/wiki/Burrows%E2%80%93Wheeler_transform) based spliced aligner.\n",
    "\n",
    "Bowtie2 outputs a SAM (Sequence Alignment Map) file, which is a standard text encoding. To save space, we'll use `samtools view -b` to encode the output as a binarized SAM file — a BAM file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/srv/scratch/training_camp/saccer3/bowtie2_index/saccer3\r\n"
     ]
    }
   ],
   "source": [
    "export bowtie_index=$YEAST_INDEX\n",
    "echo $bowtie_index\n",
    "\n",
    "#create a directory to store the aligned data \n",
    "export ALIGNMENT_DIR=\"$ANALYSIS_DIR/aligned/\"\n",
    "[[ ! -d $ALIGNMENT_DIR ]] && mkdir -p \"$ALIGNMENT_DIR\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[samopen] SAM header is present: 17 sequences.\r\n",
      "153476 reads; of these:\r\n",
      "  153476 (100.00%) were paired; of these:\r\n",
      "    7206 (4.70%) aligned concordantly 0 times\r\n",
      "    75741 (49.35%) aligned concordantly exactly 1 time\r\n",
      "    70529 (45.95%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    7206 pairs aligned concordantly 0 times; of these:\r\n",
      "      1000 (13.88%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    6206 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      12412 mates make up the pairs; of these:\r\n",
      "        10043 (80.91%) aligned 0 times\r\n",
      "        232 (1.87%) aligned exactly 1 time\r\n",
      "        2137 (17.22%) aligned >1 times\r\n",
      "96.73% overall alignment rate\r\n",
      "[samopen] SAM header is present: 17 sequences.\r\n",
      "450816 reads; of these:\r\n",
      "  450816 (100.00%) were paired; of these:\r\n",
      "    18997 (4.21%) aligned concordantly 0 times\r\n",
      "    190331 (42.22%) aligned concordantly exactly 1 time\r\n",
      "    241488 (53.57%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    18997 pairs aligned concordantly 0 times; of these:\r\n",
      "      2522 (13.28%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    16475 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      32950 mates make up the pairs; of these:\r\n",
      "        24797 (75.26%) aligned 0 times\r\n",
      "        788 (2.39%) aligned exactly 1 time\r\n",
      "        7365 (22.35%) aligned >1 times\r\n",
      "97.25% overall alignment rate\r\n",
      "[samopen] SAM header is present: 17 sequences.\r\n",
      "18854 reads; of these:\r\n",
      "  18854 (100.00%) were paired; of these:\r\n",
      "    757 (4.02%) aligned concordantly 0 times\r\n",
      "    10642 (56.44%) aligned concordantly exactly 1 time\r\n",
      "    7455 (39.54%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    757 pairs aligned concordantly 0 times; of these:\r\n",
      "      93 (12.29%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    664 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      1328 mates make up the pairs; of these:\r\n",
      "        1078 (81.17%) aligned 0 times\r\n",
      "        79 (5.95%) aligned exactly 1 time\r\n",
      "        171 (12.88%) aligned >1 times\r\n",
      "97.14% overall alignment rate\r\n",
      "[samopen] SAM header is present: 17 sequences.\r\n",
      "434218 reads; of these:\r\n",
      "  434218 (100.00%) were paired; of these:\r\n",
      "    18552 (4.27%) aligned concordantly 0 times\r\n",
      "    210067 (48.38%) aligned concordantly exactly 1 time\r\n",
      "    205599 (47.35%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    18552 pairs aligned concordantly 0 times; of these:\r\n",
      "      2759 (14.87%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    15793 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      31586 mates make up the pairs; of these:\r\n",
      "        24340 (77.06%) aligned 0 times\r\n",
      "        870 (2.75%) aligned exactly 1 time\r\n",
      "        6376 (20.19%) aligned >1 times\r\n",
      "97.20% overall alignment rate\r\n",
      "[samopen] SAM header is present: 17 sequences.\r\n",
      "157271 reads; of these:\r\n",
      "  157271 (100.00%) were paired; of these:\r\n",
      "    7658 (4.87%) aligned concordantly 0 times\r\n",
      "    78181 (49.71%) aligned concordantly exactly 1 time\r\n",
      "    71432 (45.42%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    7658 pairs aligned concordantly 0 times; of these:\r\n",
      "      1081 (14.12%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    6577 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      13154 mates make up the pairs; of these:\r\n",
      "        10689 (81.26%) aligned 0 times\r\n",
      "        245 (1.86%) aligned exactly 1 time\r\n",
      "        2220 (16.88%) aligned >1 times\r\n",
      "96.60% overall alignment rate\r\n",
      "[samopen] SAM header is present: 17 sequences.\r\n",
      "416040 reads; of these:\r\n",
      "  416040 (100.00%) were paired; of these:\r\n",
      "    19109 (4.59%) aligned concordantly 0 times\r\n",
      "    216184 (51.96%) aligned concordantly exactly 1 time\r\n",
      "    180747 (43.44%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    19109 pairs aligned concordantly 0 times; of these:\r\n",
      "      2765 (14.47%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    16344 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      32688 mates make up the pairs; of these:\r\n",
      "        26079 (79.78%) aligned 0 times\r\n",
      "        989 (3.03%) aligned exactly 1 time\r\n",
      "        5620 (17.19%) aligned >1 times\r\n",
      "96.87% overall alignment rate\r\n",
      "[samopen] SAM header is present: 17 sequences.\r\n",
      "127700 reads; of these:\r\n",
      "  127700 (100.00%) were paired; of these:\r\n",
      "    6001 (4.70%) aligned concordantly 0 times\r\n",
      "    65373 (51.19%) aligned concordantly exactly 1 time\r\n",
      "    56326 (44.11%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    6001 pairs aligned concordantly 0 times; of these:\r\n",
      "      721 (12.01%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    5280 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      10560 mates make up the pairs; of these:\r\n",
      "        8753 (82.89%) aligned 0 times\r\n",
      "        221 (2.09%) aligned exactly 1 time\r\n",
      "        1586 (15.02%) aligned >1 times\r\n",
      "96.57% overall alignment rate\r\n",
      "[samopen] SAM header is present: 17 sequences.\r\n",
      "432071 reads; of these:\r\n",
      "  432071 (100.00%) were paired; of these:\r\n",
      "    20183 (4.67%) aligned concordantly 0 times\r\n",
      "    205449 (47.55%) aligned concordantly exactly 1 time\r\n",
      "    206439 (47.78%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    20183 pairs aligned concordantly 0 times; of these:\r\n",
      "      2732 (13.54%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    17451 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      34902 mates make up the pairs; of these:\r\n",
      "        27517 (78.84%) aligned 0 times\r\n",
      "        921 (2.64%) aligned exactly 1 time\r\n",
      "        6464 (18.52%) aligned >1 times\r\n",
      "96.82% overall alignment rate\r\n",
      "[samopen] SAM header is present: 17 sequences.\r\n",
      "62968 reads; of these:\r\n",
      "  62968 (100.00%) were paired; of these:\r\n",
      "    3122 (4.96%) aligned concordantly 0 times\r\n",
      "    30519 (48.47%) aligned concordantly exactly 1 time\r\n",
      "    29327 (46.57%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    3122 pairs aligned concordantly 0 times; of these:\r\n",
      "      399 (12.78%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    2723 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      5446 mates make up the pairs; of these:\r\n",
      "        4479 (82.24%) aligned 0 times\r\n",
      "        106 (1.95%) aligned exactly 1 time\r\n",
      "        861 (15.81%) aligned >1 times\r\n",
      "96.44% overall alignment rate\r\n",
      "[samopen] SAM header is present: 17 sequences.\r\n",
      "356973 reads; of these:\r\n",
      "  356973 (100.00%) were paired; of these:\r\n",
      "    15313 (4.29%) aligned concordantly 0 times\r\n",
      "    169286 (47.42%) aligned concordantly exactly 1 time\r\n",
      "    172374 (48.29%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    15313 pairs aligned concordantly 0 times; of these:\r\n",
      "      2298 (15.01%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    13015 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      26030 mates make up the pairs; of these:\r\n",
      "        20142 (77.38%) aligned 0 times\r\n",
      "        671 (2.58%) aligned exactly 1 time\r\n",
      "        5217 (20.04%) aligned >1 times\r\n",
      "97.18% overall alignment rate\r\n",
      "[samopen] SAM header is present: 17 sequences.\r\n",
      "209758 reads; of these:\r\n",
      "  209758 (100.00%) were paired; of these:\r\n",
      "    9720 (4.63%) aligned concordantly 0 times\r\n",
      "    112594 (53.68%) aligned concordantly exactly 1 time\r\n",
      "    87444 (41.69%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    9720 pairs aligned concordantly 0 times; of these:\r\n",
      "      1534 (15.78%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    8186 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      16372 mates make up the pairs; of these:\r\n",
      "        13204 (80.65%) aligned 0 times\r\n",
      "        422 (2.58%) aligned exactly 1 time\r\n",
      "        2746 (16.77%) aligned >1 times\r\n",
      "96.85% overall alignment rate\r\n",
      "[samopen] SAM header is present: 17 sequences.\r\n",
      "324846 reads; of these:\r\n",
      "  324846 (100.00%) were paired; of these:\r\n",
      "    15358 (4.73%) aligned concordantly 0 times\r\n",
      "    172373 (53.06%) aligned concordantly exactly 1 time\r\n",
      "    137115 (42.21%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    15358 pairs aligned concordantly 0 times; of these:\r\n",
      "      2294 (14.94%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    13064 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      26128 mates make up the pairs; of these:\r\n",
      "        21026 (80.47%) aligned 0 times\r\n",
      "        812 (3.11%) aligned exactly 1 time\r\n",
      "        4290 (16.42%) aligned >1 times\r\n",
      "96.76% overall alignment rate\r\n",
      "[samopen] SAM header is present: 17 sequences.\r\n",
      "88746 reads; of these:\r\n",
      "  88746 (100.00%) were paired; of these:\r\n",
      "    4442 (5.01%) aligned concordantly 0 times\r\n",
      "    45865 (51.68%) aligned concordantly exactly 1 time\r\n",
      "    38439 (43.31%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    4442 pairs aligned concordantly 0 times; of these:\r\n",
      "      504 (11.35%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    3938 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      7876 mates make up the pairs; of these:\r\n",
      "        6577 (83.51%) aligned 0 times\r\n",
      "        142 (1.80%) aligned exactly 1 time\r\n",
      "        1157 (14.69%) aligned >1 times\r\n",
      "96.29% overall alignment rate\r\n",
      "[samopen] SAM header is present: 17 sequences.\r\n",
      "442758 reads; of these:\r\n",
      "  442758 (100.00%) were paired; of these:\r\n",
      "    20067 (4.53%) aligned concordantly 0 times\r\n",
      "    184021 (41.56%) aligned concordantly exactly 1 time\r\n",
      "    238670 (53.91%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    20067 pairs aligned concordantly 0 times; of these:\r\n",
      "      2535 (12.63%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    17532 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      35064 mates make up the pairs; of these:\r\n",
      "        27256 (77.73%) aligned 0 times\r\n",
      "        803 (2.29%) aligned exactly 1 time\r\n",
      "        7005 (19.98%) aligned >1 times\r\n",
      "96.92% overall alignment rate\r\n"
     ]
    }
   ],
   "source": [
    "for trimmed_fq1 in ${ANALYSIS_DIR}*_R1*fastq.gz; do\n",
    "\n",
    "    trimmed_fq2=$(echo $trimmed_fq1 | sed -e 's/_R1/_R2/')\n",
    "    \n",
    "    bam=$(echo \"${ALIGNMENT_DIR}${trimmed_fq1##*/}\" | sed -e 's/.fastq.gz/.bam/')\n",
    "    bowtie2 -X2000 --mm --threads 5 -x $bowtie_index -1 $trimmed_fq1 -2 $trimmed_fq2 | samtools view -bS - > $bam        \n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Part 4: Finding duplicate reads and alignment filtering\n",
    "\n",
    "During sequencing, we perform PCR, which can lead to duplicate reads. In many kinds of DNA sequencing, we want to remove duplicates so that we don't double-count signal originating from the same molecule.\n",
    "\n",
    "To do so, we use an algorithm called `sambamba` that looks for reads that mapped to exactly the same places in the genome. We also need to sort the aligned files before we can mark duplicates, since we need reads aligned to the same position to be next to each other in the file.\n",
    "\n",
    "Bowtie2 also sets certian labels (or \"flags\") in the resulting alignment file to indicate information like the score of the alignment, the orientation of both mates of the fragment, and other details.\n",
    "\n",
    "We can use these flags as a way to discard low-quality reads. [This website](https://broadinstitute.github.io/picard/explain-flags.html) provides a convenient way to interpret the meaning of these bitwise flags; for conveninece they can be encoded as numbers.\n",
    "\n",
    "Here, we want to filter reads that fall into any of the following categories:\n",
    "- the read wasn't mapped to the genome\n",
    "- the read's mate wasn't mapped to the genome\n",
    "- the alignment reported is not the primary alignment (it is a \"runner-up\" alignment)\n",
    "- the read was marked as \"low-quality\" by the sequencer software\n",
    "- the read has a mapping quality less than 30"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[bam_header_read] EOF marker is absent. The input is probably truncated.\r\n",
      "sambamba: command not found\r\n",
      "sambamba: command not found\r\n",
      "[bam_header_read] invalid BAM binary header (this is not a BAM file).\r\n",
      "[main_samview] fail to read the header from \"/dev/stdin\".\r\n",
      "[bam_header_read] EOF marker is absent. The input is probably truncated.\r\n",
      "sambamba: command not found\r\n",
      "sambamba: command not found\r\n",
      "[bam_header_read] invalid BAM binary header (this is not a BAM file).\r\n",
      "[main_samview] fail to read the header from \"/dev/stdin\".\r\n",
      "[bam_header_read] EOF marker is absent. The input is probably truncated.\r\n",
      "sambamba: command not found\r\n",
      "sambamba: command not found\r\n",
      "[bam_header_read] invalid BAM binary header (this is not a BAM file).\r\n",
      "[main_samview] fail to read the header from \"/dev/stdin\".\r\n",
      "[bam_header_read] EOF marker is absent. The input is probably truncated.\r\n",
      "sambamba: command not found\r\n",
      "sambamba: command not found\r\n",
      "[bam_header_read] invalid BAM binary header (this is not a BAM file).\r\n",
      "[main_samview] fail to read the header from \"/dev/stdin\".\r\n",
      "[bam_header_read] EOF marker is absent. The input is probably truncated.\r\n",
      "sambamba: command not found\r\n",
      "sambamba: command not found\r\n",
      "[bam_header_read] invalid BAM binary header (this is not a BAM file).\r\n",
      "[main_samview] fail to read the header from \"/dev/stdin\".\r\n",
      "[bam_header_read] EOF marker is absent. The input is probably truncated.\r\n",
      "sambamba: command not found\r\n",
      "sambamba: command not found\r\n",
      "[bam_header_read] invalid BAM binary header (this is not a BAM file).\r\n",
      "[main_samview] fail to read the header from \"/dev/stdin\".\r\n",
      "[bam_header_read] EOF marker is absent. The input is probably truncated.\r\n",
      "sambamba: command not found\r\n",
      "sambamba: command not found\r\n",
      "[bam_header_read] invalid BAM binary header (this is not a BAM file).\r\n",
      "[main_samview] fail to read the header from \"/dev/stdin\".\r\n",
      "[bam_header_read] EOF marker is absent. The input is probably truncated.\r\n",
      "sambamba: command not found\r\n",
      "sambamba: command not found\r\n",
      "[bam_header_read] invalid BAM binary header (this is not a BAM file).\r\n",
      "[main_samview] fail to read the header from \"/dev/stdin\".\r\n",
      "[bam_header_read] EOF marker is absent. The input is probably truncated.\r\n",
      "sambamba: command not found\r\n",
      "sambamba: command not found\r\n",
      "[bam_header_read] invalid BAM binary header (this is not a BAM file).\r\n",
      "[main_samview] fail to read the header from \"/dev/stdin\".\r\n",
      "[bam_header_read] EOF marker is absent. The input is probably truncated.\r\n",
      "sambamba: command not found\r\n",
      "sambamba: command not found\r\n",
      "[bam_header_read] invalid BAM binary header (this is not a BAM file).\r\n",
      "[main_samview] fail to read the header from \"/dev/stdin\".\r\n",
      "[bam_header_read] EOF marker is absent. The input is probably truncated.\r\n",
      "sambamba: command not found\r\n",
      "sambamba: command not found\r\n",
      "[bam_header_read] invalid BAM binary header (this is not a BAM file).\r\n",
      "[main_samview] fail to read the header from \"/dev/stdin\".\r\n",
      "[bam_header_read] EOF marker is absent. The input is probably truncated.\r\n",
      "sambamba: command not found\r\n",
      "sambamba: command not found\r\n",
      "[bam_header_read] invalid BAM binary header (this is not a BAM file).\r\n",
      "[main_samview] fail to read the header from \"/dev/stdin\".\r\n",
      "[bam_header_read] EOF marker is absent. The input is probably truncated.\r\n",
      "sambamba: command not found\r\n",
      "sambamba: command not found\r\n",
      "[bam_header_read] invalid BAM binary header (this is not a BAM file).\r\n",
      "[main_samview] fail to read the header from \"/dev/stdin\".\r\n",
      "[bam_header_read] EOF marker is absent. The input is probably truncated.\r\n",
      "sambamba: command not found\r\n",
      "sambamba: command not found\r\n",
      "[bam_header_read] invalid BAM binary header (this is not a BAM file).\r\n",
      "[main_samview] fail to read the header from \"/dev/stdin\".\r\n"
     ]
    }
   ],
   "source": [
    "for bam_file in ${ALIGNMENT_DIR}*.bam; do\n",
    "\n",
    "    nodup_bam_file=$(echo $bam_file | sed -e 's/.bam/.nodup.bam/')\n",
    "    \n",
    "    # Sort and remove duplicates\n",
    "    sambamba sort -m 4G -t 2 -u $bam_file | \\\n",
    "        sambamba markdup -l 0 -t 2 /dev/stdin /dev/stdout |\\\n",
    "        samtools view -F 1804 -f 2 -q 30 /dev/stdin $nodup_bam_file\n",
    "    \n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Part 5: Peak calling\n",
    "\n",
    "Now that we've aligned our reads to the genome and filtered the alignments, we want to identify areas of locally enriched signals, or \"peaks\".\n",
    "\n",
    "For ATAC-seq, peaks correspond to accessible regions. They can include promoters, enhancers, and other regulatory regions.\n",
    "\n",
    "We'll call peaks using [MACS2](http://liulab.dfci.harvard.edu/MACS/)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "PEAKS_DIR=\"${ANALYSIS_DIR}peaks/\"\n",
    "[[ ! -d $PEAKS_DIR ]] && mkdir -p \"$PEAKS_DIR\"\n",
    "\n",
    "SacCer3GenSz=12157105  # The sum of the sizes of the chromosomes in the SacCer3 genome\n",
    "\n",
    "Macs2PvalThresh=\"0.05\"  # The p-value threshold for calling peaks \n",
    "\n",
    "Macs2SmoothWindow=150  # The window size to smooth alignment signal over\n",
    "\n",
    "Macs2ShiftSize=$(python -c \"print(${Macs2SmoothWindow}*0.5)\")\n",
    "\n",
    "for nodup_bam_file in ${ALIGNMENT_DIR}*.nodup.bam; do\n",
    "    \n",
    "    # First, we need to convert each bam to a .tagAlign,\n",
    "    # which just contains the start/end positions of each read:\n",
    "    \n",
    "    tagalign_file=$(echo $nodup_bam_file | sed -e 's/.bam/.tagAlign.gz/')\n",
    "    bedtools bamtobed -i $bam | awk 'BEGIN{OFS=\"\\t\"}{$4=\"N\";$5=\"1000\";print $0}' | gzip -c > $tagalign_file\n",
    "    \n",
    "    # Now, we can run MACS:\n",
    "    output_prefix=$(echo $trimmed_fq1 | sed -e 's/.bam/.tagAlign.gz/')\n",
    "    macs2 callpeak \\\n",
    "        -t $tagalign_file -f BED -n \"$prefix\" -g \"$SacCer3GenSz\" -p $Macs2PvalThresh \\\n",
    "        --nomodel --shift -$Macs2ShiftSize --extsize $Macs2SmoothWindow -B --SPMR --keep-dup all --call-summits\n",
    "    \n",
    "done"
   ]
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
