{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Sequencing data analysis\n",
    "\n",
    "### IMPORTANT: Please make sure that your are using the bash kernel to run this notebook.\n",
    "\n",
    "### This notebook covers analysis of DNA sequencing data from raw files to processed signals.\n",
    "\n",
    "Although this analysis is for ATAC-seq data, many of the steps (especially the first section) are the same for other types of DNA sequencing experiments.\n",
    "\n",
    "We'll be doing the analysis in Bash, which is the standard language for UNIX command-line scripting."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Part 1: Setting up the data\n",
    "\n",
    "We start with raw `.fastq.gz` files, which are provided by the sequencing instrument. For each DNA molecule (read) that was sequenced, they provide the nucleotide sequence, and information about the quality of the signal of that nucleotide."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": []
    }
   ],
   "source": [
    "### Set up variables storing the location of our data\n",
    "### The proper way to load your variables is with the ~/.bashrc command, but this is very slow in iPython \n",
    "export SUNETID=\"$(whoami)\"\n",
    "export WORK_DIR=\"/srv/scratch/training_camp/tc2016/${SUNETID}\"\n",
    "export DATA_DIR=\"${WORK_DIR}/data\"\n",
    "export FASTQ_DIR=\"${DATA_DIR}/fastq/\"\n",
    "export SRC_DIR=\"${WORK_DIR}/src/training_camp/src/\"\n",
    "export ANALYSIS_DIR=\"${WORK_DIR}/analysis/\"\n",
    "export YEAST_DIR=\"/srv/scratch/training_camp/saccer3/seq\"\n",
    "export YEAST_INDEX=\"/srv/scratch/training_camp/saccer3/bowtie2_index/saccer3\"\n",
    "export YEAST_CHR=\"/srv/scratch/training_camp/saccer3/sacCer3.chrom.sizes\"\n",
    "export TMP=\"${WORK_DIR}/tmp\"\n",
    "export TEMP=$TMP \n",
    "export TMPDIR=$TMP\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now, let's check exactly which fastqs we have:\n",
    "\n",
    "(recall that the `ls` command lists the contents of a directory)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Ct_1_S22_R1.fastq.gz  Ct_300_S3_R1.fastq.gz  Ct_800_S9_R1.fastq.gz\r\n",
      "Ct_1_S22_R2.fastq.gz  Ct_300_S3_R2.fastq.gz  Ct_800_S9_R2.fastq.gz\r\n",
      "Ct_2_S23_R1.fastq.gz  Ct_3_S24_R1.fastq.gz\r\n",
      "Ct_2_S23_R2.fastq.gz  Ct_3_S24_R2.fastq.gz\r\n"
     ]
    }
   ],
   "source": [
    "ls $FASTQ_DIR"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "As a sanity check, we can also look at the size and last edited time of some of the fastqs by addind `-lrth` to the `ls` command:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 1.8G\r\n",
      "-rwxrwxr-x 1 ubuntu ubuntu 185M Sep 13 02:56 Ct_1_S22_R1.fastq.gz\r\n",
      "-rwxrwxr-x 1 ubuntu ubuntu 198M Sep 13 02:56 Ct_1_S22_R2.fastq.gz\r\n",
      "-rwxrwxr-x 1 ubuntu ubuntu 142M Sep 13 02:56 Ct_2_S23_R1.fastq.gz\r\n",
      "-rwxrwxr-x 1 ubuntu ubuntu 147M Sep 13 02:56 Ct_2_S23_R2.fastq.gz\r\n",
      "-rwxrwxr-x 1 ubuntu ubuntu 247M Sep 13 02:56 Ct_300_S3_R1.fastq.gz\r\n",
      "-rwxrwxr-x 1 ubuntu ubuntu 255M Sep 13 02:56 Ct_300_S3_R2.fastq.gz\r\n",
      "-rwxrwxr-x 1 ubuntu ubuntu 126M Sep 13 02:56 Ct_3_S24_R1.fastq.gz\r\n",
      "-rwxrwxr-x 1 ubuntu ubuntu 131M Sep 13 02:56 Ct_3_S24_R2.fastq.gz\r\n",
      "-rwxrwxr-x 1 ubuntu ubuntu 172M Sep 13 02:56 Ct_800_S9_R1.fastq.gz\r\n"
     ]
    }
   ],
   "source": [
    "ls -lrth $FASTQ_DIR | head"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Let's also inspect the format of one of the fastqs. Notice that each read takes up 4 lines:\n",
    "1. the read name\n",
    "2. the read's nucleotide sequence\n",
    "3. a '+' to indicate the record contains another line\n",
    "4. a quality score for each base (a number encoded as a letter)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@NS500418:473:H55FVAFXX:1:11101:6407:1042 1:N:0:AGGCAGAA+NCTACTCT\r\n",
      "ATACCNAGGGGCGCAATGTGCGTTCAAAGATTCGATGATTCACGGAATTCTGCAACTGTCTCTTATACACATCTCC\r\n",
      "+\r\n",
      "AAAAA#EAEEE<EEEEEEAA<EEEEEEEEEEEAEEEEEEEA/<66EEAEEEEAEE<E<EEEE/EEEE<AAEEEEAA\r\n",
      "@NS500418:473:H55FVAFXX:1:11101:7864:1042 1:N:0:AGGCAGAA+NCTACTCT\r\n",
      "GTGTTNGTTCATCTAGACAGCCGGACGGTGGCCATGGAAGTCGGAATCCGCTAAGGAGTGTGTAACAACTCACCGG\r\n",
      "+\r\n",
      "AAAAA#EEEEEEEEE/EEEEEEE<EEEAEEEAEEEEEEEEEEAEE/EEAE<EEE/EEEAEE/EEAEEA/E<AEEEA\r\n",
      "\r\n",
      "gzip: stdout: Broken pipe\r\n"
     ]
    }
   ],
   "source": [
    "zcat $(ls $FASTQ_DIR* | head -n 1) | head -n 8"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Part 2: Adapter trimming\n",
    "\n",
    "- In many kinds of DNA and RNA sequencing experiments, sometimes the sequences will read through the targeted sequence insert and into sequencing adapter or PCR primer sequences on the end of the fragment. When the insert size is shorter than the read length (like in some of our ATAC-seq reads), the adapter sequence is read by the sequencer.\n",
    "\n",
    "- We need to remove such adapter sequences because they won't align to the genome.\n",
    "\n",
    "- In ATAC-seq (the data we're analyzing), the fragment length follows a periodic distribution. Some reads have very short inserts (only a few basepairs), while other reads have inserts that are much longer (100's of basepairs — much longer than the 77bp reads we're using to read them.\n",
    "\n",
    "- We know ahead of time that the first part of the adapter sequence is `CTGTCTCTTATA`, since our reads are sequenced using a Nextera sample prep kit."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24856\r\n",
      "\r\n",
      "gzip: stdout: Broken pipe\r\n"
     ]
    }
   ],
   "source": [
    "# Let's sanity check our adapter sequence by seeing\n",
    "# how many times it occurs in the first 100000 reads.\n",
    "\n",
    "ADAPTER=\"CTGTCTCTTATA\"\n",
    "\n",
    "NUM_LINES=400000  # 4 * num_reads, since each fastq entry is 4 lines\n",
    "\n",
    "zcat $(ls $FASTQ_DIR*R1* | head -n 1) | head -n $NUM_LINES | grep $ADAPTER | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\r\n",
      "\r\n",
      "gzip: stdout: Broken pipe\r\n"
     ]
    }
   ],
   "source": [
    "# Let's also check how often a permutation (rearrangement)\n",
    "# of the adapter sequence occurs:\n",
    "\n",
    "NOT_ADAPTER=\"CGTTCTTCTATA\"  # A permutation of the adapter sequence\n",
    "\n",
    "zcat $(ls $FASTQ_DIR*R1* | head -n 1) | head -n $NUM_LINES | grep $NOT_ADAPTER | wc -l"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Notice that the correct adapter sequence occurs *many* times more in the reads than a permutation of the adapter sequene — this is an important validation that we have the right sequence.\n",
    "\n",
    "Now, we'll trim the paired-end reads using a tool called `cutadapt`:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": []
    }
   ],
   "source": [
    "#create a directory to store the trimmed data \n",
    "export TRIMMED_DIR=\"$ANALYSIS_DIR/trimmed/\"\n",
    "[[ ! -d $TRIMMED_DIR ]] && mkdir -p \"$TRIMMED_DIR\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cutadapt -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /srv/scratch/training_camp/tc2016/ubuntu/analysis//trimmed/Ct_1_S22_R1.trimmed.fastq.gz -p /srv/scratch/training_camp/tc2016/ubuntu/analysis//trimmed/Ct_1_S22_R2.trimmed.fastq.gz /srv/scratch/training_camp/tc2016/ubuntu/data/fastq/Ct_1_S22_R1.fastq.gz /srv/scratch/training_camp/tc2016/ubuntu/data/fastq/Ct_1_S22_R2.fastq.gz\r\n",
      "This is cutadapt 1.10 with Python 3.5.2\r\n",
      "Command line parameters: -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /srv/scratch/training_camp/tc2016/ubuntu/analysis//trimmed/Ct_1_S22_R1.trimmed.fastq.gz -p /srv/scratch/training_camp/tc2016/ubuntu/analysis//trimmed/Ct_1_S22_R2.trimmed.fastq.gz /srv/scratch/training_camp/tc2016/ubuntu/data/fastq/Ct_1_S22_R1.fastq.gz /srv/scratch/training_camp/tc2016/ubuntu/data/fastq/Ct_1_S22_R2.fastq.gz\r\n",
      "Trimming 2 adapters with at most 20.0% errors in paired-end mode ...\r\n",
      "Finished in 734.28 s (187 us/read; 0.32 M reads/minute).\r\n",
      "\r\n",
      "=== Summary ===\r\n",
      "\r\n",
      "Total read pairs processed:          3,929,668\r\n",
      "  Read 1 with adapter:               1,440,399 (36.7%)\r\n",
      "  Read 2 with adapter:               1,431,456 (36.4%)\r\n",
      "Pairs that were too short:               3,958 (0.1%)\r\n",
      "Pairs written (passing filters):     3,925,710 (99.9%)\r\n",
      "\r\n",
      "Total basepairs processed:   597,309,536 bp\r\n",
      "  Read 1:   298,654,768 bp\r\n",
      "  Read 2:   298,654,768 bp\r\n",
      "Total written (filtered):    535,954,991 bp (89.7%)\r\n",
      "  Read 1:   267,912,314 bp\r\n",
      "  Read 2:   268,042,677 bp\r\n",
      "\r\n",
      "=== First read: Adapter 1 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 1440399 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 13.9%\r\n",
      "  C: 38.4%\r\n",
      "  G: 24.9%\r\n",
      "  T: 22.9%\r\n",
      "  none/other: 0.0%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t60048\t61401.1\t0\t60048\r\n",
      "4\t62875\t15350.3\t0\t37454 25421\r\n",
      "5\t66744\t3837.6\t1\t28100 38644\r\n",
      "6\t45078\t959.4\t1\t28383 16695\r\n",
      "7\t32197\t239.8\t1\t27942 4255\r\n",
      "8\t34230\t60.0\t1\t29341 4448 441\r\n",
      "9\t32762\t15.0\t1\t29858 702 2202\r\n",
      "10\t31349\t3.7\t2\t29316 581 1452\r\n",
      "11\t31110\t0.9\t2\t29459 567 1084\r\n",
      "12\t30841\t0.2\t2\t30008 422 411\r\n",
      "13\t31540\t0.2\t2\t30735 380 425\r\n",
      "14\t32351\t0.2\t2\t31526 420 405\r\n",
      "15\t33230\t0.2\t2\t32513 392 325\r\n",
      "16\t32113\t0.2\t2\t31336 338 439\r\n",
      "17\t33034\t0.2\t2\t32284 357 393\r\n",
      "18\t33171\t0.2\t2\t32425 376 370\r\n",
      "19\t34858\t0.2\t2\t33960 369 529\r\n",
      "20\t36150\t0.2\t2\t35350 377 423\r\n",
      "21\t35973\t0.2\t2\t35141 402 430\r\n",
      "22\t34587\t0.2\t2\t33850 332 405\r\n",
      "23\t33083\t0.2\t2\t32264 349 470\r\n",
      "24\t36252\t0.2\t2\t35529 354 369\r\n",
      "25\t40487\t0.2\t2\t39792 416 279\r\n",
      "26\t40627\t0.2\t2\t39839 410 378\r\n",
      "27\t36178\t0.2\t2\t35512 340 326\r\n",
      "28\t34699\t0.2\t2\t33999 345 355\r\n",
      "29\t37697\t0.2\t2\t36990 338 369\r\n",
      "30\t39488\t0.2\t2\t38789 391 308\r\n",
      "31\t39966\t0.2\t2\t39281 392 293\r\n",
      "32\t39153\t0.2\t2\t38450 368 335\r\n",
      "33\t36583\t0.2\t2\t35760 301 522\r\n",
      "34\t34380\t0.2\t2\t33681 273 426\r\n",
      "35\t42471\t0.2\t2\t41812 373 286\r\n",
      "36\t45430\t0.2\t2\t44768 344 318\r\n",
      "37\t32201\t0.2\t2\t31592 302 307\r\n",
      "38\t25370\t0.2\t2\t24887 204 279\r\n",
      "39\t19155\t0.2\t2\t18551 154 450\r\n",
      "40\t15009\t0.2\t2\t14520 163 326\r\n",
      "41\t8088\t0.2\t2\t7663 60 365\r\n",
      "42\t4682\t0.2\t2\t3932 65 685\r\n",
      "43\t2796\t0.2\t2\t2327 57 412\r\n",
      "44\t1886\t0.2\t2\t1479 42 365\r\n",
      "45\t2024\t0.2\t2\t1699 33 292\r\n",
      "46\t5100\t0.2\t2\t4637 50 413\r\n",
      "47\t3859\t0.2\t2\t3475 44 340\r\n",
      "48\t2076\t0.2\t2\t1776 56 244\r\n",
      "49\t1942\t0.2\t2\t1470 39 433\r\n",
      "50\t1340\t0.2\t2\t1036 31 273\r\n",
      "51\t898\t0.2\t2\t165 30 703\r\n",
      "52\t440\t0.2\t2\t74 34 332\r\n",
      "53\t973\t0.2\t2\t67 41 865\r\n",
      "54\t543\t0.2\t2\t83 28 432\r\n",
      "55\t452\t0.2\t2\t104 25 323\r\n",
      "56\t927\t0.2\t2\t184 25 718\r\n",
      "57\t517\t0.2\t2\t134 16 367\r\n",
      "58\t502\t0.2\t2\t77 21 404\r\n",
      "59\t548\t0.2\t2\t40 19 489\r\n",
      "60\t459\t0.2\t2\t26 8 425\r\n",
      "61\t565\t0.2\t2\t53 22 490\r\n",
      "62\t330\t0.2\t2\t25 11 294\r\n",
      "63\t805\t0.2\t2\t5 16 784\r\n",
      "64\t521\t0.2\t2\t7 10 504\r\n",
      "65\t391\t0.2\t2\t5 19 367\r\n",
      "66\t387\t0.2\t2\t4 4 379\r\n",
      "67\t302\t0.2\t2\t5 28 269\r\n",
      "68\t1524\t0.2\t2\t0 55 1469\r\n",
      "69\t350\t0.2\t2\t1 18 331\r\n",
      "70\t553\t0.2\t2\t0 6 547\r\n",
      "71\t165\t0.2\t2\t0 2 163\r\n",
      "72\t216\t0.2\t2\t0 1 215\r\n",
      "73\t627\t0.2\t2\t0 18 609\r\n",
      "74\t585\t0.2\t2\t0 11 574\r\n",
      "75\t276\t0.2\t2\t0 15 261\r\n",
      "76\t280\t0.2\t2\t0 3 277\r\n",
      "\r\n",
      "=== Second read: Adapter 2 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 1431456 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 14.3%\r\n",
      "  C: 37.3%\r\n",
      "  G: 25.3%\r\n",
      "  T: 23.1%\r\n",
      "  none/other: 0.0%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t59997\t61401.1\t0\t59997\r\n",
      "4\t61914\t15350.3\t0\t36805 25109\r\n",
      "5\t65122\t3837.6\t1\t26581 38541\r\n",
      "6\t44272\t959.4\t1\t27306 16966\r\n",
      "7\t31903\t239.8\t1\t26720 5183\r\n",
      "8\t33870\t60.0\t1\t28852 4541 477\r\n",
      "9\t32485\t15.0\t1\t29411 907 2167\r\n",
      "10\t31131\t3.7\t2\t28322 1246 1563\r\n",
      "11\t30987\t0.9\t2\t28879 846 1262\r\n",
      "12\t30664\t0.2\t2\t29447 695 522\r\n",
      "13\t31297\t0.2\t2\t30180 634 483\r\n",
      "14\t32153\t0.2\t2\t31041 631 481\r\n",
      "15\t33049\t0.2\t2\t32046 595 408\r\n",
      "16\t31859\t0.2\t2\t30838 601 420\r\n",
      "17\t32913\t0.2\t2\t31803 631 479\r\n",
      "18\t32986\t0.2\t2\t31936 602 448\r\n",
      "19\t34730\t0.2\t2\t33481 597 652\r\n",
      "20\t35933\t0.2\t2\t34824 633 476\r\n",
      "21\t35777\t0.2\t2\t34615 640 522\r\n",
      "22\t34453\t0.2\t2\t33364 578 511\r\n",
      "23\t32897\t0.2\t2\t31836 526 535\r\n",
      "24\t36075\t0.2\t2\t35065 579 431\r\n",
      "25\t40356\t0.2\t2\t39228 719 409\r\n",
      "26\t40472\t0.2\t2\t39333 655 484\r\n",
      "27\t36030\t0.2\t2\t35076 549 405\r\n",
      "28\t34647\t0.2\t2\t33661 510 476\r\n",
      "29\t37591\t0.2\t2\t36539 574 478\r\n",
      "30\t39356\t0.2\t2\t38328 601 427\r\n",
      "31\t39833\t0.2\t2\t38823 617 393\r\n",
      "32\t39016\t0.2\t2\t38027 591 398\r\n",
      "33\t36497\t0.2\t2\t35329 548 620\r\n",
      "34\t34303\t0.2\t2\t33316 485 502\r\n",
      "35\t42311\t0.2\t2\t41347 590 374\r\n",
      "36\t45323\t0.2\t2\t44237 608 478\r\n",
      "37\t32107\t0.2\t2\t31260 450 397\r\n",
      "38\t25294\t0.2\t2\t24591 388 315\r\n",
      "39\t19059\t0.2\t2\t18346 266 447\r\n",
      "40\t15014\t0.2\t2\t14393 219 402\r\n",
      "41\t8101\t0.2\t2\t7581 118 402\r\n",
      "42\t4703\t0.2\t2\t3884 98 721\r\n",
      "43\t2790\t0.2\t2\t2306 67 417\r\n",
      "44\t1872\t0.2\t2\t1472 40 360\r\n",
      "45\t2009\t0.2\t2\t1676 37 296\r\n",
      "46\t5070\t0.2\t2\t4583 74 413\r\n",
      "47\t3842\t0.2\t2\t3434 48 360\r\n",
      "48\t2059\t0.2\t2\t1746 73 240\r\n",
      "49\t1990\t0.2\t2\t1459 42 489\r\n",
      "50\t1356\t0.2\t2\t1014 48 294\r\n",
      "51\t847\t0.2\t2\t151 32 664\r\n",
      "52\t409\t0.2\t2\t76 14 319\r\n",
      "53\t876\t0.2\t2\t64 55 757\r\n",
      "54\t544\t0.2\t2\t79 15 450\r\n",
      "55\t412\t0.2\t2\t94 20 298\r\n",
      "56\t922\t0.2\t2\t171 31 720\r\n",
      "57\t506\t0.2\t2\t124 13 369\r\n",
      "58\t553\t0.2\t2\t72 26 455\r\n",
      "59\t612\t0.2\t2\t32 36 544\r\n",
      "60\t445\t0.2\t2\t19 13 413\r\n",
      "61\t574\t0.2\t2\t45 17 512\r\n",
      "62\t320\t0.2\t2\t17 15 288\r\n",
      "63\t812\t0.2\t2\t4 19 789\r\n",
      "64\t505\t0.2\t2\t5 9 491\r\n",
      "65\t390\t0.2\t2\t4 18 368\r\n",
      "66\t400\t0.2\t2\t2 10 388\r\n",
      "67\t280\t0.2\t2\t5 26 249\r\n",
      "68\t1512\t0.2\t2\t0 66 1446\r\n",
      "69\t339\t0.2\t2\t1 20 318\r\n",
      "70\t595\t0.2\t2\t0 9 586\r\n",
      "71\t160\t0.2\t2\t0 3 157\r\n",
      "72\t165\t0.2\t2\t0 7 158\r\n",
      "73\t667\t0.2\t2\t0 27 640\r\n",
      "74\t584\t0.2\t2\t0 7 577\r\n",
      "75\t262\t0.2\t2\t0 18 244\r\n",
      "76\t297\t0.2\t2\t0 5 292\r\n",
      "\r\n",
      "cutadapt -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /srv/scratch/training_camp/tc2016/ubuntu/analysis//trimmed/Ct_2_S23_R1.trimmed.fastq.gz -p /srv/scratch/training_camp/tc2016/ubuntu/analysis//trimmed/Ct_2_S23_R2.trimmed.fastq.gz /srv/scratch/training_camp/tc2016/ubuntu/data/fastq/Ct_2_S23_R1.fastq.gz /srv/scratch/training_camp/tc2016/ubuntu/data/fastq/Ct_2_S23_R2.fastq.gz\r\n",
      "This is cutadapt 1.10 with Python 3.5.2\r\n",
      "Command line parameters: -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /srv/scratch/training_camp/tc2016/ubuntu/analysis//trimmed/Ct_2_S23_R1.trimmed.fastq.gz -p /srv/scratch/training_camp/tc2016/ubuntu/analysis//trimmed/Ct_2_S23_R2.trimmed.fastq.gz /srv/scratch/training_camp/tc2016/ubuntu/data/fastq/Ct_2_S23_R1.fastq.gz /srv/scratch/training_camp/tc2016/ubuntu/data/fastq/Ct_2_S23_R2.fastq.gz\r\n",
      "Trimming 2 adapters with at most 20.0% errors in paired-end mode ...\r\n",
      "Finished in 534.14 s (176 us/read; 0.34 M reads/minute).\r\n",
      "\r\n",
      "=== Summary ===\r\n",
      "\r\n",
      "Total read pairs processed:          3,028,087\r\n",
      "  Read 1 with adapter:               1,233,731 (40.7%)\r\n",
      "  Read 2 with adapter:               1,226,357 (40.5%)\r\n",
      "Pairs that were too short:               2,777 (0.1%)\r\n",
      "Pairs written (passing filters):     3,025,310 (99.9%)\r\n",
      "\r\n",
      "Total basepairs processed:   460,269,224 bp\r\n",
      "  Read 1:   230,134,612 bp\r\n",
      "  Read 2:   230,134,612 bp\r\n",
      "Total written (filtered):    406,641,404 bp (88.3%)\r\n",
      "  Read 1:   203,266,521 bp\r\n",
      "  Read 2:   203,374,883 bp\r\n",
      "\r\n",
      "=== First read: Adapter 1 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 1233731 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 13.8%\r\n",
      "  C: 37.8%\r\n",
      "  G: 24.7%\r\n",
      "  T: 23.6%\r\n",
      "  none/other: 0.0%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t44983\t47313.9\t0\t44983\r\n",
      "4\t47855\t11828.5\t0\t29513 18342\r\n",
      "5\t50904\t2957.1\t1\t22883 28021\r\n",
      "6\t35926\t739.3\t1\t23480 12446\r\n",
      "7\t26276\t184.8\t1\t23126 3150\r\n",
      "8\t28315\t46.2\t1\t24262 3718 335\r\n",
      "9\t27232\t11.6\t1\t25115 550 1567\r\n",
      "10\t25801\t2.9\t2\t24294 434 1073\r\n",
      "11\t25613\t0.7\t2\t24341 447 825\r\n",
      "12\t25295\t0.2\t2\t24641 334 320\r\n",
      "13\t26149\t0.2\t2\t25547 317 285\r\n",
      "14\t27174\t0.2\t2\t26531 338 305\r\n",
      "15\t28334\t0.2\t2\t27777 319 238\r\n",
      "16\t28287\t0.2\t2\t27611 351 325\r\n",
      "17\t28763\t0.2\t2\t28159 314 290\r\n",
      "18\t28708\t0.2\t2\t28091 324 293\r\n",
      "19\t30524\t0.2\t2\t29733 345 446\r\n",
      "20\t31602\t0.2\t2\t31021 321 260\r\n",
      "21\t31048\t0.2\t2\t30390 336 322\r\n",
      "22\t30376\t0.2\t2\t29734 330 312\r\n",
      "23\t28883\t0.2\t2\t28218 308 357\r\n",
      "24\t32716\t0.2\t2\t32091 318 307\r\n",
      "25\t38225\t0.2\t2\t37625 376 224\r\n",
      "26\t38226\t0.2\t2\t37601 370 255\r\n",
      "27\t32869\t0.2\t2\t32314 284 271\r\n",
      "28\t31710\t0.2\t2\t31146 285 279\r\n",
      "29\t34280\t0.2\t2\t33678 307 295\r\n",
      "30\t36535\t0.2\t2\t35977 302 256\r\n",
      "31\t36426\t0.2\t2\t35891 311 224\r\n",
      "32\t35547\t0.2\t2\t34934 365 248\r\n",
      "33\t32430\t0.2\t2\t31757 314 359\r\n",
      "34\t30675\t0.2\t2\t30112 257 306\r\n",
      "35\t39883\t0.2\t2\t39354 324 205\r\n",
      "36\t41537\t0.2\t2\t40930 343 264\r\n",
      "37\t27899\t0.2\t2\t27384 259 256\r\n",
      "38\t20592\t0.2\t2\t20203 186 203\r\n",
      "39\t15419\t0.2\t2\t14974 148 297\r\n",
      "40\t11813\t0.2\t2\t11449 107 257\r\n",
      "41\t6591\t0.2\t2\t6264 75 252\r\n",
      "42\t3611\t0.2\t2\t3065 54 492\r\n",
      "43\t2364\t0.2\t2\t2051 32 281\r\n",
      "44\t1677\t0.2\t2\t1396 19 262\r\n",
      "45\t1908\t0.2\t2\t1669 37 202\r\n",
      "46\t5163\t0.2\t2\t4831 60 272\r\n",
      "47\t3560\t0.2\t2\t3264 31 265\r\n",
      "48\t1861\t0.2\t2\t1665 45 151\r\n",
      "49\t1582\t0.2\t2\t1230 24 328\r\n",
      "50\t1038\t0.2\t2\t846 22 170\r\n",
      "51\t575\t0.2\t2\t109 15 451\r\n",
      "52\t301\t0.2\t2\t66 9 226\r\n",
      "53\t613\t0.2\t2\t63 33 517\r\n",
      "54\t352\t0.2\t2\t69 12 271\r\n",
      "55\t325\t0.2\t2\t56 16 253\r\n",
      "56\t541\t0.2\t2\t112 11 418\r\n",
      "57\t366\t0.2\t2\t97 6 263\r\n",
      "58\t379\t0.2\t2\t42 8 329\r\n",
      "59\t416\t0.2\t2\t28 22 366\r\n",
      "60\t309\t0.2\t2\t22 6 281\r\n",
      "61\t338\t0.2\t2\t24 11 303\r\n",
      "62\t217\t0.2\t2\t22 7 188\r\n",
      "63\t506\t0.2\t2\t11 13 482\r\n",
      "64\t387\t0.2\t2\t7 11 369\r\n",
      "65\t252\t0.2\t2\t0 18 234\r\n",
      "66\t305\t0.2\t2\t2 3 300\r\n",
      "67\t192\t0.2\t2\t4 11 177\r\n",
      "68\t1047\t0.2\t2\t0 34 1013\r\n",
      "69\t302\t0.2\t2\t0 11 291\r\n",
      "70\t378\t0.2\t2\t0 4 374\r\n",
      "71\t98\t0.2\t2\t0 0 98\r\n",
      "72\t134\t0.2\t2\t0 1 133\r\n",
      "73\t447\t0.2\t2\t0 10 437\r\n",
      "74\t399\t0.2\t2\t0 9 390\r\n",
      "75\t184\t0.2\t2\t0 14 170\r\n",
      "76\t183\t0.2\t2\t0 4 179\r\n",
      "\r\n",
      "=== Second read: Adapter 2 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 1226357 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 14.2%\r\n",
      "  C: 36.9%\r\n",
      "  G: 25.0%\r\n",
      "  T: 23.9%\r\n",
      "  none/other: 0.0%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t44408\t47313.9\t0\t44408\r\n",
      "4\t47241\t11828.5\t0\t29174 18067\r\n",
      "5\t49928\t2957.1\t1\t21742 28186\r\n",
      "6\t35187\t739.3\t1\t22731 12456\r\n",
      "7\t26026\t184.8\t1\t22126 3900\r\n",
      "8\t28018\t46.2\t1\t23892 3755 371\r\n",
      "9\t27078\t11.6\t1\t24774 729 1575\r\n",
      "10\t25673\t2.9\t2\t23734 766 1173\r\n",
      "11\t25441\t0.7\t2\t23876 675 890\r\n",
      "12\t25126\t0.2\t2\t24249 497 380\r\n",
      "13\t25983\t0.2\t2\t25197 477 309\r\n",
      "14\t26993\t0.2\t2\t26165 503 325\r\n",
      "15\t28219\t0.2\t2\t27415 509 295\r\n",
      "16\t28131\t0.2\t2\t27281 486 364\r\n",
      "17\t28613\t0.2\t2\t27775 467 371\r\n",
      "18\t28537\t0.2\t2\t27757 468 312\r\n",
      "19\t30381\t0.2\t2\t29393 521 467\r\n",
      "20\t31463\t0.2\t2\t30649 504 310\r\n",
      "21\t30868\t0.2\t2\t30051 458 359\r\n",
      "22\t30233\t0.2\t2\t29364 518 351\r\n",
      "23\t28785\t0.2\t2\t27897 454 434\r\n",
      "24\t32551\t0.2\t2\t31714 486 351\r\n",
      "25\t38097\t0.2\t2\t37250 533 314\r\n",
      "26\t38088\t0.2\t2\t37231 527 330\r\n",
      "27\t32769\t0.2\t2\t32006 455 308\r\n",
      "28\t31601\t0.2\t2\t30855 419 327\r\n",
      "29\t34159\t0.2\t2\t33362 449 348\r\n",
      "30\t36412\t0.2\t2\t35612 504 296\r\n",
      "31\t36342\t0.2\t2\t35583 451 308\r\n",
      "32\t35401\t0.2\t2\t34609 499 293\r\n",
      "33\t32366\t0.2\t2\t31435 477 454\r\n",
      "34\t30580\t0.2\t2\t29830 427 323\r\n",
      "35\t39759\t0.2\t2\t38927 555 277\r\n",
      "36\t41413\t0.2\t2\t40547 533 333\r\n",
      "37\t27826\t0.2\t2\t27145 382 299\r\n",
      "38\t20536\t0.2\t2\t19992 279 265\r\n",
      "39\t15397\t0.2\t2\t14835 211 351\r\n",
      "40\t11799\t0.2\t2\t11312 184 303\r\n",
      "41\t6606\t0.2\t2\t6221 83 302\r\n",
      "42\t3618\t0.2\t2\t3013 67 538\r\n",
      "43\t2396\t0.2\t2\t2042 43 311\r\n",
      "44\t1671\t0.2\t2\t1376 40 255\r\n",
      "45\t1930\t0.2\t2\t1663 22 245\r\n",
      "46\t5146\t0.2\t2\t4798 81 267\r\n",
      "47\t3521\t0.2\t2\t3233 44 244\r\n",
      "48\t1877\t0.2\t2\t1636 65 176\r\n",
      "49\t1588\t0.2\t2\t1207 39 342\r\n",
      "50\t1046\t0.2\t2\t840 28 178\r\n",
      "51\t534\t0.2\t2\t103 15 416\r\n",
      "52\t307\t0.2\t2\t63 13 231\r\n",
      "53\t595\t0.2\t2\t58 34 503\r\n",
      "54\t365\t0.2\t2\t65 10 290\r\n",
      "55\t284\t0.2\t2\t51 19 214\r\n",
      "56\t551\t0.2\t2\t102 17 432\r\n",
      "57\t377\t0.2\t2\t93 10 274\r\n",
      "58\t380\t0.2\t2\t37 16 327\r\n",
      "59\t390\t0.2\t2\t26 22 342\r\n",
      "60\t265\t0.2\t2\t20 7 238\r\n",
      "61\t378\t0.2\t2\t18 17 343\r\n",
      "62\t236\t0.2\t2\t19 9 208\r\n",
      "63\t500\t0.2\t2\t9 13 478\r\n",
      "64\t365\t0.2\t2\t6 10 349\r\n",
      "65\t253\t0.2\t2\t0 19 234\r\n",
      "66\t315\t0.2\t2\t2 4 309\r\n",
      "67\t185\t0.2\t2\t4 17 164\r\n",
      "68\t1106\t0.2\t2\t0 51 1055\r\n",
      "69\t243\t0.2\t2\t0 20 223\r\n",
      "70\t369\t0.2\t2\t0 5 364\r\n",
      "71\t101\t0.2\t2\t0 0 101\r\n",
      "72\t132\t0.2\t2\t0 3 129\r\n",
      "73\t461\t0.2\t2\t0 12 449\r\n",
      "74\t435\t0.2\t2\t0 4 431\r\n",
      "75\t215\t0.2\t2\t0 18 197\r\n",
      "76\t188\t0.2\t2\t0 5 183\r\n",
      "\r\n",
      "cutadapt -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /srv/scratch/training_camp/tc2016/ubuntu/analysis//trimmed/Ct_300_S3_R1.trimmed.fastq.gz -p /srv/scratch/training_camp/tc2016/ubuntu/analysis//trimmed/Ct_300_S3_R2.trimmed.fastq.gz /srv/scratch/training_camp/tc2016/ubuntu/data/fastq/Ct_300_S3_R1.fastq.gz /srv/scratch/training_camp/tc2016/ubuntu/data/fastq/Ct_300_S3_R2.fastq.gz\r\n",
      "This is cutadapt 1.10 with Python 3.5.2\r\n",
      "Command line parameters: -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /srv/scratch/training_camp/tc2016/ubuntu/analysis//trimmed/Ct_300_S3_R1.trimmed.fastq.gz -p /srv/scratch/training_camp/tc2016/ubuntu/analysis//trimmed/Ct_300_S3_R2.trimmed.fastq.gz /srv/scratch/training_camp/tc2016/ubuntu/data/fastq/Ct_300_S3_R1.fastq.gz /srv/scratch/training_camp/tc2016/ubuntu/data/fastq/Ct_300_S3_R2.fastq.gz\r\n",
      "Trimming 2 adapters with at most 20.0% errors in paired-end mode ...\r\n",
      "Finished in 948.66 s (180 us/read; 0.33 M reads/minute).\r\n",
      "\r\n",
      "=== Summary ===\r\n",
      "\r\n",
      "Total read pairs processed:          5,271,099\r\n",
      "  Read 1 with adapter:               1,921,344 (36.5%)\r\n",
      "  Read 2 with adapter:               1,911,038 (36.3%)\r\n",
      "Pairs that were too short:               5,445 (0.1%)\r\n",
      "Pairs written (passing filters):     5,265,654 (99.9%)\r\n",
      "\r\n",
      "Total basepairs processed:   801,207,048 bp\r\n",
      "  Read 1:   400,603,524 bp\r\n",
      "  Read 2:   400,603,524 bp\r\n",
      "Total written (filtered):    716,533,393 bp (89.4%)\r\n",
      "  Read 1:   358,180,437 bp\r\n",
      "  Read 2:   358,352,956 bp\r\n",
      "\r\n",
      "=== First read: Adapter 1 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 1921344 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 14.3%\r\n",
      "  C: 37.3%\r\n",
      "  G: 24.3%\r\n",
      "  T: 24.0%\r\n",
      "  none/other: 0.0%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t75301\t82360.9\t0\t75301\r\n",
      "4\t80025\t20590.2\t0\t46890 33135\r\n",
      "5\t85265\t5147.6\t1\t34215 51050\r\n",
      "6\t57407\t1286.9\t1\t35173 22234\r\n",
      "7\t39865\t321.7\t1\t34394 5471\r\n",
      "8\t43117\t80.4\t1\t35914 6597 606\r\n",
      "9\t40483\t20.1\t1\t36636 996 2851\r\n",
      "10\t38684\t5.0\t2\t35969 682 2033\r\n",
      "11\t39251\t1.3\t2\t36940 781 1530\r\n",
      "12\t38616\t0.3\t2\t37571 560 485\r\n",
      "13\t39370\t0.3\t2\t38238 543 589\r\n",
      "14\t40790\t0.3\t2\t39779 532 479\r\n",
      "15\t41676\t0.3\t2\t40698 504 474\r\n",
      "16\t40273\t0.3\t2\t39221 502 550\r\n",
      "17\t42581\t0.3\t2\t41556 516 509\r\n",
      "18\t42031\t0.3\t2\t40986 519 526\r\n",
      "19\t44960\t0.3\t2\t43686 492 782\r\n",
      "20\t45537\t0.3\t2\t44550 512 475\r\n",
      "21\t45454\t0.3\t2\t44349 605 500\r\n",
      "22\t44943\t0.3\t2\t43860 517 566\r\n",
      "23\t43194\t0.3\t2\t42152 447 595\r\n",
      "24\t48695\t0.3\t2\t47656 522 517\r\n",
      "25\t55982\t0.3\t2\t54978 575 429\r\n",
      "26\t54642\t0.3\t2\t53585 589 468\r\n",
      "27\t48188\t0.3\t2\t47264 508 416\r\n",
      "28\t47243\t0.3\t2\t46230 504 509\r\n",
      "29\t51220\t0.3\t2\t50181 540 499\r\n",
      "30\t54571\t0.3\t2\t53551 577 443\r\n",
      "31\t53394\t0.3\t2\t52421 534 439\r\n",
      "32\t53168\t0.3\t2\t52163 557 448\r\n",
      "33\t50372\t0.3\t2\t49197 478 697\r\n",
      "34\t49725\t0.3\t2\t48678 500 547\r\n",
      "35\t65234\t0.3\t2\t64253 616 365\r\n",
      "36\t69134\t0.3\t2\t67977 683 474\r\n",
      "37\t48468\t0.3\t2\t47528 498 442\r\n",
      "38\t38438\t0.3\t2\t37702 367 369\r\n",
      "39\t29226\t0.3\t2\t28364 263 599\r\n",
      "40\t22050\t0.3\t2\t21395 197 458\r\n",
      "41\t11944\t0.3\t2\t11336 123 485\r\n",
      "42\t6593\t0.3\t2\t5570 86 937\r\n",
      "43\t4048\t0.3\t2\t3515 42 491\r\n",
      "44\t2828\t0.3\t2\t2307 34 487\r\n",
      "45\t3310\t0.3\t2\t2919 49 342\r\n",
      "46\t8838\t0.3\t2\t8174 91 573\r\n",
      "47\t6358\t0.3\t2\t5769 92 497\r\n",
      "48\t3886\t0.3\t2\t3490 68 328\r\n",
      "49\t3287\t0.3\t2\t2663 45 579\r\n",
      "50\t2239\t0.3\t2\t1846 33 360\r\n",
      "51\t1169\t0.3\t2\t176 31 962\r\n",
      "52\t551\t0.3\t2\t79 18 454\r\n",
      "53\t1233\t0.3\t2\t51 34 1148\r\n",
      "54\t768\t0.3\t2\t75 19 674\r\n",
      "55\t485\t0.3\t2\t73 16 396\r\n",
      "56\t1272\t0.3\t2\t142 29 1101\r\n",
      "57\t683\t0.3\t2\t153 14 516\r\n",
      "58\t734\t0.3\t2\t50 21 663\r\n",
      "59\t718\t0.3\t2\t41 22 655\r\n",
      "60\t621\t0.3\t2\t21 10 590\r\n",
      "61\t790\t0.3\t2\t22 17 751\r\n",
      "62\t409\t0.3\t2\t13 7 389\r\n",
      "63\t1169\t0.3\t2\t7 19 1143\r\n",
      "64\t756\t0.3\t2\t2 13 741\r\n",
      "65\t574\t0.3\t2\t2 25 547\r\n",
      "66\t571\t0.3\t2\t1 8 562\r\n",
      "67\t368\t0.3\t2\t7 34 327\r\n",
      "68\t2295\t0.3\t2\t0 87 2208\r\n",
      "69\t562\t0.3\t2\t0 37 525\r\n",
      "70\t821\t0.3\t2\t0 8 813\r\n",
      "71\t229\t0.3\t2\t0 1 228\r\n",
      "72\t279\t0.3\t2\t0 7 272\r\n",
      "73\t840\t0.3\t2\t0 34 806\r\n",
      "74\t745\t0.3\t2\t0 7 738\r\n",
      "75\t444\t0.3\t2\t0 33 411\r\n",
      "76\t354\t0.3\t2\t1 8 345\r\n",
      "\r\n",
      "=== Second read: Adapter 2 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 1911038 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 14.7%\r\n",
      "  C: 35.9%\r\n",
      "  G: 24.8%\r\n",
      "  T: 24.5%\r\n",
      "  none/other: 0.0%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t75031\t82360.9\t0\t75031\r\n",
      "4\t78717\t20590.2\t0\t46238 32479\r\n",
      "5\t83544\t5147.6\t1\t32965 50579\r\n",
      "6\t56731\t1286.9\t1\t34428 22303\r\n",
      "7\t39528\t321.7\t1\t33282 6246\r\n",
      "8\t42373\t80.4\t1\t35550 6206 617\r\n",
      "9\t40423\t20.1\t1\t36395 983 3045\r\n",
      "10\t38513\t5.0\t2\t35420 996 2097\r\n",
      "11\t39024\t1.3\t2\t36566 917 1541\r\n",
      "12\t38463\t0.3\t2\t37221 671 571\r\n",
      "13\t39150\t0.3\t2\t37931 630 589\r\n",
      "14\t40658\t0.3\t2\t39391 687 580\r\n",
      "15\t41456\t0.3\t2\t40315 639 502\r\n",
      "16\t40115\t0.3\t2\t38891 623 601\r\n",
      "17\t42372\t0.3\t2\t41166 647 559\r\n",
      "18\t41849\t0.3\t2\t40631 632 586\r\n",
      "19\t44716\t0.3\t2\t43290 650 776\r\n",
      "20\t45418\t0.3\t2\t44171 682 565\r\n",
      "21\t45325\t0.3\t2\t44070 681 574\r\n",
      "22\t44766\t0.3\t2\t43514 610 642\r\n",
      "23\t43050\t0.3\t2\t41830 591 629\r\n",
      "24\t48538\t0.3\t2\t47323 647 568\r\n",
      "25\t55825\t0.3\t2\t54556 760 509\r\n",
      "26\t54481\t0.3\t2\t53232 718 531\r\n",
      "27\t48056\t0.3\t2\t46908 648 500\r\n",
      "28\t47103\t0.3\t2\t45976 586 541\r\n",
      "29\t51081\t0.3\t2\t49912 622 547\r\n",
      "30\t54424\t0.3\t2\t53206 726 492\r\n",
      "31\t53235\t0.3\t2\t52049 679 507\r\n",
      "32\t52993\t0.3\t2\t51778 704 511\r\n",
      "33\t50273\t0.3\t2\t48802 676 795\r\n",
      "34\t49649\t0.3\t2\t48352 649 648\r\n",
      "35\t65053\t0.3\t2\t63826 788 439\r\n",
      "36\t68980\t0.3\t2\t67590 806 584\r\n",
      "37\t48382\t0.3\t2\t47231 640 511\r\n",
      "38\t38383\t0.3\t2\t37460 456 467\r\n",
      "39\t29147\t0.3\t2\t28152 359 636\r\n",
      "40\t22009\t0.3\t2\t21252 267 490\r\n",
      "41\t11943\t0.3\t2\t11231 172 540\r\n",
      "42\t6549\t0.3\t2\t5535 90 924\r\n",
      "43\t4099\t0.3\t2\t3476 62 561\r\n",
      "44\t2803\t0.3\t2\t2297 51 455\r\n",
      "45\t3360\t0.3\t2\t2902 60 398\r\n",
      "46\t8764\t0.3\t2\t8127 122 515\r\n",
      "47\t6273\t0.3\t2\t5742 80 451\r\n",
      "48\t3888\t0.3\t2\t3457 92 339\r\n",
      "49\t3289\t0.3\t2\t2643 56 590\r\n",
      "50\t2233\t0.3\t2\t1834 37 362\r\n",
      "51\t1080\t0.3\t2\t171 27 882\r\n",
      "52\t536\t0.3\t2\t74 20 442\r\n",
      "53\t1185\t0.3\t2\t49 42 1094\r\n",
      "54\t720\t0.3\t2\t69 16 635\r\n",
      "55\t477\t0.3\t2\t82 13 382\r\n",
      "56\t1249\t0.3\t2\t136 36 1077\r\n",
      "57\t647\t0.3\t2\t148 10 489\r\n",
      "58\t687\t0.3\t2\t48 23 616\r\n",
      "59\t635\t0.3\t2\t38 27 570\r\n",
      "60\t618\t0.3\t2\t16 18 584\r\n",
      "61\t793\t0.3\t2\t18 23 752\r\n",
      "62\t444\t0.3\t2\t11 13 420\r\n",
      "63\t1175\t0.3\t2\t5 25 1145\r\n",
      "64\t681\t0.3\t2\t1 12 668\r\n",
      "65\t505\t0.3\t2\t0 23 482\r\n",
      "66\t525\t0.3\t2\t1 11 513\r\n",
      "67\t409\t0.3\t2\t6 29 374\r\n",
      "68\t2202\t0.3\t2\t0 61 2141\r\n",
      "69\t531\t0.3\t2\t0 29 502\r\n",
      "70\t921\t0.3\t2\t0 13 908\r\n",
      "71\t197\t0.3\t2\t0 1 196\r\n",
      "72\t271\t0.3\t2\t0 2 269\r\n",
      "73\t851\t0.3\t2\t0 45 806\r\n",
      "74\t828\t0.3\t2\t0 5 823\r\n",
      "75\t439\t0.3\t2\t0 34 405\r\n",
      "76\t397\t0.3\t2\t1 5 391\r\n",
      "\r\n",
      "cutadapt -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /srv/scratch/training_camp/tc2016/ubuntu/analysis//trimmed/Ct_3_S24_R1.trimmed.fastq.gz -p /srv/scratch/training_camp/tc2016/ubuntu/analysis//trimmed/Ct_3_S24_R2.trimmed.fastq.gz /srv/scratch/training_camp/tc2016/ubuntu/data/fastq/Ct_3_S24_R1.fastq.gz /srv/scratch/training_camp/tc2016/ubuntu/data/fastq/Ct_3_S24_R2.fastq.gz\r\n",
      "This is cutadapt 1.10 with Python 3.5.2\r\n",
      "Command line parameters: -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /srv/scratch/training_camp/tc2016/ubuntu/analysis//trimmed/Ct_3_S24_R1.trimmed.fastq.gz -p /srv/scratch/training_camp/tc2016/ubuntu/analysis//trimmed/Ct_3_S24_R2.trimmed.fastq.gz /srv/scratch/training_camp/tc2016/ubuntu/data/fastq/Ct_3_S24_R1.fastq.gz /srv/scratch/training_camp/tc2016/ubuntu/data/fastq/Ct_3_S24_R2.fastq.gz\r\n",
      "Trimming 2 adapters with at most 20.0% errors in paired-end mode ...\r\n",
      "Finished in 487.68 s (184 us/read; 0.33 M reads/minute).\r\n",
      "\r\n",
      "=== Summary ===\r\n",
      "\r\n",
      "Total read pairs processed:          2,648,492\r\n",
      "  Read 1 with adapter:                 875,320 (33.0%)\r\n",
      "  Read 2 with adapter:                 869,705 (32.8%)\r\n",
      "Pairs that were too short:               2,586 (0.1%)\r\n",
      "Pairs written (passing filters):     2,645,906 (99.9%)\r\n",
      "\r\n",
      "Total basepairs processed:   402,570,784 bp\r\n",
      "  Read 1:   201,285,392 bp\r\n",
      "  Read 2:   201,285,392 bp\r\n",
      "Total written (filtered):    366,393,470 bp (91.0%)\r\n",
      "  Read 1:   183,169,459 bp\r\n",
      "  Read 2:   183,224,011 bp\r\n",
      "\r\n",
      "=== First read: Adapter 1 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 875320 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 13.9%\r\n",
      "  C: 38.0%\r\n",
      "  G: 25.1%\r\n",
      "  T: 23.0%\r\n",
      "  none/other: 0.0%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t40306\t41382.7\t0\t40306\r\n",
      "4\t43071\t10345.7\t0\t24437 18634\r\n",
      "5\t45564\t2586.4\t1\t18133 27431\r\n",
      "6\t29882\t646.6\t1\t18072 11810\r\n",
      "7\t20753\t161.7\t1\t17829 2924\r\n",
      "8\t21901\t40.4\t1\t18068 3543 290\r\n",
      "9\t20357\t10.1\t1\t18321 454 1582\r\n",
      "10\t19336\t2.5\t2\t17992 387 957\r\n",
      "11\t19380\t0.6\t2\t18252 336 792\r\n",
      "12\t18753\t0.2\t2\t18249 258 246\r\n",
      "13\t19445\t0.2\t2\t18925 254 266\r\n",
      "14\t19517\t0.2\t2\t19027 238 252\r\n",
      "15\t20112\t0.2\t2\t19649 252 211\r\n",
      "16\t19773\t0.2\t2\t19288 211 274\r\n",
      "17\t19908\t0.2\t2\t19414 230 264\r\n",
      "18\t20117\t0.2\t2\t19640 227 250\r\n",
      "19\t21030\t0.2\t2\t20464 218 348\r\n",
      "20\t21553\t0.2\t2\t21047 245 261\r\n",
      "21\t21370\t0.2\t2\t20861 251 258\r\n",
      "22\t20711\t0.2\t2\t20184 225 302\r\n",
      "23\t19974\t0.2\t2\t19455 196 323\r\n",
      "24\t21704\t0.2\t2\t21223 203 278\r\n",
      "25\t24505\t0.2\t2\t24097 216 192\r\n",
      "26\t24575\t0.2\t2\t24130 210 235\r\n",
      "27\t21427\t0.2\t2\t21012 200 215\r\n",
      "28\t20510\t0.2\t2\t20089 174 247\r\n",
      "29\t21445\t0.2\t2\t21011 179 255\r\n",
      "30\t22978\t0.2\t2\t22587 167 224\r\n",
      "31\t23330\t0.2\t2\t22939 216 175\r\n",
      "32\t22871\t0.2\t2\t22460 221 190\r\n",
      "33\t21019\t0.2\t2\t20464 182 373\r\n",
      "34\t19972\t0.2\t2\t19533 148 291\r\n",
      "35\t24949\t0.2\t2\t24578 188 183\r\n",
      "36\t26251\t0.2\t2\t25829 207 215\r\n",
      "37\t18187\t0.2\t2\t17815 168 204\r\n",
      "38\t13719\t0.2\t2\t13406 110 203\r\n",
      "39\t10446\t0.2\t2\t10059 108 279\r\n",
      "40\t8083\t0.2\t2\t7797 67 219\r\n",
      "41\t4491\t0.2\t2\t4191 58 242\r\n",
      "42\t2589\t0.2\t2\t2049 45 495\r\n",
      "43\t1531\t0.2\t2\t1237 20 274\r\n",
      "44\t971\t0.2\t2\t736 21 214\r\n",
      "45\t1058\t0.2\t2\t834 17 207\r\n",
      "46\t2605\t0.2\t2\t2323 31 251\r\n",
      "47\t1884\t0.2\t2\t1633 24 227\r\n",
      "48\t1005\t0.2\t2\t805 35 165\r\n",
      "49\t952\t0.2\t2\t611 17 324\r\n",
      "50\t666\t0.2\t2\t489 12 165\r\n",
      "51\t517\t0.2\t2\t65 19 433\r\n",
      "52\t288\t0.2\t2\t47 15 226\r\n",
      "53\t574\t0.2\t2\t42 33 499\r\n",
      "54\t318\t0.2\t2\t45 10 263\r\n",
      "55\t262\t0.2\t2\t60 7 195\r\n",
      "56\t482\t0.2\t2\t80 23 379\r\n",
      "57\t323\t0.2\t2\t70 10 243\r\n",
      "58\t364\t0.2\t2\t30 15 319\r\n",
      "59\t390\t0.2\t2\t20 12 358\r\n",
      "60\t300\t0.2\t2\t9 9 282\r\n",
      "61\t302\t0.2\t2\t30 8 264\r\n",
      "62\t222\t0.2\t2\t20 5 197\r\n",
      "63\t468\t0.2\t2\t4 16 448\r\n",
      "64\t352\t0.2\t2\t2 5 345\r\n",
      "65\t251\t0.2\t2\t2 21 228\r\n",
      "66\t251\t0.2\t2\t1 7 243\r\n",
      "67\t190\t0.2\t2\t2 13 175\r\n",
      "68\t1001\t0.2\t2\t1 44 956\r\n",
      "69\t268\t0.2\t2\t0 19 249\r\n",
      "70\t273\t0.2\t2\t0 4 269\r\n",
      "71\t93\t0.2\t2\t0 0 93\r\n",
      "72\t122\t0.2\t2\t0 6 116\r\n",
      "73\t409\t0.2\t2\t0 18 391\r\n",
      "74\t397\t0.2\t2\t0 11 386\r\n",
      "75\t190\t0.2\t2\t0 17 173\r\n",
      "76\t177\t0.2\t2\t0 5 172\r\n",
      "\r\n",
      "=== Second read: Adapter 2 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 869705 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 14.4%\r\n",
      "  C: 37.3%\r\n",
      "  G: 25.3%\r\n",
      "  T: 23.0%\r\n",
      "  none/other: 0.0%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t39911\t41382.7\t0\t39911\r\n",
      "4\t41929\t10345.7\t0\t24239 17690\r\n",
      "5\t44550\t2586.4\t1\t17354 27196\r\n",
      "6\t29312\t646.6\t1\t17555 11757\r\n",
      "7\t20514\t161.7\t1\t17260 3254\r\n",
      "8\t21588\t40.4\t1\t17853 3384 351\r\n",
      "9\t20224\t10.1\t1\t18109 535 1580\r\n",
      "10\t19295\t2.5\t2\t17630 549 1116\r\n",
      "11\t19276\t0.6\t2\t17931 515 830\r\n",
      "12\t18710\t0.2\t2\t17999 356 355\r\n",
      "13\t19345\t0.2\t2\t18700 368 277\r\n",
      "14\t19462\t0.2\t2\t18846 327 289\r\n",
      "15\t20026\t0.2\t2\t19417 346 263\r\n",
      "16\t19693\t0.2\t2\t19071 319 303\r\n",
      "17\t19827\t0.2\t2\t19221 322 284\r\n",
      "18\t20043\t0.2\t2\t19420 307 316\r\n",
      "19\t20962\t0.2\t2\t20269 312 381\r\n",
      "20\t21444\t0.2\t2\t20851 334 259\r\n",
      "21\t21273\t0.2\t2\t20629 351 293\r\n",
      "22\t20599\t0.2\t2\t19971 334 294\r\n",
      "23\t19926\t0.2\t2\t19274 296 356\r\n",
      "24\t21601\t0.2\t2\t20989 347 265\r\n",
      "25\t24448\t0.2\t2\t23848 357 243\r\n",
      "26\t24487\t0.2\t2\t23897 308 282\r\n",
      "27\t21356\t0.2\t2\t20848 288 220\r\n",
      "28\t20433\t0.2\t2\t19879 281 273\r\n",
      "29\t21379\t0.2\t2\t20849 280 250\r\n",
      "30\t22916\t0.2\t2\t22363 314 239\r\n",
      "31\t23302\t0.2\t2\t22779 286 237\r\n",
      "32\t22822\t0.2\t2\t22212 348 262\r\n",
      "33\t20975\t0.2\t2\t20281 268 426\r\n",
      "34\t19917\t0.2\t2\t19369 226 322\r\n",
      "35\t24921\t0.2\t2\t24338 343 240\r\n",
      "36\t26179\t0.2\t2\t25557 341 281\r\n",
      "37\t18136\t0.2\t2\t17662 242 232\r\n",
      "38\t13679\t0.2\t2\t13276 203 200\r\n",
      "39\t10475\t0.2\t2\t9980 152 343\r\n",
      "40\t8104\t0.2\t2\t7725 113 266\r\n",
      "41\t4475\t0.2\t2\t4154 69 252\r\n",
      "42\t2565\t0.2\t2\t2025 62 478\r\n",
      "43\t1553\t0.2\t2\t1216 34 303\r\n",
      "44\t991\t0.2\t2\t729 21 241\r\n",
      "45\t1024\t0.2\t2\t829 24 171\r\n",
      "46\t2612\t0.2\t2\t2297 52 263\r\n",
      "47\t1913\t0.2\t2\t1619 25 269\r\n",
      "48\t1006\t0.2\t2\t798 46 162\r\n",
      "49\t960\t0.2\t2\t607 26 327\r\n",
      "50\t692\t0.2\t2\t485 18 189\r\n",
      "51\t575\t0.2\t2\t71 22 482\r\n",
      "52\t295\t0.2\t2\t47 14 234\r\n",
      "53\t528\t0.2\t2\t39 38 451\r\n",
      "54\t324\t0.2\t2\t43 19 262\r\n",
      "55\t309\t0.2\t2\t56 14 239\r\n",
      "56\t452\t0.2\t2\t77 20 355\r\n",
      "57\t329\t0.2\t2\t66 12 251\r\n",
      "58\t345\t0.2\t2\t27 25 293\r\n",
      "59\t401\t0.2\t2\t16 16 369\r\n",
      "60\t263\t0.2\t2\t7 9 247\r\n",
      "61\t280\t0.2\t2\t25 10 245\r\n",
      "62\t235\t0.2\t2\t18 7 210\r\n",
      "63\t502\t0.2\t2\t4 14 484\r\n",
      "64\t356\t0.2\t2\t0 13 343\r\n",
      "65\t237\t0.2\t2\t2 11 224\r\n",
      "66\t301\t0.2\t2\t1 7 293\r\n",
      "67\t202\t0.2\t2\t1 16 185\r\n",
      "68\t973\t0.2\t2\t0 43 930\r\n",
      "69\t249\t0.2\t2\t0 24 225\r\n",
      "70\t307\t0.2\t2\t0 6 301\r\n",
      "71\t120\t0.2\t2\t0 3 117\r\n",
      "72\t114\t0.2\t2\t0 0 114\r\n",
      "73\t423\t0.2\t2\t0 20 403\r\n",
      "74\t389\t0.2\t2\t0 7 382\r\n",
      "75\t188\t0.2\t2\t0 14 174\r\n",
      "76\t178\t0.2\t2\t0 3 175\r\n",
      "\r\n",
      "cutadapt -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /srv/scratch/training_camp/tc2016/ubuntu/analysis//trimmed/Ct_800_S9_R1.trimmed.fastq.gz -p /srv/scratch/training_camp/tc2016/ubuntu/analysis//trimmed/Ct_800_S9_R2.trimmed.fastq.gz /srv/scratch/training_camp/tc2016/ubuntu/data/fastq/Ct_800_S9_R1.fastq.gz /srv/scratch/training_camp/tc2016/ubuntu/data/fastq/Ct_800_S9_R2.fastq.gz\r\n",
      "This is cutadapt 1.10 with Python 3.5.2\r\n",
      "Command line parameters: -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA -o /srv/scratch/training_camp/tc2016/ubuntu/analysis//trimmed/Ct_800_S9_R1.trimmed.fastq.gz -p /srv/scratch/training_camp/tc2016/ubuntu/analysis//trimmed/Ct_800_S9_R2.trimmed.fastq.gz /srv/scratch/training_camp/tc2016/ubuntu/data/fastq/Ct_800_S9_R1.fastq.gz /srv/scratch/training_camp/tc2016/ubuntu/data/fastq/Ct_800_S9_R2.fastq.gz\r\n",
      "Trimming 2 adapters with at most 20.0% errors in paired-end mode ...\r\n",
      "Finished in 674.42 s (183 us/read; 0.33 M reads/minute).\r\n",
      "\r\n",
      "=== Summary ===\r\n",
      "\r\n",
      "Total read pairs processed:          3,676,166\r\n",
      "  Read 1 with adapter:               1,333,350 (36.3%)\r\n",
      "  Read 2 with adapter:               1,326,286 (36.1%)\r\n",
      "Pairs that were too short:               3,868 (0.1%)\r\n",
      "Pairs written (passing filters):     3,672,298 (99.9%)\r\n",
      "\r\n",
      "Total basepairs processed:   558,777,232 bp\r\n",
      "  Read 1:   279,388,616 bp\r\n",
      "  Read 2:   279,388,616 bp\r\n",
      "Total written (filtered):    499,010,029 bp (89.3%)\r\n",
      "  Read 1:   249,445,043 bp\r\n",
      "  Read 2:   249,564,986 bp\r\n",
      "\r\n",
      "=== First read: Adapter 1 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 1333350 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 14.2%\r\n",
      "  C: 37.3%\r\n",
      "  G: 24.3%\r\n",
      "  T: 24.1%\r\n",
      "  none/other: 0.0%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t51504\t57440.1\t0\t51504\r\n",
      "4\t54364\t14360.0\t0\t31336 23028\r\n",
      "5\t58203\t3590.0\t1\t23181 35022\r\n",
      "6\t38503\t897.5\t1\t22975 15528\r\n",
      "7\t26441\t224.4\t1\t22465 3976\r\n",
      "8\t27253\t56.1\t1\t23440 3355 458\r\n",
      "9\t27314\t14.0\t1\t24542 686 2086\r\n",
      "10\t25965\t3.5\t2\t24217 449 1299\r\n",
      "11\t26702\t0.9\t2\t25113 480 1109\r\n",
      "12\t26122\t0.2\t2\t25388 326 408\r\n",
      "13\t27043\t0.2\t2\t26313 319 411\r\n",
      "14\t27669\t0.2\t2\t26936 347 386\r\n",
      "15\t28381\t0.2\t2\t27723 331 327\r\n",
      "16\t27208\t0.2\t2\t26468 339 401\r\n",
      "17\t28262\t0.2\t2\t27605 285 372\r\n",
      "18\t28199\t0.2\t2\t27530 281 388\r\n",
      "19\t30214\t0.2\t2\t29359 307 548\r\n",
      "20\t31247\t0.2\t2\t30556 310 381\r\n",
      "21\t31500\t0.2\t2\t30764 362 374\r\n",
      "22\t31026\t0.2\t2\t30350 307 369\r\n",
      "23\t29618\t0.2\t2\t28883 282 453\r\n",
      "24\t33877\t0.2\t2\t33139 339 399\r\n",
      "25\t39604\t0.2\t2\t38923 363 318\r\n",
      "26\t38408\t0.2\t2\t37689 352 367\r\n",
      "27\t32640\t0.2\t2\t32080 307 253\r\n",
      "28\t31534\t0.2\t2\t30883 277 374\r\n",
      "29\t34771\t0.2\t2\t34073 316 382\r\n",
      "30\t37541\t0.2\t2\t36872 309 360\r\n",
      "31\t37474\t0.2\t2\t36822 344 308\r\n",
      "32\t37240\t0.2\t2\t36536 373 331\r\n",
      "33\t34871\t0.2\t2\t34087 271 513\r\n",
      "34\t35330\t0.2\t2\t34586 296 448\r\n",
      "35\t49405\t0.2\t2\t48700 412 293\r\n",
      "36\t52654\t0.2\t2\t51826 440 388\r\n",
      "37\t34841\t0.2\t2\t34244 284 313\r\n",
      "38\t26942\t0.2\t2\t26439 205 298\r\n",
      "39\t21398\t0.2\t2\t20732 189 477\r\n",
      "40\t16376\t0.2\t2\t15904 127 345\r\n",
      "41\t9055\t0.2\t2\t8609 80 366\r\n",
      "42\t4955\t0.2\t2\t4164 70 721\r\n",
      "43\t3054\t0.2\t2\t2615 37 402\r\n",
      "44\t2148\t0.2\t2\t1733 30 385\r\n",
      "45\t2613\t0.2\t2\t2287 39 287\r\n",
      "46\t7640\t0.2\t2\t7189 77 374\r\n",
      "47\t5305\t0.2\t2\t4944 60 301\r\n",
      "48\t2755\t0.2\t2\t2459 44 252\r\n",
      "49\t2458\t0.2\t2\t1921 29 508\r\n",
      "50\t1655\t0.2\t2\t1375 20 260\r\n",
      "51\t866\t0.2\t2\t145 22 699\r\n",
      "52\t387\t0.2\t2\t63 12 312\r\n",
      "53\t890\t0.2\t2\t37 51 802\r\n",
      "54\t469\t0.2\t2\t48 11 410\r\n",
      "55\t389\t0.2\t2\t46 18 325\r\n",
      "56\t872\t0.2\t2\t99 28 745\r\n",
      "57\t588\t0.2\t2\t177 10 401\r\n",
      "58\t523\t0.2\t2\t62 19 442\r\n",
      "59\t583\t0.2\t2\t18 17 548\r\n",
      "60\t458\t0.2\t2\t16 9 433\r\n",
      "61\t530\t0.2\t2\t24 16 490\r\n",
      "62\t345\t0.2\t2\t10 8 327\r\n",
      "63\t761\t0.2\t2\t4 11 746\r\n",
      "64\t550\t0.2\t2\t0 15 535\r\n",
      "65\t402\t0.2\t2\t0 25 377\r\n",
      "66\t465\t0.2\t2\t0 4 461\r\n",
      "67\t271\t0.2\t2\t1 21 249\r\n",
      "68\t1682\t0.2\t2\t0 76 1606\r\n",
      "69\t408\t0.2\t2\t1 22 385\r\n",
      "70\t529\t0.2\t2\t0 6 523\r\n",
      "71\t179\t0.2\t2\t0 3 176\r\n",
      "72\t197\t0.2\t2\t0 2 195\r\n",
      "73\t561\t0.2\t2\t0 15 546\r\n",
      "74\t623\t0.2\t2\t0 10 613\r\n",
      "75\t277\t0.2\t2\t0 23 254\r\n",
      "76\t263\t0.2\t2\t3 2 258\r\n",
      "\r\n",
      "=== Second read: Adapter 2 ===\r\n",
      "\r\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 1326286 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-12 bp: 2\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 14.7%\r\n",
      "  C: 35.8%\r\n",
      "  G: 24.9%\r\n",
      "  T: 24.6%\r\n",
      "  none/other: 0.0%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t52084\t57440.1\t0\t52084\r\n",
      "4\t53384\t14360.0\t0\t30668 22716\r\n",
      "5\t56578\t3590.0\t1\t22077 34501\r\n",
      "6\t37778\t897.5\t1\t22360 15418\r\n",
      "7\t26140\t224.4\t1\t21605 4535\r\n",
      "8\t26988\t56.1\t1\t23168 3350 470\r\n",
      "9\t27145\t14.0\t1\t24310 756 2079\r\n",
      "10\t25939\t3.5\t2\t23678 746 1515\r\n",
      "11\t26532\t0.9\t2\t24848 567 1117\r\n",
      "12\t25989\t0.2\t2\t25055 458 476\r\n",
      "13\t26912\t0.2\t2\t25973 459 480\r\n",
      "14\t27518\t0.2\t2\t26593 474 451\r\n",
      "15\t28273\t0.2\t2\t27438 430 405\r\n",
      "16\t27131\t0.2\t2\t26156 478 497\r\n",
      "17\t28115\t0.2\t2\t27246 423 446\r\n",
      "18\t28079\t0.2\t2\t27207 425 447\r\n",
      "19\t30042\t0.2\t2\t29075 430 537\r\n",
      "20\t31133\t0.2\t2\t30241 463 429\r\n",
      "21\t31378\t0.2\t2\t30469 462 447\r\n",
      "22\t30962\t0.2\t2\t30078 411 473\r\n",
      "23\t29549\t0.2\t2\t28613 402 534\r\n",
      "24\t33769\t0.2\t2\t32853 448 468\r\n",
      "25\t39467\t0.2\t2\t38602 511 354\r\n",
      "26\t38293\t0.2\t2\t37321 543 429\r\n",
      "27\t32566\t0.2\t2\t31775 436 355\r\n",
      "28\t31448\t0.2\t2\t30589 401 458\r\n",
      "29\t34633\t0.2\t2\t33795 462 376\r\n",
      "30\t37427\t0.2\t2\t36569 443 415\r\n",
      "31\t37420\t0.2\t2\t36536 452 432\r\n",
      "32\t37124\t0.2\t2\t36217 510 397\r\n",
      "33\t34848\t0.2\t2\t33816 412 620\r\n",
      "34\t35226\t0.2\t2\t34303 434 489\r\n",
      "35\t49288\t0.2\t2\t48281 605 402\r\n",
      "36\t52490\t0.2\t2\t51412 649 429\r\n",
      "37\t34736\t0.2\t2\t33967 399 370\r\n",
      "38\t26851\t0.2\t2\t26200 316 335\r\n",
      "39\t21289\t0.2\t2\t20570 241 478\r\n",
      "40\t16359\t0.2\t2\t15749 214 396\r\n",
      "41\t9046\t0.2\t2\t8533 126 387\r\n",
      "42\t5009\t0.2\t2\t4118 99 792\r\n",
      "43\t3092\t0.2\t2\t2586 48 458\r\n",
      "44\t2134\t0.2\t2\t1709 37 388\r\n",
      "45\t2666\t0.2\t2\t2267 39 360\r\n",
      "46\t7644\t0.2\t2\t7143 93 408\r\n",
      "47\t5317\t0.2\t2\t4913 71 333\r\n",
      "48\t2764\t0.2\t2\t2431 61 272\r\n",
      "49\t2403\t0.2\t2\t1893 40 470\r\n",
      "50\t1664\t0.2\t2\t1353 37 274\r\n",
      "51\t793\t0.2\t2\t141 23 629\r\n",
      "52\t408\t0.2\t2\t61 11 336\r\n",
      "53\t809\t0.2\t2\t43 45 721\r\n",
      "54\t462\t0.2\t2\t45 18 399\r\n",
      "55\t376\t0.2\t2\t45 16 315\r\n",
      "56\t771\t0.2\t2\t97 23 651\r\n",
      "57\t556\t0.2\t2\t165 15 376\r\n",
      "58\t504\t0.2\t2\t59 22 423\r\n",
      "59\t530\t0.2\t2\t16 26 488\r\n",
      "60\t440\t0.2\t2\t13 13 414\r\n",
      "61\t548\t0.2\t2\t16 14 518\r\n",
      "62\t326\t0.2\t2\t7 11 308\r\n",
      "63\t732\t0.2\t2\t3 13 716\r\n",
      "64\t563\t0.2\t2\t1 7 555\r\n",
      "65\t367\t0.2\t2\t0 22 345\r\n",
      "66\t489\t0.2\t2\t0 8 481\r\n",
      "67\t311\t0.2\t2\t1 25 285\r\n",
      "68\t1613\t0.2\t2\t1 63 1549\r\n",
      "69\t417\t0.2\t2\t1 27 389\r\n",
      "70\t547\t0.2\t2\t0 9 538\r\n",
      "71\t150\t0.2\t2\t0 0 150\r\n",
      "72\t187\t0.2\t2\t0 9 178\r\n",
      "73\t604\t0.2\t2\t0 18 586\r\n",
      "74\t583\t0.2\t2\t0 12 571\r\n",
      "75\t303\t0.2\t2\t0 24 279\r\n",
      "76\t275\t0.2\t2\t3 4 268\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "for R1_fastq in ${FASTQ_DIR}*_R1*fastq.gz; do\n",
    "    \n",
    "    # Get the read 2 fastq file from the filename of read 1\n",
    "    R2_fastq=$(echo $R1_fastq | sed -e 's/R1/R2/')\n",
    "    \n",
    "    # Generate names for the trimmed fastq files\n",
    "\n",
    "    trimmed_R1_fastq=$TRIMMED_DIR$(echo $(basename $R1_fastq)| sed -e 's/.fastq.gz/.trimmed.fastq.gz/')\n",
    "    trimmed_R2_fastq=$TRIMMED_DIR$(echo $(basename $R2_fastq)| sed -e 's/.fastq.gz/.trimmed.fastq.gz/')   \n",
    "    echo cutadapt -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA \\\n",
    "        -o ${trimmed_R1_fastq} \\\n",
    "        -p ${trimmed_R2_fastq} \\\n",
    "        $R1_fastq \\\n",
    "        $R2_fastq\n",
    "    cutadapt -m 5 -e 0.20 -a CTGTCTCTTATA -A CTGTCTCTTATA \\\n",
    "        -o ${trimmed_R1_fastq} \\\n",
    "        -p ${trimmed_R2_fastq} \\\n",
    "        $R1_fastq \\\n",
    "        $R2_fastq\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": false
   },
   "source": [
    "## Part 3: Alignment\n",
    "\n",
    "Now, we're ready to align our trimmed reads to the Yeast SacCer3 reference genome.\n",
    "\n",
    "We'll use [Bowtie2](http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml), which is a [Burrows-Wheeler](https://en.wikipedia.org/wiki/Burrows%E2%80%93Wheeler_transform) based spliced aligner.\n",
    "\n",
    "Bowtie2 outputs a SAM (Sequence Alignment Map) file, which is a standard text encoding. To save space, we'll use `samtools view -b` to encode the output as a binarized SAM file — a BAM file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/srv/scratch/training_camp/saccer3/bowtie2_index/saccer3\r\n"
     ]
    }
   ],
   "source": [
    "#set the bowtie index\n",
    "export bowtie_index=$YEAST_INDEX\n",
    "echo $bowtie_index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": []
    }
   ],
   "source": [
    "#create a directory to store the aligned data \n",
    "export ALIGNMENT_DIR=\"$ANALYSIS_DIR/aligned/\"\n",
    "[[ ! -d $ALIGNMENT_DIR ]] && mkdir -p \"$ALIGNMENT_DIR\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[samopen] SAM header is present: 17 sequences.\r\n",
      "153476 reads; of these:\r\n",
      "  153476 (100.00%) were paired; of these:\r\n",
      "    7206 (4.70%) aligned concordantly 0 times\r\n",
      "    75741 (49.35%) aligned concordantly exactly 1 time\r\n",
      "    70529 (45.95%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    7206 pairs aligned concordantly 0 times; of these:\r\n",
      "      1000 (13.88%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    6206 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      12412 mates make up the pairs; of these:\r\n",
      "        10043 (80.91%) aligned 0 times\r\n",
      "        232 (1.87%) aligned exactly 1 time\r\n",
      "        2137 (17.22%) aligned >1 times\r\n",
      "96.73% overall alignment rate\r\n",
      "[samopen] SAM header is present: 17 sequences.\r\n",
      "450816 reads; of these:\r\n",
      "  450816 (100.00%) were paired; of these:\r\n",
      "    18997 (4.21%) aligned concordantly 0 times\r\n",
      "    190331 (42.22%) aligned concordantly exactly 1 time\r\n",
      "    241488 (53.57%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    18997 pairs aligned concordantly 0 times; of these:\r\n",
      "      2522 (13.28%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    16475 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      32950 mates make up the pairs; of these:\r\n",
      "        24797 (75.26%) aligned 0 times\r\n",
      "        788 (2.39%) aligned exactly 1 time\r\n",
      "        7365 (22.35%) aligned >1 times\r\n",
      "97.25% overall alignment rate\r\n",
      "[samopen] SAM header is present: 17 sequences.\r\n",
      "3925710 reads; of these:\r\n",
      "  3925710 (100.00%) were paired; of these:\r\n",
      "    273596 (6.97%) aligned concordantly 0 times\r\n",
      "    1778827 (45.31%) aligned concordantly exactly 1 time\r\n",
      "    1873287 (47.72%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    273596 pairs aligned concordantly 0 times; of these:\r\n",
      "      26806 (9.80%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    246790 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      493580 mates make up the pairs; of these:\r\n",
      "        402256 (81.50%) aligned 0 times\r\n",
      "        17544 (3.55%) aligned exactly 1 time\r\n",
      "        73780 (14.95%) aligned >1 times\r\n",
      "94.88% overall alignment rate\r\n",
      "[samopen] SAM header is present: 17 sequences.\r\n",
      "3025310 reads; of these:\r\n",
      "  3025310 (100.00%) were paired; of these:\r\n",
      "    205484 (6.79%) aligned concordantly 0 times\r\n",
      "    1367132 (45.19%) aligned concordantly exactly 1 time\r\n",
      "    1452694 (48.02%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    205484 pairs aligned concordantly 0 times; of these:\r\n",
      "      22358 (10.88%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    183126 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      366252 mates make up the pairs; of these:\r\n",
      "        296087 (80.84%) aligned 0 times\r\n",
      "        12368 (3.38%) aligned exactly 1 time\r\n",
      "        57797 (15.78%) aligned >1 times\r\n",
      "95.11% overall alignment rate\r\n",
      "[samopen] SAM header is present: 17 sequences.\r\n",
      "5265654 reads; of these:\r\n",
      "  5265654 (100.00%) were paired; of these:\r\n",
      "    313855 (5.96%) aligned concordantly 0 times\r\n",
      "    2411180 (45.79%) aligned concordantly exactly 1 time\r\n",
      "    2540619 (48.25%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    313855 pairs aligned concordantly 0 times; of these:\r\n",
      "      35277 (11.24%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    278578 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      557156 mates make up the pairs; of these:\r\n",
      "        442853 (79.48%) aligned 0 times\r\n",
      "        23691 (4.25%) aligned exactly 1 time\r\n",
      "        90612 (16.26%) aligned >1 times\r\n",
      "95.79% overall alignment rate\r\n",
      "[samopen] SAM header is present: 17 sequences.\r\n",
      "2645906 reads; of these:\r\n",
      "  2645906 (100.00%) were paired; of these:\r\n",
      "    158288 (5.98%) aligned concordantly 0 times\r\n",
      "    1181733 (44.66%) aligned concordantly exactly 1 time\r\n",
      "    1305885 (49.35%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    158288 pairs aligned concordantly 0 times; of these:\r\n",
      "      16937 (10.70%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    141351 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      282702 mates make up the pairs; of these:\r\n",
      "        223749 (79.15%) aligned 0 times\r\n",
      "        10568 (3.74%) aligned exactly 1 time\r\n",
      "        48385 (17.12%) aligned >1 times\r\n",
      "95.77% overall alignment rate\r\n",
      "[samopen] SAM header is present: 17 sequences.\r\n",
      "3672298 reads; of these:\r\n",
      "  3672298 (100.00%) were paired; of these:\r\n",
      "    237694 (6.47%) aligned concordantly 0 times\r\n",
      "    1865980 (50.81%) aligned concordantly exactly 1 time\r\n",
      "    1568624 (42.72%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    237694 pairs aligned concordantly 0 times; of these:\r\n",
      "      25612 (10.78%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    212082 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      424164 mates make up the pairs; of these:\r\n",
      "        350929 (82.73%) aligned 0 times\r\n",
      "        18495 (4.36%) aligned exactly 1 time\r\n",
      "        54740 (12.91%) aligned >1 times\r\n",
      "95.22% overall alignment rate\r\n",
      "[samopen] SAM header is present: 17 sequences.\r\n",
      "18854 reads; of these:\r\n",
      "  18854 (100.00%) were paired; of these:\r\n",
      "    757 (4.02%) aligned concordantly 0 times\r\n",
      "    10642 (56.44%) aligned concordantly exactly 1 time\r\n",
      "    7455 (39.54%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    757 pairs aligned concordantly 0 times; of these:\r\n",
      "      93 (12.29%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    664 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      1328 mates make up the pairs; of these:\r\n",
      "        1078 (81.17%) aligned 0 times\r\n",
      "        79 (5.95%) aligned exactly 1 time\r\n",
      "        171 (12.88%) aligned >1 times\r\n",
      "97.14% overall alignment rate\r\n",
      "[samopen] SAM header is present: 17 sequences.\r\n",
      "434218 reads; of these:\r\n",
      "  434218 (100.00%) were paired; of these:\r\n",
      "    18552 (4.27%) aligned concordantly 0 times\r\n",
      "    210067 (48.38%) aligned concordantly exactly 1 time\r\n",
      "    205599 (47.35%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    18552 pairs aligned concordantly 0 times; of these:\r\n",
      "      2759 (14.87%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    15793 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      31586 mates make up the pairs; of these:\r\n",
      "        24340 (77.06%) aligned 0 times\r\n",
      "        870 (2.75%) aligned exactly 1 time\r\n",
      "        6376 (20.19%) aligned >1 times\r\n",
      "97.20% overall alignment rate\r\n",
      "[samopen] SAM header is present: 17 sequences.\r\n",
      "157271 reads; of these:\r\n",
      "  157271 (100.00%) were paired; of these:\r\n",
      "    7658 (4.87%) aligned concordantly 0 times\r\n",
      "    78181 (49.71%) aligned concordantly exactly 1 time\r\n",
      "    71432 (45.42%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    7658 pairs aligned concordantly 0 times; of these:\r\n",
      "      1081 (14.12%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    6577 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      13154 mates make up the pairs; of these:\r\n",
      "        10689 (81.26%) aligned 0 times\r\n",
      "        245 (1.86%) aligned exactly 1 time\r\n",
      "        2220 (16.88%) aligned >1 times\r\n",
      "96.60% overall alignment rate\r\n",
      "[samopen] SAM header is present: 17 sequences.\r\n",
      "416040 reads; of these:\r\n",
      "  416040 (100.00%) were paired; of these:\r\n",
      "    19109 (4.59%) aligned concordantly 0 times\r\n",
      "    216184 (51.96%) aligned concordantly exactly 1 time\r\n",
      "    180747 (43.44%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    19109 pairs aligned concordantly 0 times; of these:\r\n",
      "      2765 (14.47%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    16344 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      32688 mates make up the pairs; of these:\r\n",
      "        26079 (79.78%) aligned 0 times\r\n",
      "        989 (3.03%) aligned exactly 1 time\r\n",
      "        5620 (17.19%) aligned >1 times\r\n",
      "96.87% overall alignment rate\r\n",
      "[samopen] SAM header is present: 17 sequences.\r\n",
      "127700 reads; of these:\r\n",
      "  127700 (100.00%) were paired; of these:\r\n",
      "    6001 (4.70%) aligned concordantly 0 times\r\n",
      "    65373 (51.19%) aligned concordantly exactly 1 time\r\n",
      "    56326 (44.11%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    6001 pairs aligned concordantly 0 times; of these:\r\n",
      "      721 (12.01%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    5280 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      10560 mates make up the pairs; of these:\r\n",
      "        8753 (82.89%) aligned 0 times\r\n",
      "        221 (2.09%) aligned exactly 1 time\r\n",
      "        1586 (15.02%) aligned >1 times\r\n",
      "96.57% overall alignment rate\r\n",
      "[samopen] SAM header is present: 17 sequences.\r\n",
      "432071 reads; of these:\r\n",
      "  432071 (100.00%) were paired; of these:\r\n",
      "    20183 (4.67%) aligned concordantly 0 times\r\n",
      "    205449 (47.55%) aligned concordantly exactly 1 time\r\n",
      "    206439 (47.78%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    20183 pairs aligned concordantly 0 times; of these:\r\n",
      "      2732 (13.54%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    17451 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      34902 mates make up the pairs; of these:\r\n",
      "        27517 (78.84%) aligned 0 times\r\n",
      "        921 (2.64%) aligned exactly 1 time\r\n",
      "        6464 (18.52%) aligned >1 times\r\n",
      "96.82% overall alignment rate\r\n",
      "[samopen] SAM header is present: 17 sequences.\r\n",
      "62968 reads; of these:\r\n",
      "  62968 (100.00%) were paired; of these:\r\n",
      "    3122 (4.96%) aligned concordantly 0 times\r\n",
      "    30519 (48.47%) aligned concordantly exactly 1 time\r\n",
      "    29327 (46.57%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    3122 pairs aligned concordantly 0 times; of these:\r\n",
      "      399 (12.78%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    2723 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      5446 mates make up the pairs; of these:\r\n",
      "        4479 (82.24%) aligned 0 times\r\n",
      "        106 (1.95%) aligned exactly 1 time\r\n",
      "        861 (15.81%) aligned >1 times\r\n",
      "96.44% overall alignment rate\r\n",
      "[samopen] SAM header is present: 17 sequences.\r\n",
      "356973 reads; of these:\r\n",
      "  356973 (100.00%) were paired; of these:\r\n",
      "    15313 (4.29%) aligned concordantly 0 times\r\n",
      "    169286 (47.42%) aligned concordantly exactly 1 time\r\n",
      "    172374 (48.29%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    15313 pairs aligned concordantly 0 times; of these:\r\n",
      "      2298 (15.01%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    13015 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      26030 mates make up the pairs; of these:\r\n",
      "        20142 (77.38%) aligned 0 times\r\n",
      "        671 (2.58%) aligned exactly 1 time\r\n",
      "        5217 (20.04%) aligned >1 times\r\n",
      "97.18% overall alignment rate\r\n",
      "[samopen] SAM header is present: 17 sequences.\r\n",
      "209758 reads; of these:\r\n",
      "  209758 (100.00%) were paired; of these:\r\n",
      "    9720 (4.63%) aligned concordantly 0 times\r\n",
      "    112594 (53.68%) aligned concordantly exactly 1 time\r\n",
      "    87444 (41.69%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    9720 pairs aligned concordantly 0 times; of these:\r\n",
      "      1534 (15.78%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    8186 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      16372 mates make up the pairs; of these:\r\n",
      "        13204 (80.65%) aligned 0 times\r\n",
      "        422 (2.58%) aligned exactly 1 time\r\n",
      "        2746 (16.77%) aligned >1 times\r\n",
      "96.85% overall alignment rate\r\n",
      "[samopen] SAM header is present: 17 sequences.\r\n",
      "324846 reads; of these:\r\n",
      "  324846 (100.00%) were paired; of these:\r\n",
      "    15358 (4.73%) aligned concordantly 0 times\r\n",
      "    172373 (53.06%) aligned concordantly exactly 1 time\r\n",
      "    137115 (42.21%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    15358 pairs aligned concordantly 0 times; of these:\r\n",
      "      2294 (14.94%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    13064 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      26128 mates make up the pairs; of these:\r\n",
      "        21026 (80.47%) aligned 0 times\r\n",
      "        812 (3.11%) aligned exactly 1 time\r\n",
      "        4290 (16.42%) aligned >1 times\r\n",
      "96.76% overall alignment rate\r\n",
      "[samopen] SAM header is present: 17 sequences.\r\n",
      "88746 reads; of these:\r\n",
      "  88746 (100.00%) were paired; of these:\r\n",
      "    4442 (5.01%) aligned concordantly 0 times\r\n",
      "    45865 (51.68%) aligned concordantly exactly 1 time\r\n",
      "    38439 (43.31%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    4442 pairs aligned concordantly 0 times; of these:\r\n",
      "      504 (11.35%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    3938 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      7876 mates make up the pairs; of these:\r\n",
      "        6577 (83.51%) aligned 0 times\r\n",
      "        142 (1.80%) aligned exactly 1 time\r\n",
      "        1157 (14.69%) aligned >1 times\r\n",
      "96.29% overall alignment rate\r\n",
      "[samopen] SAM header is present: 17 sequences.\r\n",
      "442758 reads; of these:\r\n",
      "  442758 (100.00%) were paired; of these:\r\n",
      "    20067 (4.53%) aligned concordantly 0 times\r\n",
      "    184021 (41.56%) aligned concordantly exactly 1 time\r\n",
      "    238670 (53.91%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    20067 pairs aligned concordantly 0 times; of these:\r\n",
      "      2535 (12.63%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    17532 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      35064 mates make up the pairs; of these:\r\n",
      "        27256 (77.73%) aligned 0 times\r\n",
      "        803 (2.29%) aligned exactly 1 time\r\n",
      "        7005 (19.98%) aligned >1 times\r\n",
      "96.92% overall alignment rate\r\n"
     ]
    }
   ],
   "source": [
    "for trimmed_fq1 in ${TRIMMED_DIR}*_R1*fastq.gz; do\n",
    "\n",
    "    trimmed_fq2=$(echo $trimmed_fq1 | sed -e 's/_R1/_R2/')\n",
    "    \n",
    "    bam=$(echo \"${ALIGNMENT_DIR}${trimmed_fq1##*/}\" | sed -e 's/.fastq.gz/.bam/')\n",
    "    bowtie2 -X2000 --mm --threads 10 -x $bowtie_index -1 $trimmed_fq1 -2 $trimmed_fq2 | samtools view -bS - > $bam        \n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Part 4: Finding duplicate reads and alignment filtering\n",
    "\n",
    "During sequencing, we perform PCR, which can lead to duplicate reads. In many kinds of DNA sequencing, we want to remove duplicates so that we don't double-count signal originating from the same molecule.\n",
    "\n",
    "To do so, we use an algorithm called `sambamba` that looks for reads that mapped to exactly the same places in the genome. We also need to sort the aligned files before we can mark duplicates, since we need reads aligned to the same position to be next to each other in the file.\n",
    "\n",
    "Bowtie2 also sets certian labels (or \"flags\") in the resulting alignment file to indicate information like the score of the alignment, the orientation of both mates of the fragment, and other details.\n",
    "\n",
    "We can use these flags as a way to discard low-quality reads. [This website](https://broadinstitute.github.io/picard/explain-flags.html) provides a convenient way to interpret the meaning of these bitwise flags; for conveninece they can be encoded as numbers.\n",
    "\n",
    "Here, we want to filter reads that fall into any of the following categories:\n",
    "- the read wasn't mapped to the genome\n",
    "- the read's mate wasn't mapped to the genome\n",
    "- the alignment reported is not the primary alignment (it is a \"runner-up\" alignment)\n",
    "- the read was marked as \"low-quality\" by the sequencer software\n",
    "- the read has a mapping quality less than 30"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "for bam_file in ${ALIGNMENT_DIR}*.trimmed.bam; do\n",
    "\n",
    "    bam_file_sorted=$(echo $bam_file | sed -e 's/.bam/.sorted.bam/')\n",
    "    bam_file_dup=$(echo $bam_file | sed -e 's/.bam/.sorted.dup.bam/')\n",
    "    nodup_bam_file=$(echo $bam_file | sed -e 's/.bam/.nodup.bam/')\n",
    "    \n",
    "    # Sort and remove duplicates\n",
    "    sambamba sort -m 4G -t 10 -u $bam_file \n",
    "    sambamba markdup -l 0 -t 10 $bam_file_sorted $bam_file_dup\n",
    "    samtools view -F 1804 -f 2 -q 30 -b $bam_file_dup  > $nodup_bam_file\n",
    "    \n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Part 5: Peak calling\n",
    "\n",
    "Now that we've aligned our reads to the genome and filtered the alignments, we want to identify areas of locally enriched signals, or \"peaks\".\n",
    "\n",
    "For ATAC-seq, peaks correspond to accessible regions. They can include promoters, enhancers, and other regulatory regions.\n",
    "\n",
    "We'll call peaks using [MACS2](http://liulab.dfci.harvard.edu/MACS/)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#create a directory to store the tagAlign data \n",
    "TAGALIGN_DIR=\"${ANALYSIS_DIR}tagAlign/\"\n",
    "[[ ! -d $TAGALIGN_DIR ]] && mkdir -p \"$TAGALIGN_DIR\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#create a directory to store the MACS peaks \n",
    "PEAKS_DIR=\"${ANALYSIS_DIR}peaks/\"\n",
    "[[ ! -d $PEAKS_DIR ]] && mkdir -p \"$PEAKS_DIR\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "SacCer3GenSz=12157105  # The sum of the sizes of the chromosomes in the SacCer3 genome\n",
    "\n",
    "Macs2PvalThresh=\"0.05\"  # The p-value threshold for calling peaks \n",
    "\n",
    "Macs2SmoothWindow=150  # The window size to smooth alignment signal over\n",
    "Macs2ShiftSize=$(python -c \"print(int(${Macs2SmoothWindow}/2))\")\n",
    "\n",
    "for nodup_bam_file in ${ALIGNMENT_DIR}*.nodup.bam; do\n",
    "    \n",
    "    # First, we need to convert each bam to a .tagAlign,\n",
    "    # which just contains the start/end positions of each read:\n",
    "    \n",
    "    tagalign_file=$TAGALIGN_DIR$(echo $(basename $nodup_bam_file) | sed -e 's/.bam/.tagAlign.gz/')\n",
    "    bedtools bamtobed -i $nodup_bam_file | awk 'BEGIN{OFS=\"\\t\"}{$4=\"N\";$5=\"1000\";print $0}' | gzip -c > $tagalign_file\n",
    "    \n",
    "    # Now, we can run MACS:\n",
    "    output_prefix=$PEAKS_DIR$(echo $(basename $tagalign_file)| sed -e 's/.tagAlign.gz//')\n",
    "     macs2 callpeak \\\n",
    "        -t $tagalign_file -f BED -n $output_prefix -g \"$SacCer3GenSz\" -p $Macs2PvalThresh \\\n",
    "        --nomodel --shift -$Macs2ShiftSize --extsize $Macs2SmoothWindow -B --SPMR --keep-dup all --call-summits\n",
    "\n",
    "    #We also generate a fold change file comparing the sample to the control(DMSO)\n",
    "    macs2 bdgcmp -t $output_prefix\\_treat_pileup.bdg -c $output_prefix\\_control_lambda.bdg -o $output_prefix\\_FE.bdg -m FE\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Finally, we merge the peaks across all conditions to create a master list of peaks for analysis. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "cd $PEAKS_DIR\n",
    "#concatenate all .narrowPeak files together \n",
    "cat *narrowPeak > all.peaks.bed \n",
    "\n",
    "#sort the concatenated file \n",
    "bedtools sort -i all.peaks.bed > all.peaks.sorted.bed \n",
    "\n",
    "#merge the sorted, concatenated fileto join overlapping peaks \n",
    "bedtools merge -i all.peaks.sorted.bed | sed -n 'p;='  | paste -d\"\\t\" - - >  all_merged.peaks.bed\n",
    "gzip -f all_merged.peaks.bed \n"
   ]
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
